---
title: "SVs Analysis, HPG intestine project"
author: "Laura Sisk-Hackworth"
date: "2023-06-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load Packages
```{r}
library(FSA)
library(ggplot2)
library(tidyverse)
library(corrplot)
library(ggpubr)
library(psych)
library(vegan)
library(nlme)
library(metap)
library(caTools)
library(randomForest)
library(ROCR)
library(car)
library(reshape2)
library(RColorBrewer)
library(ape)
library(coda4microbiome)
library(cowplot)
library(svglite)
#phyloseq
```
Colors:
Order Female Mut, Female WT, Male Mut, Male WT
current = "#332288","#E066A6", "#E6A316","#256769"
sample_type :"#E69F00","#009E73"
Cohort Current = "#CC79A7","#0072B2"

# Colors
```{r}
group_color_list <- c(Female_Mutant = "#332288", Male_Mutant = "#E6A316", 
                      Female_Wild_Type = "#E066A6", Male_Wild_Type = "#256769")
sample_type_color_list <- c(Lumen = "#E69F00", Mucosa = "#009E73")
cohort_color_list <- c(A = "#CC79A7", B = "#0072B2")
week_color_list <- c(`3` = "#feb24c", `6` = "#fc4e2a", `10` = "#800026")
section_color_list <- c(Cecum_Lumen = "#A6CEE3", Cecum_Mucosa=  "#1F78B4", Duodenum_Lumen =  "#B2DF8A", Duodenum_Mucosa =  "#33A02C", feces =  "#FB9A99" , Ileum_Lumen = "#E31A1C", Ileum_Mucosa =  "#FDBF6F", Cecum = "#A6CEE3", Duodenum = "#B2DF8A", Ileum = "#E31A1C", Feces = "#FB9A99", Unknown = "#808080")
```


# 0 Import data 
## 0.1 Metadata
```{r}
#Cecum
cecum_map_all <- read.csv("data/cecum/cecum_metadata.csv", header = TRUE, row.names = 1)
cecum_map <- cecum_map_all[7:160,]
head(cecum_map)
#since I took one of the samples out due to low sequence depth so I could have a higher rarefaction number:
cecum_map_alpha <- cecum_map[!(cecum_map$Mouse_Label=="414_CL"),]
head(cecum_map_alpha)
str(cecum_map_alpha)
#ileum
ileum_map_all <- read.csv("data/ileum/ileum_metadata.csv", header = TRUE, row.names = 1)
ileum_map <- ileum_map_all[7:160,]
head(ileum_map)
#since I took one of the samples out due to low sequence depth so I could have a higher rarefaction number:
ileum_map_alpha <- ileum_map
head(ileum_map_alpha)

#Duodenum
duodenum_map_all <- read.csv("data/duodenum/duodenum_metadata.csv", header = TRUE, row.names = 1)
duodenum_map <- duodenum_map_all[7:160,]
head(duodenum_map)
#since I took one of the samples out due to low sequence depth so I could have a higher rarefaction number:
duodenum_map_alpha <- duodenum_map[!(duodenum_map$Mouse_Label=="461_DL"),]
head(duodenum_map_alpha)
 
#feces
feces_map_week10 <- read.csv("data/feces/feces_metadata_week10.csv", header = TRUE, row.names = 1)
feces_map <- feces_map_week10
head(feces_map)

#since I took one of the samples out due to low sequence depth so I could have a higher rarefaction number:
feces_map_alpha <- feces_map[!(feces_map$Mouse_Label=="427_10"),]
head(feces_map_alpha)

all_envs_map <- rbind(duodenum_map, ileum_map, cecum_map, feces_map)
intestine_map <- rbind(duodenum_map, ileum_map, cecum_map)
```
## 0.2 Alpha Diveristy data
```{r}
#cecum
cecum_alpha_all <- read.csv("data/cecum/cecum_alpha_diversity_all_filtered.csv", header = TRUE, row.names = 1)
cecum_alpha <-cecum_alpha_all[7:159,]
tail(cecum_alpha)

ileum_alpha_all <- read.csv("data/ileum/ileum_alpha_diversity_all_filtered.csv", header = TRUE, row.names = 1)
ileum_alpha <-ileum_alpha_all[7:160,]
tail(ileum_alpha)

duodenum_alpha_all <- read.csv("data/duodenum/duodenum_alpha_diversity_all_filtered.csv", header = TRUE, row.names = 1)
duodenum_alpha <-duodenum_alpha_all[7:159,]
tail(duodenum_alpha)

feces_alpha_all <- read.csv("data/feces/feces_alpha_diversity_all_filtered.csv", header = TRUE, row.names = 1)
feces_alpha <-feces_alpha_all[10:239,]
tail(feces_alpha)
```


## 0.3 SV, Genus, and Family count tables
```{r}
#cecum
cecum_SVs_all <- read.csv("data/cecum/ASV_table_mito_removedtable.zerofiltered.csv", header = TRUE, row.names = 1)


cecum_SVs <- cecum_SVs_all[,7:160]
cecum_map <- cecum_map_all[7:160,]
head(cecum_SVs)

cecum_genus_all <- read.csv("data/cecum/cecum_genus_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)


cecum_genus <- cecum_genus_all[,7:160]
head(cecum_genus)

cecum_family_all <- read.csv("data/cecum/cecum_family_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)

cecum_family <- cecum_family_all[,7:160]
head(cecum_family)

#ileum

ileum_SVs_all <- read.csv("data/ileum/ASV_table_mito_removedtable.zerofiltered.csv", header = TRUE, row.names = 1)


ileum_SVs <- ileum_SVs_all[,7:160]
ileum_map <- ileum_map_all[7:160,]
head(ileum_SVs)

ileum_genus_all <- read.csv("data/ileum/ileum_genus_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)


ileum_genus <- ileum_genus_all[,7:160]
head(ileum_genus)

ileum_family_all <- read.csv("data/ileum/ileum_family_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)


ileum_family <- ileum_family_all[,7:160]
head(ileum_family)

#duodenum
duodenum_SVs_all <- read.csv("data/duodenum/ASV_table_mito_removedtable.zerofiltered.csv", header = TRUE, row.names = 1)


duodenum_SVs <- duodenum_SVs_all[,7:160]
duodenum_map <- duodenum_map_all[7:160,]
head(duodenum_SVs)

duodenum_genus_all <- read.csv("data/duodenum/duodenum_genus_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)


duodenum_genus <- duodenum_genus_all[,7:160]
head(duodenum_genus)

duodenum_family_all <- read.csv("data/duodenum/duodenum_family_PT_table.from_biom_simple_name.csv", header = TRUE, row.names = 1)


duodenum_family <- duodenum_family_all[,7:160]
head(duodenum_family)

#feces
feces_SVs_week10 <- read.csv("data/feces/ASV_table_mito_removedtable.zerofiltered_week10.csv", header = TRUE, row.names = 1)

feces_SVs <- feces_SVs_week10
head(feces_SVs)

feces_genus_week10 <- read.csv("data/feces/feces_genus_PT_table.from_biom_simple_name_week10.csv", header = TRUE, row.names = 1)


feces_genus <- feces_genus_week10 
head(feces_genus)

feces_family_week10 <- read.csv("data/feces/feces_family_PT_table.from_biom_simple_name_week10.csv", header = TRUE, row.names = 1)


feces_family <- feces_family_week10
head(feces_family)
```

All Intestine data (no feces)
All Intestine Samples
```{r}
#SVs
duo_ile_SVs <- merge(duodenum_SVs, ileum_SVs, by = 'row.names', all = TRUE)
rownames(duo_ile_SVs) <- duo_ile_SVs$Row.names
duo_ile_SVs <- duo_ile_SVs[, 2:309]
dim(duo_ile_SVs)

intestine_SVs_names <- merge(duo_ile_SVs, cecum_SVs, by = 'row.names', all = TRUE)
rownames(intestine_SVs_names) <- intestine_SVs_names$Row.names

intestine_SVs <- intestine_SVs_names[, 2:463]
dim(intestine_SVs)
dim(intestine_map)

intestine_SVs[is.na(intestine_SVs)] <- 0


#Genus
duo_ile_genus <- merge(duodenum_genus, ileum_genus, by = 'row.names', all = TRUE)
rownames(duo_ile_genus) <- duo_ile_genus$Row.names
duo_ile_genus <- duo_ile_genus[, 2:309]
dim(duo_ile_genus)

intestine_genus_names <- merge(duo_ile_genus, cecum_genus, by = 'row.names', all = TRUE)
rownames(intestine_genus_names) <- intestine_genus_names$Row.names

intestine_genus <- intestine_genus_names[, 2:463]
dim(intestine_genus)
dim(intestine_map)

intestine_genus[is.na(intestine_genus)] <- 0

#Family
duo_ile_family <- merge(duodenum_family, ileum_family, by = 'row.names', all = TRUE)
rownames(duo_ile_family) <- duo_ile_family$Row.names
duo_ile_family <- duo_ile_family[, 2:309]
dim(duo_ile_family)

intestine_family_names <- merge(duo_ile_family, cecum_family, by = 'row.names', all = TRUE)
rownames(intestine_family_names) <- intestine_family_names$Row.names

intestine_family <- intestine_family_names[, 2:463]
dim(intestine_family)
dim(intestine_map)

intestine_family[is.na(intestine_family)] <- 0
```

For all family data (all sample types, environments), make color palette for barplots to use in all the environments 
Get unique family names from all environments
```{r}
#unique_family <-sort(unique(c(row.names(duodenum_family), row.names(ileum_family), row.names(cecum_family), row.names(feces_family))))
#length(unique_family)
#print(unique_family)
```
Colors
```{r}
family_color_list <- c(Acholeplasmataceae = "#A6CEE3",Actinomycetaceae = "#1F78B4",
                       Aerococcaceae = "#B2DF8A", Alcaligenaceae = "#33A02C", 
                       Anaerovoracaceae = "#A6CEE3", Bacillaceae = "#B15928", 
                       Bacteroidaceae =  "#FF7F00"  ,  
                       Bacteroidales_family = "#1F78B4", 
                       Beijerinckiaceae = "#1F78B4", Bifidobacteriaceae = "#A6CEE3",
                       Burkholderiaceae = "#B2DF8A", Butyricicoccaceae = "#FB9A99",
                       Caulobacteraceae = "#1F78B4", 
                       `Clostridia_UCG-014` = "#A6CEE3",
                       Clostridia_vadinBB60_group = "#1F78B4", 
                       Clostridiaceae = "#FDBF6F", 
                       Comamonadaceae = "#A6CEE3", Corynebacteriaceae = "#1F78B4",
                       Deferribacteraceae = "#B2DF8A", Eggerthellaceae = "#E31A1C",
                       Enterobacteriaceae = "#B2DF8A", Enterococcaceae = "#33A02C",
                       Erysipelatoclostridiaceae = "#FB9A99",  
                       Erysipelotrichaceae = "#A6CEE3", Halomonadaceae = "#1F78B4",
                       Lachnospiraceae = "#A6CEE3", Lactobacillaceae = "#1F78B4",
                       Listeriaceae = "#B2DF8A", Marinifilaceae = "#33A02C" ,
                       Monoglobaceae = "#6A3D9A" , Moraxellaceae = "#33A02C", 
                       Muribaculaceae = "#33A02C", 
                       Obscuribacteraceae = "#A6CEE3",Oscillospiraceae ="#FFFF99" ,
                       Peptococcaceae = "#1F78B4", Prevotellaceae = "#6A3D9A" ,
                       Pseudomonadaceae = "#B2DF8A", RF39 = "#33A02C", 
                       Rhizobiaceae = "#FB9A99", Rikenellaceae ="#B2DF8A",
                       Ruminococcaceae = "#CAB2D6" , 
                       Saccharimonadaceae = "#6A3D9A" , 
                       Staphylococcaceae = "#A6CEE3", Streptococcaceae = "#1F78B4",
                       Thermicanaceae = "#B2DF8A", 
                       Xanthobacteraceae = "#A6CEE3",Xanthomonadaceae  = "#1F78B4")
```
Light blue = "#A6CEE3"
Dark blue = "#1F78B4"
Light green = "#B2DF8A"
Dark green = "#33A02C"
Pink = "#FB9A99" 
Red = "#E31A1C" 
Light Orange = "#FDBF6F"
Orange = "#FF7F00" 
lavender = "#CAB2D6" 
purple = "#6A3D9A" 
yellow = "#FFFF99" 
brown = "#B15928"


## 0.4 Import Unifrac Matrices 
(exported from qiime)- 
```{r}
#cecum
cecum_weighted_filtered_all_cohorts <- read.csv("./data/cecum/unifrac_data/weighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

cecum_unweighted_filtered_all_cohorts <- read.csv("./data/cecum/unifrac_data/unweighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

cecum_weighted_filtered_c45 <- cecum_weighted_filtered_all_cohorts[7:159,7:159]
cecum_unweighted_filtered_c45 <- cecum_unweighted_filtered_all_cohorts[7:159,7:159]

#ileum
ileum_weighted_filtered_all_cohorts <- read.csv("./data/ileum/unifrac_data/weighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

ileum_unweighted_filtered_all_cohorts <- read.csv("./data/ileum/unifrac_data/unweighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

ileum_weighted_filtered_c45 <- ileum_weighted_filtered_all_cohorts[7:160,7:160]
ileum_unweighted_filtered_c45 <- ileum_unweighted_filtered_all_cohorts[7:160,7:160]

#duodenum
duodenum_weighted_filtered_all_cohorts <- read.csv("./data/duodenum/unifrac_data/weighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

duodenum_unweighted_filtered_all_cohorts <- read.csv("./data/duodenum/unifrac_data/unweighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

duodenum_weighted_filtered_c45 <- duodenum_weighted_filtered_all_cohorts[7:159,7:159]
duodenum_unweighted_filtered_c45 <- duodenum_unweighted_filtered_all_cohorts[7:159,7:159]

#feces
feces_weighted_filtered_all_cohorts <- read.csv("./data/feces/unifrac_data/weighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

feces_unweighted_filtered_all_cohorts <- read.csv("./data/feces/unifrac_data/unweighted-unifrac-distance-matrix-filtered.tsv", header = TRUE, row.names = 1, sep = "\t")

feces_weighted_filtered_c45 <- feces_weighted_filtered_all_cohorts[10:239,10:239]
feces_unweighted_filtered_c45 <- feces_unweighted_filtered_all_cohorts[10:239,10:239]
```

## 0.5 SV taxonomy
```{r}
duodenum_SV_taxonomy <- read.csv("data/duodenum/duodenum_taxonomy.csv", header = TRUE, row.names = 1)

ileum_SV_taxonomy <- read.csv("data/ileum/ileum_taxonomy.csv", header = TRUE, row.names = 1)

cecum_SV_taxonomy <- read.csv("data/cecum/cecum_taxonomy.csv", header = TRUE, row.names = 1)
```
```{r}
duo_ile_SV_taxonomy <- rbind(duodenum_SV_taxonomy, ileum_SV_taxonomy[!rownames(ileum_SV_taxonomy) %in% rownames(duodenum_SV_taxonomy),])


intestine_SV_taxonomy <- rbind(duo_ile_SV_taxonomy, cecum_SV_taxonomy[!rownames(cecum_SV_taxonomy) %in% rownames(duo_ile_SV_taxonomy),])

dim(intestine_SV_taxonomy)
```


## 0.6  Lists for sample subsetting later
```{r}
#for subsetting rows (just samples)
cecum_female_list <-cecum_map[(cecum_map$Sex=="Female"),]$Mouse_Label
cecum_male_list <-cecum_map[(cecum_map$Sex=="Male"),]$Mouse_Label

cecum_wild_type_list <-cecum_map[(cecum_map$Genotype=="Wild_Type"),]$Mouse_Label
cecum_mutant_list <-cecum_map[(cecum_map$Genotype=="Mutant"),]$Mouse_Label

cecum_female_wild_type_list <- cecum_map[(cecum_map$Group=="Female_Wild-Type"),]$Mouse_Label
cecum_male_wild_type_list <- cecum_map[(cecum_map$Group=="Male_Wild-Type"),]$Mouse_Label
cecum_female_mutant_list <- cecum_map[(cecum_map$Group=="Female_Mutant"),]$Mouse_Label
cecum_male_mutant_list <- cecum_map[(cecum_map$Group=="Male_Mutant"),]$Mouse_Label

cecum_lumen_list <- cecum_map[(cecum_map$Sample_Type=="Lumen"),]$Mouse_Label
cecum_mucosa_list <- cecum_map[(cecum_map$Sample_Type=="Mucosa"),]$Mouse_Label

cecum_cohort_4_list <- cecum_map[(cecum_map$Cohort=="A"),]$Mouse_Label
cecum_cohort_5_list <- cecum_map[(cecum_map$Cohort=="B"),]$Mouse_Label

#for subsetting cols (has an X)
cecum_female_cols_list <-cecum_map[(cecum_map$Sex=="Female"),]$Mouse_Label_Cols
cecum_male_cols_list <-cecum_map[(cecum_map$Sex=="Male"),]$Mouse_Label_Cols

cecum_wild_type_cols_list <-cecum_map[(cecum_map$Genotype=="Wild_Type"),]$Mouse_Label_Cols
cecum_mutant_cols_list <-cecum_map[(cecum_map$Genotype=="Mutant"),]$Mouse_Label_Cols

cecum_female_wild_type_cols_list <- cecum_map[(cecum_map$Group=="Female_Wild-Type"),]$Mouse_Label_Cols
cecum_male_wild_type_cols_list <- cecum_map[(cecum_map$Group=="Male_Wild-Type"),]$Mouse_Label_Cols
cecum_female_mutant_cols_list <- cecum_map[(cecum_map$Group=="Female_Mutant"),]$Mouse_Label_Cols
cecum_male_mutant_cols_list <- cecum_map[(cecum_map$Group=="Male_Mutant"),]$Mouse_Label_Cols

cecum_lumen_cols_list <- cecum_map[(cecum_map$Sample_Type=="Lumen"),]$Mouse_Label_Cols
cecum_mucosa_cols_list <- cecum_map[(cecum_map$Sample_Type=="Mucosa"),]$Mouse_Label_Cols

cecum_cohort_4_cols_list<- cecum_map[(cecum_map$Cohort=="A"),]$Mouse_Label_Cols
cecum_cohort_5_cols_list<- cecum_map[(cecum_map$Cohort=="B"),]$Mouse_Label_Cols

#Ileum
#for subsetting rows (just samples)
ileum_female_list <-ileum_map[(ileum_map$Sex=="Female"),]$Mouse_Label
ileum_male_list <-ileum_map[(ileum_map$Sex=="Male"),]$Mouse_Label

ileum_wild_type_list <-ileum_map[(ileum_map$Genotype=="Wild_Type"),]$Mouse_Label
ileum_mutant_list <-ileum_map[(ileum_map$Genotype=="Mutant"),]$Mouse_Label

ileum_female_wild_type_list <- ileum_map[(ileum_map$Group=="Female_Wild-Type"),]$Mouse_Label
ileum_male_wild_type_list <- ileum_map[(ileum_map$Group=="Male_Wild-Type"),]$Mouse_Label
ileum_female_mutant_list <- ileum_map[(ileum_map$Group=="Female_Mutant"),]$Mouse_Label
ileum_male_mutant_list <- ileum_map[(ileum_map$Group=="Male_Mutant"),]$Mouse_Label

ileum_lumen_list <- ileum_map[(ileum_map$Sample_Type=="Lumen"),]$Mouse_Label
ileum_mucosa_list <- ileum_map[(ileum_map$Sample_Type=="Mucosa"),]$Mouse_Label

ileum_cohort_4_list <- ileum_map[(ileum_map$Cohort=="A"),]$Mouse_Label
ileum_cohort_5_list <- ileum_map[(ileum_map$Cohort=="B"),]$Mouse_Label

#for subsetting cols (has an X)
ileum_female_cols_list <-ileum_map[(ileum_map$Sex=="Female"),]$Mouse_Label_Cols
ileum_male_cols_list <-ileum_map[(ileum_map$Sex=="Male"),]$Mouse_Label_Cols

ileum_wild_type_cols_list <-ileum_map[(ileum_map$Genotype=="Wild_Type"),]$Mouse_Label_Cols
ileum_mutant_cols_list <-ileum_map[(ileum_map$Genotype=="Mutant"),]$Mouse_Label_Cols

ileum_female_wild_type_cols_list <- ileum_map[(ileum_map$Group=="Female_Wild-Type"),]$Mouse_Label_Cols
ileum_male_wild_type_cols_list <- ileum_map[(ileum_map$Group=="Male_Wild-Type"),]$Mouse_Label_Cols
ileum_female_mutant_cols_list <- ileum_map[(ileum_map$Group=="Female_Mutant"),]$Mouse_Label_Cols
ileum_male_mutant_cols_list <- ileum_map[(ileum_map$Group=="Male_Mutant"),]$Mouse_Label_Cols

ileum_lumen_cols_list <- ileum_map[(ileum_map$Sample_Type=="Lumen"),]$Mouse_Label_Cols
ileum_mucosa_cols_list <- ileum_map[(ileum_map$Sample_Type=="Mucosa"),]$Mouse_Label_Cols

ileum_cohort_4_cols_list<- ileum_map[(ileum_map$Cohort=="A"),]$Mouse_Label_Cols
ileum_cohort_5_cols_list<- ileum_map[(ileum_map$Cohort=="B"),]$Mouse_Label_Cols

#Duodenum
#for subsetting rows (just samples)
duodenum_female_list <-duodenum_map[(duodenum_map$Sex=="Female"),]$Mouse_Label
duodenum_male_list <-duodenum_map[(duodenum_map$Sex=="Male"),]$Mouse_Label

duodenum_wild_type_list <-duodenum_map[(duodenum_map$Genotype=="Wild_Type"),]$Mouse_Label
duodenum_mutant_list <-duodenum_map[(duodenum_map$Genotype=="Mutant"),]$Mouse_Label

duodenum_female_wild_type_list <- duodenum_map[(duodenum_map$Group=="Female_Wild-Type"),]$Mouse_Label
duodenum_male_wild_type_list <- duodenum_map[(duodenum_map$Group=="Male_Wild-Type"),]$Mouse_Label
duodenum_female_mutant_list <- duodenum_map[(duodenum_map$Group=="Female_Mutant"),]$Mouse_Label
duodenum_male_mutant_list <- duodenum_map[(duodenum_map$Group=="Male_Mutant"),]$Mouse_Label

duodenum_lumen_list <- duodenum_map[(duodenum_map$Sample_Type=="Lumen"),]$Mouse_Label
duodenum_mucosa_list <- duodenum_map[(duodenum_map$Sample_Type=="Mucosa"),]$Mouse_Label

duodenum_cohort_4_list <- duodenum_map[(duodenum_map$Cohort=="A"),]$Mouse_Label
duodenum_cohort_5_list <- duodenum_map[(duodenum_map$Cohort=="B"),]$Mouse_Label

#for subsetting cols (has an X)
duodenum_female_cols_list <-duodenum_map[(duodenum_map$Sex=="Female"),]$Mouse_Label_Cols
duodenum_male_cols_list <-duodenum_map[(duodenum_map$Sex=="Male"),]$Mouse_Label_Cols

duodenum_wild_type_cols_list <-duodenum_map[(duodenum_map$Genotype=="Wild_Type"),]$Mouse_Label_Cols
duodenum_mutant_cols_list <-duodenum_map[(duodenum_map$Genotype=="Mutant"),]$Mouse_Label_Cols

duodenum_female_wild_type_cols_list <- duodenum_map[(duodenum_map$Group=="Female_Wild-Type"),]$Mouse_Label_Cols
duodenum_male_wild_type_cols_list <- duodenum_map[(duodenum_map$Group=="Male_Wild-Type"),]$Mouse_Label_Cols
duodenum_female_mutant_cols_list <- duodenum_map[(duodenum_map$Group=="Female_Mutant"),]$Mouse_Label_Cols
duodenum_male_mutant_cols_list <- duodenum_map[(duodenum_map$Group=="Male_Mutant"),]$Mouse_Label_Cols

duodenum_lumen_cols_list <- duodenum_map[(duodenum_map$Sample_Type=="Lumen"),]$Mouse_Label_Cols
duodenum_mucosa_cols_list <- duodenum_map[(duodenum_map$Sample_Type=="Mucosa"),]$Mouse_Label_Cols

duodenum_cohort_4_cols_list<- duodenum_map[(duodenum_map$Cohort=="A"),]$Mouse_Label_Cols
duodenum_cohort_5_cols_list<- duodenum_map[(duodenum_map$Cohort=="B"),]$Mouse_Label_Cols


#Feces
#for subsetting rows (just samples)
feces_female_list <-feces_map[(feces_map$Sex=="Female"),]$Mouse_Label
feces_male_list <-feces_map[(feces_map$Sex=="Male"),]$Mouse_Label

feces_wild_type_list <-feces_map[(feces_map$Genotype=="Wild_Type"),]$Mouse_Label
feces_mutant_list <-feces_map[(feces_map$Genotype=="Mutant"),]$Mouse_Label

feces_female_wild_type_list <- feces_map[(feces_map$Group=="Female_Wild-Type"),]$Mouse_Label
feces_male_wild_type_list <- feces_map[(feces_map$Group=="Male_Wild-Type"),]$Mouse_Label
feces_female_mutant_list <- feces_map[(feces_map$Group=="Female_Mutant"),]$Mouse_Label
feces_male_mutant_list <- feces_map[(feces_map$Group=="Male_Mutant"),]$Mouse_Label

feces_week3_list <- feces_map[(feces_map$Week=="3"),]$Mouse_Label
feces_week6_list <- feces_map[(feces_map$Week=="6"),]$Mouse_Label
feces_week10_list <- feces_map[(feces_map$Week=="10"),]$Mouse_Label

feces_week3_cols_list <- feces_map[(feces_map$Week=="3"),]$Mouse_Label_Cols
feces_week6_cols_list <- feces_map[(feces_map$Week=="6"),]$Mouse_Label_Cols
feces_week10_cols_list <- feces_map[(feces_map$Week=="10"),]$Mouse_Label_Cols

feces_cohort_4_list <- feces_map[(feces_map$Cohort=="A"),]$Mouse_Label
feces_cohort_5_list <- feces_map[(feces_map$Cohort=="B"),]$Mouse_Label

#for subsetting cols (has an X)
feces_female_cols_list <-feces_map[(feces_map$Sex=="Female"),]$Mouse_Label_Cols
feces_male_cols_list <-feces_map[(feces_map$Sex=="Male"),]$MMouse_Label_Cols

feces_wild_type_cols_list <-feces_map[(feces_map$Genotype=="Wild_Type"),]$MMouse_Label_Cols
feces_mutant_cols_list <-feces_map[(feces_map$Genotype=="Mutant"),]$Mouse_Label_Cols

feces_female_wild_type_cols_list <- feces_map[(feces_map$Group=="Female_Wild-Type"),]$Mouse_Label_Cols
feces_male_wild_type_cols_list <- feces_map[(feces_map$Group=="Male_Wild-Type"),]$Mouse_Label_Cols
feces_female_mutant_cols_list <- feces_map[(feces_map$Group=="Female_Mutant"),]$Mouse_Label_Cols
feces_male_mutant_cols_list <- feces_map[(feces_map$Group=="Male_Mutant"),]$Mouse_Label_Cols

feces_week3_cols_list <- feces_map[(feces_map$Week=="3"),]$Mouse_Label_Cols
feces_week3_cols_list <- feces_map[(feces_map$Week=="6"),]$Mouse_Label_Cols
feces_week3_cols_list <- feces_map[(feces_map$Week=="10"),]$Mouse_Label_Cols

feces_cohort_4_cols_list<- feces_map[(feces_map$Cohort=="A"),]$Mouse_Label_Cols
feces_cohort_5_cols_list<- feces_map[(feces_map$Cohort=="B"),]$Mouse_Label_Cols
```
## 0.7 Sourcetracker data 
```{r}
fecal_sourcetracker_means <- read.csv("data/sourcetracker/combined_fecal_200_means.csv", header = TRUE, row.names = 1)
```

## 0.8 PICRUSt data
```{r}
duodenum_KO <- read.csv("data/duodenum/KO_table_duodenum.csv", header = TRUE, row.names = 1)

ileum_KO <- read.csv("data/ileum/KO_table_ileum.csv", header = TRUE, row.names = 1)

cecum_KO <- read.csv("data/cecum/KO_table_cecum.csv", header = TRUE, row.names = 1)

bile_acids_glucuronidase_KO_names <- read.csv("data/duodenum/KO_terms_relevant.csv", header = TRUE, row.names = 1)

bile_acids_glucuronidase_KO  <- row.names(bile_acids_glucuronidase_KO_names)
```

# 1 Alpha Diversity
## 1.1 Make Data frames with metadata
```{r}
#Duodenum
duodenum_map_alpha$Cohort <- factor(duodenum_map_alpha$Cohort, levels = c("A", "B"))

duodenum_alpha$Sex <- factor(duodenum_map_alpha$Sex)
duodenum_alpha$Group <- factor(duodenum_map_alpha$Group)
duodenum_alpha$Sample_Type <- factor(duodenum_map_alpha$Sample_Type)
duodenum_alpha$Genotype <- factor(duodenum_map_alpha$Genotype)
duodenum_alpha$Cohort <- factor(duodenum_map_alpha$Cohort)
duodenum_alpha$Mouse <- factor(duodenum_map_alpha$Mouse)
duodenum_alpha$Weight <- duodenum_map_alpha$Weight

str(duodenum_alpha)

#Ileum

ileum_map_alpha$Cohort <- factor(ileum_map_alpha$Cohort, levels = c("A", "B"))

ileum_alpha$Sex <- factor(ileum_map_alpha$Sex)
ileum_alpha$Group <- factor(ileum_map_alpha$Group)
ileum_alpha$Sample_Type <- factor(ileum_map_alpha$Sample_Type)
ileum_alpha$Genotype <- factor(ileum_map_alpha$Genotype)
ileum_alpha$Cohort <- factor(ileum_map_alpha$Cohort)
ileum_alpha$Mouse <- factor(ileum_map_alpha$Mouse)
ileum_alpha$Weight <- ileum_map_alpha$Weight

str(ileum_alpha)

#cecum
cecum_map_alpha$Cohort <- factor(cecum_map_alpha$Cohort, levels = c("A", "B"))

cecum_alpha$Sex <- factor(cecum_map_alpha$Sex)
cecum_alpha$Group <- factor(cecum_map_alpha$Group)
cecum_alpha$Sample_Type <- factor(cecum_map_alpha$Sample_Type)
cecum_alpha$Genotype <- factor(cecum_map_alpha$Genotype)
cecum_alpha$Cohort <- factor(cecum_map_alpha$Cohort)
cecum_alpha$Mouse <- factor(cecum_map_alpha$Mouse)
cecum_alpha$Weight <- cecum_map_alpha$Weight

str(cecum_alpha)
```
## 1.2 LME
### 1.2.1 Duodenum
Shannon
```{r}
lme_duodenum_shannon_sex_genotype_sample_type <- lme(Shannon ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = duodenum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_duodenum_shannon_sex_genotype_sample_type)

Anova(lme_duodenum_shannon_sex_genotype_sample_type)

#write.csv(Anova(lme_duodenum_shannon_sex_genotype_sample_type), file="./results/environmental_analysis/duodenum/alpha/lme/lme_duodenum_shannon_sex_genotype_sample_type_anova_table.csv")

#capture.output(summary(lme_duodenum_shannon_sex_genotype_sample_type),file="./results/environmental_analysis/duodenum/alpha/lme/lme_duodenum_shannon_sex_genotype_sample_type_summary.csv" )
```
Faith PD
```{r}
lme_duodenum_faith_sex_genotype_sample_type <- lme(Faith_PD ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = duodenum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_duodenum_faith_sex_genotype_sample_type)

Anova(lme_duodenum_faith_sex_genotype_sample_type)

write.csv(Anova(lme_duodenum_faith_sex_genotype_sample_type), file="./results/environmental_analysis/duodenum/alpha/lme/lme_duodenum_faith_sex_genotype_sample_type_anova_table.csv")

capture.output(summary(lme_duodenum_faith_sex_genotype_sample_type),file="./results/environmental_analysis/duodenum/alpha/lme/lme_duodenum_faith_sex_genotype_sample_type_summary.csv" )
```
### 1.2.2 Ileum
Shannon
```{r}
lme_ileum_shannon_sex_genotype_sample_type <- lme(Shannon ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = ileum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_ileum_shannon_sex_genotype_sample_type)

Anova(lme_ileum_shannon_sex_genotype_sample_type)

write.csv(Anova(lme_ileum_shannon_sex_genotype_sample_type), file="./results/environmental_analysis/ileum/alpha/lme/lme_ileum_shannon_sex_genotype_sample_type_anova_table.csv")

capture.output(summary(lme_ileum_shannon_sex_genotype_sample_type),file="./results/environmental_analysis/ileum/alpha/lme/lme_ileum_shannon_sex_genotype_sample_type_summary.csv" )
```
Faith PD
```{r}
lme_ileum_faith_sex_genotype_sample_type <- lme(Faith_PD ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = ileum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_ileum_faith_sex_genotype_sample_type)

Anova(lme_ileum_faith_sex_genotype_sample_type)

write.csv(Anova(lme_ileum_faith_sex_genotype_sample_type), file="./results/environmental_analysis/ileum/alpha/lme/lme_ileum_faith_sex_genotype_sample_type_anova_table.csv")

capture.output(summary(lme_ileum_faith_sex_genotype_sample_type),file="./results/environmental_analysis/ileum/alpha/lme/lme_ileum_faith_sex_genotype_sample_type_summary.csv" )
```
### 1.2.3 Cecum
Shannon
```{r}
lme_cecum_shannon_sex_genotype_sample_type <- lme(Shannon ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = cecum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_cecum_shannon_sex_genotype_sample_type)

Anova(lme_cecum_shannon_sex_genotype_sample_type)

write.csv(Anova(lme_cecum_shannon_sex_genotype_sample_type), file="./results/environmental_analysis/cecum/alpha/lme/lme_cecum_shannon_sex_genotype_sample_type_anova_table.csv")

capture.output(summary(lme_cecum_shannon_sex_genotype_sample_type),file="./results/environmental_analysis/cecum/alpha/lme/lme_cecum_shannon_sex_genotype_sample_type_summary.csv" )
```
Faith PD
```{r}
lme_cecum_faith_sex_genotype_sample_type <- lme(Faith_PD ~ Sex*Genotype*Sample_Type,
                            random = list(~1|Cohort, ~1|Mouse), 
                            data = cecum_alpha,
                              method = "ML", na.action = na.omit)

summary(lme_cecum_faith_sex_genotype_sample_type)

Anova(lme_cecum_faith_sex_genotype_sample_type)

write.csv(Anova(lme_cecum_faith_sex_genotype_sample_type), file="./results/environmental_analysis/cecum/alpha/lme/lme_cecum_faith_sex_genotype_sample_type_anova_table.csv")

capture.output(summary(lme_cecum_faith_sex_genotype_sample_type),file="./results/environmental_analysis/cecum/alpha/lme/lme_cecum_faith_sex_genotype_sample_type_summary.csv" )
```


## 1.3 Descriptive stats
```{r}
duodenum_alpha_desciptive <- duodenum_alpha %>%                            
  group_by(Group, Sample_Type) %>% 
  summarize(median = median(Faith_PD),
            mean = mean(Faith_PD), 
            sd = sd(Faith_PD))
ileum_alpha_desciptive <- ileum_alpha %>%                            
  group_by(Group, Sample_Type) %>% 
  summarize(median = median(Faith_PD),
            mean = mean(Faith_PD), 
            sd = sd(Faith_PD))
cecum_alpha_desciptive <- cecum_alpha %>%                            
  group_by(Group, Sample_Type) %>% 
  summarize(median = median(Faith_PD),
            mean = mean(Faith_PD), 
            sd = sd(Faith_PD))
duodenum_alpha_desciptive$Section <- "Duodenum"
ileum_alpha_desciptive$Section <- "Ileum"
cecum_alpha_desciptive$Section <- "Cecum"
intestine_alpha_descriptive <- rbind(duodenum_alpha_desciptive, ileum_alpha_desciptive)
intestine_alpha_descriptive <- rbind(intestine_alpha_descriptive , cecum_alpha_desciptive)
write.csv(intestine_alpha_descriptive,file="./results/environmental_analysis/duodenum/alpha/descriptive_stats_alpha_intestine.csv")
```

## 1.4. Family level Richness
### 1.4.1 Calculate richness
```{r}
duodenum_family_t <- data.frame(t(duodenum_family))
duodenum_family_count <- length(colnames(duodenum_family_t))
duodenum_family_t$Group_Sample_Type <- duodenum_map$Group_Sample_Type
duodenum_family_t$Cohort <- duodenum_map$Cohort
duodenum_family_t$Sample <- rownames(duodenum_family_t )
duodenum_family_t$Sex <- duodenum_map$Sex
duodenum_family_t$Group <- duodenum_map$Group
duodenum_family_t$Genotype <- duodenum_map$Genotype
duodenum_family_t$Sample_Type <- duodenum_map$Sample_Type
duodenum_family_t$Mouse_Label <- duodenum_map$Mouse_Label
duodenum_family_t$Mouse <- duodenum_map$Mouse
duodenum_family_t$Richness <- rowSums(duodenum_family_t[,1:duodenum_family_count] > 0)

ileum_family_t <- data.frame(t(ileum_family))
ileum_family_count <- length(colnames(ileum_family_t))
ileum_family_t$Group_Sample_Type <- ileum_map$Group_Sample_Type
ileum_family_t$Cohort <- ileum_map$Cohort
ileum_family_t$Sample <- rownames(ileum_family_t )
ileum_family_t$Sex <- ileum_map$Sex
ileum_family_t$Group <- ileum_map$Group
ileum_family_t$Genotype <- ileum_map$Genotype
ileum_family_t$Sample_Type <- ileum_map$Sample_Type
ileum_family_t$Mouse_Label <- ileum_map$Mouse_Label
ileum_family_t$Mouse <- ileum_map$Mouse
ileum_family_t$Richness <- rowSums(ileum_family_t[,1:ileum_family_count] > 0)

cecum_family_t <- data.frame(t(cecum_family))
cecum_family_count <- length(colnames(cecum_family_t))
cecum_family_t$Group_Sample_Type <- cecum_map$Group_Sample_Type
cecum_family_t$Cohort <- cecum_map$Cohort
cecum_family_t$Sample <- rownames(cecum_family_t )
cecum_family_t$Sex <- cecum_map$Sex
cecum_family_t$Group <- cecum_map$Group
cecum_family_t$Genotype <- cecum_map$Genotype
cecum_family_t$Sample_Type <- cecum_map$Sample_Type
cecum_family_t$Mouse_Label <- cecum_map$Mouse_Label
cecum_family_t$Mouse <- cecum_map$Mouse
cecum_family_t$Richness <- rowSums(cecum_family_t[,1:cecum_family_count] > 0)
```
### 1.4.2 Graph individual sections
```{r}
p_duodenum_family_richness_group <- ggplot(data = duodenum_family_t) +
    geom_boxplot(aes(x = Group, y = Richness, color = Group))+
  geom_point(aes(x = Group, y = Richness, color = Group))  +
  scale_color_manual(values = group_color_list) +
  labs(title = "Duodenum") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  guides(color=guide_legend(title="Sample Type")) +
  facet_grid(~Sample_Type) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  ylim(5,30)
p_duodenum_family_richness_group

ggsave('./results/environmental_analysis/duodenum/alpha/family/ileum_family_richness_group.png',width=11, height=10, plot = p_duodenum_family_richness_group)

p_ileum_family_richness_group <- ggplot(data = ileum_family_t) +
    geom_boxplot(aes(x = Group, y = Richness, color = Group))+
  geom_point(aes(x = Group, y = Richness, color = Group))  +
  scale_color_manual(values = group_color_list) +
  labs(title = "ileum") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  guides(color=guide_legend(title="Sample Type")) +
  facet_grid(~Sample_Type) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  ylim(5,30)
p_ileum_family_richness_group

ggsave('./results/environmental_analysis/ileum/alpha/family/ileum_family_richness_group.png',width=11, height=10, plot = p_ileum_family_richness_group)

p_cecum_family_richness_group <- ggplot(data = cecum_family_t) +
    geom_boxplot(aes(x = Group, y = Richness, color = Group))+
  geom_point(aes(x = Group, y = Richness, color = Group))  +
  scale_color_manual(values = group_color_list) +
  labs(title = "cecum") +
  theme_classic() +
  theme(text = element_text(size = 20)) +
  guides(color=guide_legend(title="Sample Type")) +
  facet_grid(~Sample_Type) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  ylim(5,30)
p_cecum_family_richness_group

ggsave('./results/environmental_analysis/cecum/alpha/family/cecum_family_richness_group.png',width=11, height=10, plot = p_cecum_family_richness_group)
```



# 2 Beta Diversity
## 2.1 Taxa Barplots
### 2.1.1 Duodenum Taxa Bar Plots
First: Do bar for all samples
Make family data relative
```{r}
duodenum_family_rel <- as.data.frame(apply(duodenum_family,2,function(x){x/sum(x)}))
head(duodenum_family_rel)

duodenum_family_rel_t <- as.data.frame(t(duodenum_family_rel))
duodenum_family_rel_t$Group_Sample_Type <- duodenum_map$Group_Sample_Type
duodenum_family_rel_t$Cohort <- duodenum_map$Cohort
duodenum_family_rel_t$Sample <- rownames(duodenum_family_rel_t )
duodenum_family_rel_t$Sex <- duodenum_map$Sex
duodenum_family_rel_t$Group <- duodenum_map$Group
duodenum_family_rel_t$Genotype <- duodenum_map$Genotype
duodenum_family_rel_t$Sample_Type <- duodenum_map$Sample_Type
duodenum_family_rel_t$Mouse_Label <- duodenum_map$Mouse_Label
duodenum_family_rel_t$Mouse <- duodenum_map$Mouse

duodenum_family_list <- as.list(colnames(duodenum_family_rel_t[,1:41]))
duodenum_family_metadata_list <- as.list(colnames(duodenum_family_rel_t[,42:50]))

duodenum_family_rel_t_stack = melt(duodenum_family_rel_t, id = c("Sample"), id.vars=c(duodenum_family_metadata_list), measure.vars=c(duodenum_family_list)) 

duodenum_family_rel_t_stack <- duodenum_family_rel_t_stack[order(-duodenum_family_rel_t_stack$value),]
duodenum_family_rel_t_stack$variable <- factor(duodenum_family_rel_t_stack$variable,levels=unique(duodenum_family_rel_t_stack$variable))
```

Graph all samples:
```{r}
p_duodenum_family = ggplot(duodenum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Relative Abundance") +
    facet_wrap(~Group_Sample_Type, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_duodenum_family

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_stacked.png",p_duodenum_family, width = 30, height = 10, limitsize = FALSE)

p_duodenum_family_facet = ggplot(duodenum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Relative Abundance") +
    facet_grid(Sample_Type~Group, scales = "free_x",  shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_duodenum_family_facet

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_stacked_facet.png",p_duodenum_family_facet, width = 20, height = 10, limitsize = FALSE)
```

Graph just mucosa
```{r}
p_duodenum_mucosa_family = ggplot(duodenum_family_rel_t_stack[(duodenum_family_rel_t_stack$Sample_Type=="Mucosa"),]) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "duodenum Family Relative Abundance") +
    facet_wrap(~Group, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family")  +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_duodenum_mucosa_family

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/mucosa_samples_family_stacked.png",p_duodenum_mucosa_family, width = 30, height = 10, limitsize = FALSE)
```

Second, do sample typ averages:

Take means for each Group_Sample_type in duodenum_family_rel_t
By cohort
```{r}
duodenum_family_mean_rel_t <- duodenum_family_rel_t[,1:43] %>% group_by(Group_Sample_Type, Cohort) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
duodenum_family_mean_rel_t

duodenum_family_mean_list <- as.list(colnames(duodenum_family_mean_rel_t[,3:31]))
duodenum_family_mean_metadata_list <-as.list(colnames(duodenum_family_mean_rel_t[,1:2]))

duodenum_family_mean_rel_t_stack = melt(duodenum_family_mean_rel_t, id = c("Group_Sample_Type"), id.vars =c(duodenum_family_mean_metadata_list), measure.vars=c(duodenum_family_mean_list)) 

duodenum_family_mean_rel_t_stack <- duodenum_family_mean_rel_t_stack[order(-duodenum_family_mean_rel_t_stack$value),]
duodenum_family_mean_rel_t_stack$variable <- factor(duodenum_family_mean_rel_t_stack$variable,levels=unique(duodenum_family_mean_rel_t_stack$variable))
```
Graph
```{r}
p_duodenum_family_mean_cohort = ggplot(duodenum_family_mean_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Mean Relative Abundance") +
    facet_wrap(~Cohort) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_duodenum_family_mean_cohort

p_duodenum_family_mean_cohort_flip <- p_duodenum_family_mean_cohort + 
    coord_flip() 

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_stacked_by_cohort.png",p_duodenum_family_mean_cohort, width = 15, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_stacked_by_cohort_flip.png",p_duodenum_family_mean_cohort_flip, width = 20, height = 5, limitsize = FALSE)
```
Cohorts combined
```{r}
duodenum_family_mean_both_cohorts_rel_t <- duodenum_family_rel_t[,1:42] %>% group_by(Group_Sample_Type) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
duodenum_family_mean_both_cohorts_rel_t

duodenum_family_mean_both_cohorts_list <- as.list(colnames(duodenum_family_mean_both_cohorts_rel_t[,2:42]))

duodenum_family_mean_both_cohorts_rel_t_stack = melt(duodenum_family_mean_both_cohorts_rel_t, id = c("Group_Sample_Type"), measure.vars=c(duodenum_family_mean_both_cohorts_list)) 

duodenum_family_mean_both_cohorts_rel_t_stack <- duodenum_family_mean_both_cohorts_rel_t_stack[order(-duodenum_family_mean_both_cohorts_rel_t_stack$value),]
duodenum_family_mean_both_cohorts_rel_t_stack$variable <- factor(duodenum_family_mean_both_cohorts_rel_t_stack$variable,levels=unique(duodenum_family_mean_both_cohorts_rel_t_stack$variable))

p_duodenum_family_mean_both_cohorts = ggplot(duodenum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_duodenum_family_mean_both_cohorts

p_duodenum_family_mean_both_cohorts_flip <- p_duodenum_family_mean_both_cohorts + 
    coord_flip() 

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked.png",p_duodenum_family_mean_both_cohorts, width = 10, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip.png",p_duodenum_family_mean_both_cohorts_flip, width = 20, height = 5, limitsize = FALSE)
```
order by genotype
```{r}
duodenum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type_gen <- factor(duodenum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type, levels = c("Female_Mutant_Lumen", "Male_Mutant_Lumen", "Female_Wild_Type_Lumen", "Male_Wild_Type_Lumen", "Female_Mutant_Mucosa", "Male_Mutant_Mucosa","Female_Wild_Type_Mucosa", "Male_Wild_Type_Mucosa" ))

p_duodenum_family_mean_both_cohorts_genotype = ggplot(duodenum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 20))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_duodenum_family_mean_both_cohorts_genotype

p_duodenum_family_mean_both_cohorts_flip_genotype <- p_duodenum_family_mean_both_cohorts_genotype + 
    coord_flip() +
  theme(axis.text.x = element_text(angle = 0.0001, vjust = 0.5, hjust=1))

#ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_genotype.png",p_duodenum_family_mean_both_cohorts_genotype, width = 10, height = 10, limitsize = FALSE)

#ggsave("./results/environmental_analysis/duodenum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip_by_genotype.png",p_duodenum_family_mean_both_cohorts_flip_genotype, width = 20, height = 5, limitsize = FALSE)
```

### 2.1.2 Ileum Taxa Bar Plots
First: Do bar for all samples
Make family data relative
```{r}
ileum_family_rel <- as.data.frame(apply(ileum_family,2,function(x){x/sum(x)}))
head(ileum_family_rel)

ileum_family_rel_t <- as.data.frame(t(ileum_family_rel))
ileum_family_rel_t$Group_Sample_Type <- ileum_map$Group_Sample_Type
ileum_family_rel_t$Cohort <- ileum_map$Cohort
ileum_family_rel_t$Sample <- rownames(ileum_family_rel_t )
ileum_family_rel_t$Sex <- ileum_map$Sex
ileum_family_rel_t$Group <- ileum_map$Group
ileum_family_rel_t$Genotype <- ileum_map$Genotype
ileum_family_rel_t$Sample_Type <- ileum_map$Sample_Type
ileum_family_rel_t$Mouse_Label <- ileum_map$Mouse_Label
ileum_family_rel_t$Mouse <- ileum_map$Mouse

ileum_family_list <- as.list(colnames(ileum_family_rel_t[,1:38]))
ileum_family_metadata_list <- as.list(colnames(ileum_family_rel_t[,39:47]))

ileum_family_rel_t_stack = melt(ileum_family_rel_t, id = c("Sample"), id.vars=c(ileum_family_metadata_list), measure.vars=c(ileum_family_list)) 

ileum_family_rel_t_stack <- ileum_family_rel_t_stack[order(-ileum_family_rel_t_stack$value),]
ileum_family_rel_t_stack$variable <- factor(ileum_family_rel_t_stack$variable,levels=unique(ileum_family_rel_t_stack$variable))

#Graph all samples
p_ileum_family = ggplot(ileum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Relative Abundance") +
    facet_wrap(~Group_Sample_Type, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_ileum_family

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_stacked.png",p_ileum_family, width = 30, height = 10, limitsize = FALSE)
```
Facet graph?
```{r}
p_ileum_family_facet = ggplot(ileum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Relative Abundance") +
    facet_grid(Sample_Type~Group, scales = "free_x", shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_ileum_family_facet

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_stacked_facet.png",p_ileum_family_facet, width = 20, height = 10, limitsize = FALSE)
```
Second, do sample typ averages:
Take means for each Group_Sample_type in ileum_family_rel_t
By cohort
```{r}
ileum_family_mean_rel_t <- ileum_family_rel_t[,1:40] %>% group_by(Group_Sample_Type, Cohort) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
ileum_family_mean_rel_t

ileum_family_mean_list <- as.list(colnames(ileum_family_mean_rel_t[,3:31]))
ileum_family_mean_metadata_list <-as.list(colnames(ileum_family_mean_rel_t[,1:2]))

ileum_family_mean_rel_t_stack = melt(ileum_family_mean_rel_t, id = c("Group_Sample_Type"), id.vars =c(ileum_family_mean_metadata_list), measure.vars=c(ileum_family_mean_list)) 

ileum_family_mean_rel_t_stack <- ileum_family_mean_rel_t_stack[order(-ileum_family_mean_rel_t_stack$value),]
ileum_family_mean_rel_t_stack$variable <- factor(ileum_family_mean_rel_t_stack$variable,levels=unique(ileum_family_mean_rel_t_stack$variable))

p_ileum_family_mean_cohort = ggplot(ileum_family_mean_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Mean Relative Abundance") +
    facet_wrap(~Cohort) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_ileum_family_mean_cohort

p_ileum_family_mean_cohort_flip <- p_ileum_family_mean_cohort + 
    coord_flip() 

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_stacked_by_cohort.png",p_ileum_family_mean_cohort, width = 15, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_stacked_by_cohort_flip.png",p_ileum_family_mean_cohort_flip, width = 20, height = 5, limitsize = FALSE)
```
Cohorts combined
```{r}
ileum_family_mean_both_cohorts_rel_t <- ileum_family_rel_t[,1:39] %>% group_by(Group_Sample_Type) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
ileum_family_mean_both_cohorts_rel_t


ileum_family_mean_both_cohorts_list <- as.list(colnames(ileum_family_mean_both_cohorts_rel_t[,2:39]))

ileum_family_mean_both_cohorts_rel_t_stack = melt(ileum_family_mean_both_cohorts_rel_t, id = c("Group_Sample_Type"), measure.vars=c(ileum_family_mean_both_cohorts_list)) 

ileum_family_mean_both_cohorts_rel_t_stack <- ileum_family_mean_both_cohorts_rel_t_stack[order(-ileum_family_mean_both_cohorts_rel_t_stack$value),]
ileum_family_mean_both_cohorts_rel_t_stack$variable <- factor(ileum_family_mean_both_cohorts_rel_t_stack$variable,levels=unique(ileum_family_mean_both_cohorts_rel_t_stack$variable))

p_ileum_family_mean_both_cohorts = ggplot(ileum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_ileum_family_mean_both_cohorts

p_ileum_family_mean_both_cohorts_flip <- p_ileum_family_mean_both_cohorts + 
    coord_flip() 

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked.png",p_ileum_family_mean_both_cohorts, width = 10, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip.png",p_ileum_family_mean_both_cohorts_flip, width = 20, height = 5, limitsize = FALSE)
```
Order x axis by genotype
```{r}
ileum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type_gen <- factor(ileum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type, levels = c("Female_Mutant_Lumen", "Male_Mutant_Lumen", "Female_Wild_Type_Lumen", "Male_Wild_Type_Lumen", "Female_Mutant_Mucosa", "Male_Mutant_Mucosa", "Female_Wild_Type_Mucosa", "Male_Wild_Type_Mucosa"))

p_ileum_family_mean_both_cohorts_genotype = ggplot(ileum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 20))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_ileum_family_mean_both_cohorts_genotype

p_ileum_family_mean_both_cohorts_flip_genotype <- p_ileum_family_mean_both_cohorts_genotype + 
    coord_flip() +
  theme(axis.text.x = element_text(angle = 0.0001, vjust = 0.5, hjust=1))

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_genotype.png",p_ileum_family_mean_both_cohorts_genotype, width = 10, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/ileum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip_by_genotype.png",p_ileum_family_mean_both_cohorts_flip_genotype, width = 20, height = 6, limitsize = FALSE)
```

### 2.1.3 Cecum Taxa Bar Plots
All Samples:
```{r}
#First: Do bar for all samples
Make family data relative
cecum_family_rel <- as.data.frame(apply(cecum_family,2,function(x){x/sum(x)}))
head(cecum_family_rel)


#Stack the data

cecum_family_rel_t <- as.data.frame(t(cecum_family_rel))
cecum_family_rel_t$Group_Sample_Type <- cecum_map$Group_Sample_Type
cecum_family_rel_t$Cohort <- cecum_map$Cohort
cecum_family_rel_t$Sample <- rownames(cecum_family_rel_t )
cecum_family_rel_t$Sex <- cecum_map$Sex
cecum_family_rel_t$Group <- cecum_map$Group
cecum_family_rel_t$Genotype <- cecum_map$Genotype
cecum_family_rel_t$Sample_Type <- cecum_map$Sample_Type
cecum_family_rel_t$Mouse_Label <- cecum_map$Mouse_Label
cecum_family_rel_t$Mouse <- cecum_map$Mouse


cecum_family_list <- as.list(colnames(cecum_family_rel_t[,1:29]))
cecum_family_metadata_list <- as.list(colnames(cecum_family_rel_t[,30:38]))

cecum_family_rel_t_stack = melt(cecum_family_rel_t, id = c("Sample"), id.vars=c(cecum_family_metadata_list), measure.vars=c(cecum_family_list)) 

#Stack
cecum_family_rel_t_stack <- cecum_family_rel_t_stack[order(-cecum_family_rel_t_stack$value),]
cecum_family_rel_t_stack$variable <- factor(cecum_family_rel_t_stack$variable,levels=unique(cecum_family_rel_t_stack$variable))


cecum_family_rel_t_stack$Group_Sample_Type <- factor(cecum_family_rel_t_stack$Group_Sample_Type)
```
Graph all samples:
```{r}
p_cecum_family = ggplot(cecum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Relative Abundance") +
    facet_wrap(~Group_Sample_Type, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_cecum_family

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_stacked.png",p_cecum_family, width = 30, height = 10, limitsize = FALSE)

p_cecum_family_facet = ggplot(cecum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Relative Abundance") +
    facet_grid(Sample_Type~Group, scales = "free_x", shrink = TRUE) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) 
p_cecum_family_facet

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_stacked_facet.png",p_cecum_family_facet, width = 20, height = 10, limitsize = FALSE)
```

Second, do sample typ averages:
Take means for each Group_Sample_type in cecum_family_rel_t
By cohort
```{r}
cecum_family_mean_rel_t <- cecum_family_rel_t[,1:31] %>% group_by(Group_Sample_Type, Cohort) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
cecum_family_mean_rel_t
#stack by sample type and cohort
cecum_family_mean_list <- as.list(colnames(cecum_family_mean_rel_t[,3:31]))
cecum_family_mean_metadata_list <-as.list(colnames(cecum_family_mean_rel_t[,1:2]))

cecum_family_mean_rel_t_stack = melt(cecum_family_mean_rel_t, id = c("Group_Sample_Type"), id.vars =c(cecum_family_mean_metadata_list), measure.vars=c(cecum_family_mean_list)) 

cecum_family_mean_rel_t_stack <- cecum_family_mean_rel_t_stack[order(-cecum_family_mean_rel_t_stack$value),]
cecum_family_mean_rel_t_stack$variable <- factor(cecum_family_mean_rel_t_stack$variable,levels=unique(cecum_family_mean_rel_t_stack$variable))
```
Graph
```{r}
p_cecum_family_mean_cohort = ggplot(cecum_family_mean_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Mean Relative Abundance") +
    facet_wrap(~Cohort) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_cecum_family_mean_cohort

p_cecum_family_mean_cohort_flip <- p_cecum_family_mean_cohort + 
    coord_flip() 

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_stacked_by_cohort.png",p_cecum_family_mean_cohort, width = 15, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_stacked_by_cohort_flip.png",p_cecum_family_mean_cohort_flip, width = 20, height = 5, limitsize = FALSE)
```
Cohorts combined
```{r}
cecum_family_mean_both_cohorts_rel_t <- cecum_family_rel_t[,1:30] %>% group_by(Group_Sample_Type) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
cecum_family_mean_both_cohorts_rel_t

#Stack
cecum_family_mean_both_cohorts_list <- as.list(colnames(cecum_family_mean_both_cohorts_rel_t[,2:30]))

cecum_family_mean_both_cohorts_rel_t_stack = melt(cecum_family_mean_both_cohorts_rel_t, id = c("Group_Sample_Type"), measure.vars=c(cecum_family_mean_both_cohorts_list)) 

cecum_family_mean_both_cohorts_rel_t_stack <- cecum_family_mean_both_cohorts_rel_t_stack[order(-cecum_family_mean_both_cohorts_rel_t_stack$value),]
cecum_family_mean_both_cohorts_rel_t_stack$variable <- factor(cecum_family_mean_both_cohorts_rel_t_stack$variable,levels=unique(cecum_family_mean_both_cohorts_rel_t_stack$variable))
```
Graph
```{r}
p_cecum_family_mean_both_cohorts = ggplot(cecum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group_Sample_Type, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_cecum_family_mean_both_cohorts

p_cecum_family_mean_both_cohorts_flip <- p_cecum_family_mean_both_cohorts + 
    coord_flip() 

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by.png",p_cecum_family_mean_both_cohorts, width = 10, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip.png",p_cecum_family_mean_both_cohorts_flip, width = 20, height = 5, limitsize = FALSE)
```
Order x axis by genotype
```{r}
cecum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type_gen <- factor(cecum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type, levels = c("Female_Mutant_Lumen", "Male_Mutant_Lumen", "Female_Wild_Type_Lumen", "Male_Wild_Type_Lumen", "Female_Mutant_Mucosa", "Male_Mutant_Mucosa", "Female_Wild_Type_Mucosa", "Male_Wild_Type_Mucosa"))

p_cecum_family_mean_both_cohorts_genotype = ggplot(cecum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 20))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") 
p_cecum_family_mean_both_cohorts_genotype

p_cecum_family_mean_both_cohorts_flip_genotype <- p_cecum_family_mean_both_cohorts_genotype + 
    coord_flip() +
  theme(axis.text.x = element_text(angle = 0.0001, vjust = 0.5, hjust=1))

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_genotype.png",p_cecum_family_mean_both_cohorts_genotype, width = 10, height = 10, limitsize = FALSE)

ggsave("./results/environmental_analysis/cecum/taxa_barplots/all_samples_family_mean_both_cohorts_stacked_by_flip_by_genotype.png",p_cecum_family_mean_both_cohorts_flip_genotype, width = 20, height = 5, limitsize = FALSE)
```

### 2.1.4 Taxa Barplots Stats for taxa barplots
Output values for descriptive Statistics
```{r}
write.csv(duodenum_family_mean_both_cohorts_rel_t, 
"./results/environmental_analysis/duodenum/taxa_barplots/duodenum_family_mean_both_cohorts_relative_abundance.csv")

write.csv(ileum_family_mean_both_cohorts_rel_t, 
"./results/environmental_analysis/ileum/taxa_barplots/ileum_family_mean_both_cohorts_relative_abundance.csv")

write.csv(cecum_family_mean_both_cohorts_rel_t, 
"./results/environmental_analysis/cecum/taxa_barplots/cecum_family_mean_both_cohorts_relative_abundance.csv")
```

Top 10 most abundant LMEs-
```{r}
duodenum_family_10_colmeans <- data.frame(sort(format(colMeans(duodenum_family_mean_both_cohorts_rel_t[2:42]), scientific = FALSE), decreasing= TRUE)[1:10])
duodenum_family_10_colmeans$Family <- rownames(duodenum_family_10_colmeans)
colnames(duodenum_family_10_colmeans) <- c("Duodenum_Abundance", "Family")


ileum_family_10_colmeans <-data.frame( sort(format(colMeans(ileum_family_mean_both_cohorts_rel_t[2:39]), scientific = FALSE), decreasing= TRUE)[1:10] )
ileum_family_10_colmeans$Family <- rownames(ileum_family_10_colmeans)
colnames(ileum_family_10_colmeans) <- c("Ileum_Abundance", "Family")


cecum_family_10_colmeans <-data.frame( sort(format(colMeans(cecum_family_mean_both_cohorts_rel_t[2:30]), scientific = FALSE), decreasing= TRUE)[1:10])
cecum_family_10_colmeans$Family <- rownames(cecum_family_10_colmeans)
colnames(cecum_family_10_colmeans) <- c("Cecum_Abundance", "Family")


intestine_top_10_family <- list(duodenum_family_10_colmeans, ileum_family_10_colmeans, cecum_family_10_colmeans) %>% reduce(full_join, by='Family', all = TRUE)
intestine_top_10_family[is.na(intestine_top_10_family)] <- 0

intestine_top_10_family2 <- intestine_top_10_family[, c(2,1,3,4)]

intestine_top_10_family2$Duodenum_Abundance<- as.numeric(intestine_top_10_family2$Duodenum_Abundance)
intestine_top_10_family2$Ileum_Abundance<- as.numeric(intestine_top_10_family2$Ileum_Abundance)
intestine_top_10_family2$Cecum_Abundance<- as.numeric(intestine_top_10_family2$Cecum_Abundance)
intestine_top_10_family$Means <-apply(intestine_top_10_family2[,2:4],1,mean)

intestine_top_10_family_order<- intestine_top_10_family[order(intestine_top_10_family$Means, decreasing = TRUE),]
intestine_top_10_family_abundances <- intestine_top_10_family_order[1:10,]$Family
```

Now do LMEs for intestine_top_10_family_abundances
want columns as families, rows as samples
```{r}
intestine_family_top_10_all_samples <- data.frame(t(filter(intestine_family_tx, rownames(intestine_family_tx) %in% intestine_top_10_family_abundances)))

intestine_family_top_10_all_samples$Sex <- intestine_map$Sex
intestine_family_top_10_all_samples$Genotype <- intestine_map$Genotype
intestine_family_top_10_all_samples$Sample_Type <- intestine_map$Sample_Type
intestine_family_top_10_all_samples$Cohort <- intestine_map$Cohort
intestine_family_top_10_all_samples$Mouse <- intestine_map$Mouse
intestine_family_top_10_all_samples$Weight <- intestine_map$Weight
intestine_family_top_10_all_samples$Section <- intestine_map$Section
```

```{r}
#change the number of names to how many genera there are
intestine_top_10_family_names <- colnames(intestine_family_top_10_all_samples)[1:10]
num_intestine_top_10_family <- length(intestine_top_10_family_names)

# create a named list to hold the fitted models
intestine_family_model_list <- as.list(1:num_intestine_top_10_family)
names(intestine_family_model_list) <- intestine_top_10_family_names

intestine_family_model_anova_list <- as.list(1:num_intestine_top_10_family)
names(intestine_family_model_anova_list) <- intestine_top_10_family_names

# loop over names
for(i in intestine_top_10_family_names){ 
  # create temporary data matrix and model formula
  tmp_top_10 <- intestine_family_top_10_all_samples[, c(i,"Sex","Genotype","Sample_Type","Cohort", "Mouse", "Section")]
  fml_top_10 <- as.formula( paste( i, "~", paste(c("Sex","Genotype", "Sample_Type", "Section"), collapse="*") ) )
  # assign fit to list by name
  intestine_family_model_list[[i]] <- lme(fml_top_10, random = list(~1|Cohort, ~1|Mouse), method = "ML", data= tmp_top_10)
  intestine_family_model_anova_list[[i]] <- Anova(lme(fml_top_10, random = list(~1|Cohort, ~1|Mouse), method = "ML", data= tmp_top_10))
}

capture.output(intestine_family_model_anova_list, file="./results/environmental_analysis/all_envs/diff_abundance_lme/lme_intestine_top_10_family_anova.csv")

capture.output(intestine_family_model_list, file="./results/environmental_analysis/all_envs/diff_abundance_lme/lme_intestine_top_10_family_models.csv")
```

Do FDR correction 
```{r}
p_values_intestine_top_10_family_anova_table <- data.frame(lapply(intestine_family_model_anova_list, "[", , "Pr(>Chisq)"))
rownames(p_values_intestine_top_10_family_anova_table) <- rownames(intestine_family_model_anova_list$Lachnospiraceae)

write.csv(p_values_intestine_top_10_family_anova_table, file="./results/environmental_analysis/all_envs/diff_abundance_lme/lme_intestine_top_10_family_anova_p_value_table.csv")

p_values_intestine_top_10_family_anova_table_fdr <- t(data.frame(apply(p_values_intestine_top_10_family_anova_table, 1, function(x) p.adjust(x, method = "fdr"))))

write.csv(p_values_intestine_top_10_family_anova_table_fdr, file="./results/environmental_analysis/all_envs/diff_abundance_lme/lme_intestine_top_10_family_anova_p_value_table_FDR.csv")
```

## 2.2 Heatmaps
### 2.2.1 All Intestine
SVs 
```{r}
intestine_SVs_zerorep <- intestine_SVs
intestine_SVs_zerorep[intestine_SVs_zerorep == 0] <- 0.001
tail(intestine_SVs_zerorep)

intestine_SVs_tx<- data.frame(t(clr(t(intestine_SVs_zerorep))))

#prep to stack
intestine_SVs_tx_t <- data.frame(t(intestine_SVs_tx))
intestine_SVs_tx_t$Group <- factor(intestine_map$Group, levels = c("Female_Mutant", "Male_Mutant", "Female_Wild_Type", "Male_Wild_Type" ))
intestine_SVs_tx_t$Sample_Type_Env <- intestine_map$Sample_Type_Env

intestine_SVs_tx_group_by_group_sample <- intestine_SVs_tx_t%>%  # [,1:30]
  group_by(Group, Sample_Type_Env) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
intestine_SVs_tx_group_by_group_sample
```
Stack
```{r}
intestine_SVs_tx_group_by_group_sample_list <- as.list(colnames(intestine_SVs_tx_group_by_group_sample[,3:371]))

intestine_SVs_tx_group_by_group_sample_stack = melt(intestine_SVs_tx_group_by_group_sample, id.vars = c("Group", "Sample_Type_Env"), measure.vars=c(intestine_SVs_tx_group_by_group_sample_list)) 

intestine_SVs_tx_group_by_group_sample_stack <- intestine_SVs_tx_group_by_group_sample_stack[order(-intestine_SVs_tx_group_by_group_sample_stack$value),]
intestine_SVs_tx_group_by_group_sample_stack$variable <- factor(intestine_SVs_tx_group_by_group_sample_stack$variable,levels=unique(intestine_SVs_tx_group_by_group_sample_stack$variable))

colnames(intestine_SVs_tx_group_by_group_sample_stack) <- c("Group", "Sample_Type_Env", "SVs", "CLR_Abundance")

intestine_SVs_tx_group_by_group_sample_stack$Sample_Type_Env <- factor(intestine_SVs_tx_group_by_group_sample_stack$Sample_Type_Env,levels= c("Duodenum_Lumen", "Duodenum_Mucosa", "Ileum_Lumen", "Ileum_Mucosa","Cecum_Lumen", "Cecum_Mucosa"))
```

Genus
genus clr aitchison
Zero replace the table
```{r}
intestine_genus_zerorep <- intestine_genus
intestine_genus_zerorep[intestine_genus_zerorep == 0] <- 0.001
tail(intestine_genus_zerorep)

intestine_genus_tx<- data.frame(t(clr(t(intestine_genus_zerorep))))

#prep to stack
intestine_genus_tx_t <- data.frame(t(intestine_genus_tx))
intestine_genus_tx_t$Group <- factor(intestine_map$Group, levels = c("Female_Mutant", "Male_Mutant", "Female_Wild_Type", "Male_Wild_Type" ))
intestine_genus_tx_t$Sample_Type_Env <- intestine_map$Sample_Type_Env

intestine_genus_tx_group_by_group_sample <- intestine_genus_tx_t%>%  # [,1:30]
  group_by(Group, Sample_Type_Env) %>% 
  summarise(across(everything(), mean),
            .groups = 'drop')  %>%
  as.data.frame()
intestine_genus_tx_group_by_group_sample
```
Stack
```{r}
intestine_genus_tx_group_by_group_sample_list <- as.list(colnames(intestine_genus_tx_group_by_group_sample[,3:98]))

intestine_genus_tx_group_by_group_sample_stack = melt(intestine_genus_tx_group_by_group_sample, id.vars = c("Group", "Sample_Type_Env"), measure.vars=c(intestine_genus_tx_group_by_group_sample_list)) 

intestine_genus_tx_group_by_group_sample_stack <- intestine_genus_tx_group_by_group_sample_stack[order(-intestine_genus_tx_group_by_group_sample_stack$value),]
intestine_genus_tx_group_by_group_sample_stack$variable <- factor(intestine_genus_tx_group_by_group_sample_stack$variable,levels=unique(intestine_genus_tx_group_by_group_sample_stack$variable))

colnames(intestine_genus_tx_group_by_group_sample_stack) <- c("Group", "Sample_Type_Env", "Genus", "CLR_Abundance")

intestine_genus_tx_group_by_group_sample_stack$Sample_Type_Env <- factor(intestine_genus_tx_group_by_group_sample_stack$Sample_Type_Env,levels= c("Duodenum_Lumen", "Duodenum_Mucosa", "Ileum_Lumen", "Ileum_Mucosa","Cecum_Lumen", "Cecum_Mucosa"))
```

Plot in heatmap
```{r}
p_heatmap_genus_sample_type_env_facet_group <- ggplot(intestine_genus_tx_group_by_group_sample_stack, aes(Sample_Type_Env, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_distiller(palette = "YlGnBu") +
  #scale_fill_viridis_c(option = "D", begin = 0.1) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Group, nrow = 1) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_group

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_group.png',width=25, height=35, plot = p_heatmap_genus_sample_type_env_facet_group)

p_heatmap_genus_sample_type_env_facet_sample_type <- ggplot(intestine_genus_tx_group_by_group_sample_stack, aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_distiller(palette = "YlGnBu") +
  #scale_fill_viridis_c(option = "D", begin = 0.1) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Sample_Type_Env, nrow = 1) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_sample_type

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_sample_type.png',width=32, height=35, plot = p_heatmap_genus_sample_type_env_facet_sample_type)
```


## 2.3 clr-aitchison (Aitchison distance)
clr- define function
```{r}
clr <- function(x) sweep(log(x), 1, rowMeans(log(x)), "-")
#
```
How does this code work? Here: 
clr <- function(x), sweep(log(X, 1, rowMeans(log(X), "-")
-sweep means to return an array of some statistic done on that array. In this case, the array that is being acted upon is log(x), or the natural log of the data frame you are inputting to the function
- 1 means do the stat on rows (in the data frame you are doing clr on, samples should be rows and features should be the columns)
-the stat being extracted from the array ln(x) here is the rowMeans(log(x)), or the average of the natural logarithms of the counts in that sample. This number is also the inverse log of the geometric mean: e^(average of the sample logarithms) = geometric mean
-the function the stat is being applied to the array log(x) is is "-" (subtraction)
-so, for each variable in each row, the average of the logs of all counts in a sample is being subtracted from the log of each variable count in that sample. This is equivalent to taking the natural log of the variable count in that sample divided the geometric mean (the way the clr transform is typically written). It is equivalent because if ln(count/geometric mean) = clr count #this is the usual way of writing clr
ln(count)- ln (geometric mean) = 
ln(count) - ln(e^average of the sample logarithms) = 
ln(count) - average of the sample logarithms = clr count #this is the way the above formula is doing it


### 2.3.1. Zero replace the SV tables and clr
Duodenum
```{r}
duodenum_SVs_zerorep <- duodenum_SVs
duodenum_SVs_zerorep[duodenum_SVs_zerorep == 0] <- 0.001
tail(duodenum_SVs_zerorep)

duodenum_genus_zerorep <- duodenum_genus
duodenum_genus_zerorep[duodenum_genus_zerorep == 0] <- 0.001
tail(duodenum_genus_zerorep)

duodenum_family_zerorep <- duodenum_family
duodenum_family_zerorep[duodenum_family_zerorep == 0] <- 0.001
tail(duodenum_family_zerorep)

duodenum_SVs_tx<- data.frame(t(clr(t(duodenum_SVs_zerorep))))
head(duodenum_SVs_tx)

duodenum_genus_tx <- data.frame(t(clr(t(duodenum_genus_zerorep))))
head(duodenum_genus_tx)

duodenum_family_tx <- data.frame(t(clr(t(duodenum_family_zerorep))))
head(duodenum_family_tx)
```
ileum
```{r}
ileum_SVs_zerorep <- ileum_SVs
ileum_SVs_zerorep[ileum_SVs_zerorep == 0] <- 0.001
tail(ileum_SVs_zerorep)

ileum_genus_zerorep <- ileum_genus
ileum_genus_zerorep[ileum_genus_zerorep == 0] <- 0.001
tail(ileum_genus_zerorep)

ileum_family_zerorep <- ileum_family
ileum_family_zerorep[ileum_family_zerorep == 0] <- 0.001
tail(ileum_family_zerorep)

ileum_SVs_tx<- data.frame(t(clr(t(ileum_SVs_zerorep))))
head(ileum_SVs_tx)

ileum_genus_tx <- data.frame(t(clr(t(ileum_genus_zerorep))))
head(ileum_genus_tx)

ileum_family_tx <- data.frame(t(clr(t(ileum_family_zerorep))))
head(ileum_family_tx)
```
cecum
```{r}
cecum_SVs_zerorep <- cecum_SVs
cecum_SVs_zerorep[cecum_SVs_zerorep == 0] <- 0.001
tail(cecum_SVs_zerorep)

cecum_genus_zerorep <- cecum_genus
cecum_genus_zerorep[cecum_genus_zerorep == 0] <- 0.001
tail(cecum_genus_zerorep)

cecum_family_zerorep <- cecum_family
cecum_family_zerorep[cecum_family_zerorep == 0] <- 0.001
tail(cecum_family_zerorep)

cecum_SVs_tx<- data.frame(t(clr(t(cecum_SVs_zerorep))))
head(cecum_SVs_tx)

cecum_genus_tx <- data.frame(t(clr(t(cecum_genus_zerorep))))
head(cecum_genus_tx)

cecum_family_tx <- data.frame(t(clr(t(cecum_family_zerorep))))
head(cecum_family_tx)
```

### 2.3.2  Mixed effect permanova SVs
Duodenum
```{r}
duodenum_dist_c45<-vegdist(as.matrix(t(duodenum_SVs_tx)) , method='aitchison')

duodenum_SVs_c45_mixed_permanova <- with(duodenum_map, adonis2(duodenum_dist_c45 ~ Sex*Genotype*Sample_Type, data = duodenum_map, permutations = 9999, strata = duodenum_map$Cohort))

capture.output(duodenum_SVs_c45_mixed_permanova, file = "./results/environmental_analysis/duodenum/permanova/aitchison/both_cohorts/duodenum_SVs_c45_mixed_permanova.csv")
```

ileum
```{r}
ileum_dist_c45<-vegdist(as.matrix(t(ileum_SVs_tx)) , method='aitchison')

ileum_SVs_c45_mixed_permanova <- with(ileum_map, adonis2(ileum_dist_c45 ~ Sex*Genotype*Sample_Type, data = ileum_map, permutations = 9999, strata = ileum_map$Cohort))

capture.output(ileum_SVs_c45_mixed_permanova, file = "./results/environmental_analysis/ileum/permanova/aitchison/both_cohorts/ileum_SVs_c45_mixed_permanova.csv")
```
cecum
```{r}
cecum_dist_c45<-vegdist(as.matrix(t(cecum_SVs_tx)) , method='aitchison')

cecum_SVs_c45_mixed_permanova <- with(cecum_map, adonis2(cecum_dist_c45 ~ Sex*Genotype*Sample_Type, data = cecum_map, permutations = 9999, strata = cecum_map$Cohort))

capture.output(cecum_SVs_c45_mixed_permanova, file = "./results/environmental_analysis/cecum/permanova/aitchison/both_cohorts/cecum_SVs_c45_mixed_permanova.csv")
```
### 2.3.3 NMDS SVs
Duodenum
```{r}
duodenum_SVs_NMDS = metaMDS(t(duodenum_SVs_tx), distance = "euclidean", parallel = 2, k = 2)

duodenum_SVs_scores = as.data.frame(scores(duodenum_SVs_NMDS))
duodenum_SVs_scores$Cohort <- duodenum_map$Cohort
duodenum_SVs_scores$Sex <- duodenum_map$Sex
duodenum_SVs_scores$Group <- duodenum_map$Group
duodenum_SVs_scores$Genotype <- duodenum_map$Genotype
duodenum_SVs_scores$Sample_Type <- duodenum_map$Sample_Type
duodenum_SVs_scores$Group_Sample_Type <- duodenum_map$Group_Sample_Type
```
```{r}
duodenum_genus_SV_counts_df_t <- data.frame(t(duodenum_genus_tx))
duodenum_SVs_genus <- envfit(ord = duodenum_SVs_NMDS, env = duodenum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(duodenum_SVs_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
duodenum_SVs_genus_scores <- as.data.frame(scores(duodenum_SVs_genus, display = "vectors"))

#add genus names
duodenum_SVs_genus_scores$Genus <- rownames(duodenum_SVs_genus_scores)

#add p values to filter
duodenum_SVs_genus_scores$pval <- duodenum_SVs_genus$vectors$pvals

duodenum_SVs_genus_scores$r2 <- duodenum_SVs_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
duodenum_SVs_genus_scores_sig <- arrange(subset(duodenum_SVs_genus_scores, pval<=0.00066), desc(r2))

write.csv(duodenum_SVs_genus_scores, './results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_SVs_genus_scores_sig.csv')

duodenum_SVs_genus_scores_sig <-duodenum_SVs_genus_scores_sig[1:10,] 
dim(duodenum_SVs_genus_scores_sig)
```
graph
```{r}
p_NMDS_duodenum_SVs_sample_type_group_genus <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.18) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  geom_segment(data = duodenum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*25, y=0, yend=NMDS2*25),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = duodenum_SVs_genus_scores_sig,
            aes(x=NMDS1*25, y=NMDS2*25, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) +
  theme(text = element_text(size = 15))  
p_NMDS_duodenum_SVs_sample_type_group_genus

ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_sample_type_genus.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_sample_type_group_genus)
```

graph
```{r}
p_NMDS_duodenum_SVs_genotype <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_genotype

p_NMDS_duodenum_SVs_sex <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_sex

#Cohort
p_NMDS_duodenum_SVs_cohort_both <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort, shape = Sample_Type)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_cohort_both


p_NMDS_duodenum_SVs_cohort <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_cohort

#Sample type
p_NMDS_duodenum_SVs_Sample_Type_both <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_Sample_Type_both

p_NMDS_duodenum_SVs_Sample_Type_both_group <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_wrap(~Group, nrow = 1)
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_SVs_Sample_Type_both_group

#Save
ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_genotype_by_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_genotype)

ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_sex_by_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_sex)

ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_cohort.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_cohort_both)

ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_cohort_by_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_cohort)

ggsave('./results/environmental_analysis/duodenum/NMDS/aitchison/both_cohorts/duodenum_c45_SVs_NMDS_clr_no_facet_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_SVs_Sample_Type_both)
```
ileum
```{r}
ileum_SVs_NMDS = metaMDS(t(ileum_SVs_tx), distance = "euclidean", parallel = 2, k = 2)

ileum_SVs_scores = as.data.frame(scores(ileum_SVs_NMDS))
ileum_SVs_scores$Cohort <- ileum_map$Cohort
ileum_SVs_scores$Sex <- ileum_map$Sex
ileum_SVs_scores$Group <- ileum_map$Group
ileum_SVs_scores$Genotype <- ileum_map$Genotype
ileum_SVs_scores$Sample_Type <- ileum_map$Sample_Type
ileum_SVs_scores$Group_Sample_Type <- ileum_map$Group_Sample_Type
```

```{r}
ileum_genus_SV_counts_df_t <- data.frame(t(ileum_genus_tx))
ileum_SVs_genus <- envfit(ord = ileum_SVs_NMDS, env = ileum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(ileum_SVs_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
ileum_SVs_genus_scores <- as.data.frame(scores(ileum_SVs_genus, display = "vectors"))

#add genus names
ileum_SVs_genus_scores$Genus <- rownames(ileum_SVs_genus_scores)

#add p values to filter
ileum_SVs_genus_scores$pval <- ileum_SVs_genus$vectors$pvals

ileum_SVs_genus_scores$r2 <- ileum_SVs_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
ileum_SVs_genus_scores_sig <- arrange(subset(ileum_SVs_genus_scores, pval<=0.00067), desc(r2))

write.csv(ileum_SVs_genus_scores, './results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_SVs_genus_scores_sig.csv')

ileum_SVs_genus_scores_sig <-ileum_SVs_genus_scores_sig[1:10,] 
dim(ileum_SVs_genus_scores_sig)
```
graph
```{r}
p_NMDS_ileum_SVs_sample_type_group_genus <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.18) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  geom_segment(data = ileum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*30, y=0, yend=NMDS2*30),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = ileum_SVs_genus_scores_sig,
            aes(x=NMDS1*30, y=NMDS2*30, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) +
  theme(text = element_text(size = 15))  
p_NMDS_ileum_SVs_sample_type_group_genus

ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_sample_type_genus.png',width=8, height=5, plot = p_NMDS_ileum_SVs_sample_type_group_genus)
```

Graph
```{r}
p_NMDS_ileum_SVs_genotype <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_genotype

p_NMDS_ileum_SVs_sex <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_sex

#Cohort
p_NMDS_ileum_SVs_cohort_both <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort, shape = Sample_Type)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_cohort_both


p_NMDS_ileum_SVs_cohort <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_cohort

#Sample type
p_NMDS_ileum_SVs_Sample_Type_both <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_Sample_Type_both

p_NMDS_ileum_SVs_Sample_Type_both_group <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_wrap(~Group, nrow = 1)
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_SVs_Sample_Type_both_group

#Save
ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_genotype_by_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_SVs_genotype)

ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_sex_by_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_SVs_sex)

ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_cohort.png',width=8, height=5, plot = p_NMDS_ileum_SVs_cohort_both)

ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_cohort_by_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_SVs_cohort)

ggsave('./results/environmental_analysis/ileum/NMDS/aitchison/both_cohorts/ileum_c45_SVs_NMDS_clr_no_facet_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_SVs_Sample_Type_both)
```
cecum
```{r}
cecum_SVs_NMDS = metaMDS(t(cecum_SVs_tx), distance = "euclidean", parallel = 2, k = 2)

cecum_SVs_scores = as.data.frame(scores(cecum_SVs_NMDS))
cecum_SVs_scores$Cohort <- cecum_map$Cohort
cecum_SVs_scores$Sex <- cecum_map$Sex
cecum_SVs_scores$Group <- cecum_map$Group
cecum_SVs_scores$Genotype <- cecum_map$Genotype
cecum_SVs_scores$Sample_Type <- cecum_map$Sample_Type
cecum_SVs_scores$Group_Sample_Type <- cecum_map$Group_Sample_Type
```

```{r}
cecum_genus_SV_counts_df_t <- data.frame(t(cecum_genus_tx))
cecum_SVs_genus <- envfit(ord = cecum_SVs_NMDS, env = cecum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(cecum_SVs_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
cecum_SVs_genus_scores <- as.data.frame(scores(cecum_SVs_genus, display = "vectors"))

#add genus names
cecum_SVs_genus_scores$Genus <- rownames(cecum_SVs_genus_scores)

#add p values to filter
cecum_SVs_genus_scores$pval <- cecum_SVs_genus$vectors$pvals

cecum_SVs_genus_scores$r2 <- cecum_SVs_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
cecum_SVs_genus_scores_sig <- arrange(subset(cecum_SVs_genus_scores, pval<=0.00067), desc(r2))

write.csv(cecum_SVs_genus_scores, './results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_SVs_genus_scores_sig.csv')

cecum_SVs_genus_scores_sig <-cecum_SVs_genus_scores_sig[1:10,] 
dim(cecum_SVs_genus_scores_sig)
```
graph
```{r}
p_NMDS_cecum_SVs_sample_type_group_genus <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.18) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  geom_segment(data = cecum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*30, y=0, yend=NMDS2*30),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = cecum_SVs_genus_scores_sig,
            aes(x=NMDS1*30, y=NMDS2*30, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) +
  theme(text = element_text(size = 15))  
p_NMDS_cecum_SVs_sample_type_group_genus

ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_sample_type_genus.png',width=8, height=5, plot = p_NMDS_cecum_SVs_sample_type_group_genus)
```
Graph
```{r}
p_NMDS_cecum_SVs_genotype <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_genotype

p_NMDS_cecum_SVs_sex <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_sex

#Cohort
p_NMDS_cecum_SVs_cohort_both <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort, shape = Sample_Type)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_cohort_both


p_NMDS_cecum_SVs_cohort <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_cohort

#Sample type
p_NMDS_cecum_SVs_Sample_Type_both <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_Sample_Type_both

p_NMDS_cecum_SVs_Sample_Type_both_group <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type, shape = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color = Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Clr aitchison") +
  facet_wrap(~Group, nrow = 1)
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_SVs_Sample_Type_both_group

#Save
ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_genotype_by_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_SVs_genotype)

ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_sex_by_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_SVs_sex)

ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_cohort.png',width=8, height=5, plot = p_NMDS_cecum_SVs_cohort_both)

ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_cohort_by_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_SVs_cohort)

ggsave('./results/environmental_analysis/cecum/NMDS/aitchison/both_cohorts/cecum_c45_SVs_NMDS_clr_no_facet_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_SVs_Sample_Type_both)
```

### 2.3.4 All Intestine NMDS for Aitchison NMDS and permanova
Combine SV tables
```{r}
duo_ile_SVs <- merge(duodenum_SVs, ileum_SVs, by = 'row.names', all = TRUE)
rownames(duo_ile_SVs) <- duo_ile_SVs$Row.names
duo_ile_SVs <- duo_ile_SVs[, 2:309]
dim(duo_ile_SVs)

cec_fec_SVs <- merge(cecum_SVs, feces_SVs, by = 'row.names', all = TRUE)
rownames(cec_fec_SVs) <- cec_fec_SVs$Row.names
cec_fec_SVs <- cec_fec_SVs[, 2:232]
dim(cec_fec_SVs)

all_SVs_names <- merge(duo_ile_SVs, cec_fec_SVs, by = 'row.names', all = TRUE)
rownames(all_SVs_names) <- all_SVs_names$Row.names

all_SVs <- all_SVs_names[, 2:540]
dim(all_SVs)
dim(all_envs_map)

all_SVs[is.na(all_SVs)] <- 0
```

Combine family tables
```{r}
duo_ile_family <- merge(duodenum_family, ileum_family, by = 'row.names', all = TRUE)
rownames(duo_ile_family) <- duo_ile_family$Row.names
duo_ile_family <- duo_ile_family[, 2:309]
dim(duo_ile_family)

cec_fec_family <- merge(cecum_family, feces_family, by = 'row.names', all = TRUE)
rownames(cec_fec_family) <- cec_fec_family$Row.names
cec_fec_family <- cec_fec_family[, 2:232]
dim(cec_fec_family)

all_family_names <- merge(duo_ile_family, cec_fec_family, by = 'row.names', all = TRUE)
rownames(all_family_names) <- all_family_names$Row.names

all_family <- all_family_names[, 2:540]
dim(all_family)
dim(all_envs_map)
```

replace NAs with 0
```{r}
all_family[is.na(all_family)] <- 0
```

#### 2.3.4.1 SVs clr aitchison 
Zero replace the SV table
```{r}
all_SVs_zerorep <- all_SVs
all_SVs_zerorep[all_SVs_zerorep == 0] <- 0.001
tail(all_SVs_zerorep)
```

```{r}
all_SVs_tx<- data.frame(t(clr(t(all_SVs_zerorep))))
head(all_SVs_tx)
```


Get distances
```{r}
all_envs_clr_SVs_dist_c45<-vegdist(as.matrix(t(all_SVs_tx)) , method='euclidean')
```

Both Cohorts, all weeks
Permanova- mixed effects 
```{r}
all_SVs_clr_perm_c45_mixed_effects <- with(all_envs_map, adonis2(all_envs_clr_SVs_dist_c45 ~ Sex*Genotype*Sample_Type*Section, data = all_envs_map, permutations = 9999, strata = all_envs_map$Cohort))

capture.output(all_SVs_clr_perm_c45_mixed_effects,file="./results/environmental_analysis/all_envs/permanova/SVs_clr/all_envs_c45s_permanova_mixed_effects.csv")

```


Get dispersion and centroid
```{r}

all_envs_beta_dist_aitchison_c45 <- betadisper(all_envs_clr_SVs_dist_c45, group = all_envs_map$Group_Sample_Type_env, type = "centroid", bias.adjust = FALSE, sqrt.dist = FALSE)

all_envs_beta_dist_aitchison_c45_centroids <- as.data.frame(all_envs_beta_dist_aitchison_c45$centroids)

all_envs_beta_dist_aitchison_c45_distances <- as.data.frame(all_envs_beta_dist_aitchison_c45$distances)
colnames(all_envs_beta_dist_aitchison_c45_distances) <- c("Distances")


all_envs_beta_dist_aitchison_c45_distances$Cohort <- all_envs_map$Cohort
all_envs_beta_dist_aitchison_c45_distances$Sex <- all_envs_map$Sex
all_envs_beta_dist_aitchison_c45_distances$Genotype <- all_envs_map$Genotype
all_envs_beta_dist_aitchison_c45_distances$Group <- all_envs_map$Group
all_envs_beta_dist_aitchison_c45_distances$Mouse <- all_envs_map$Mouse
all_envs_beta_dist_aitchison_c45_distances$Sample_Type <- all_envs_map$Sample_Type
all_envs_beta_dist_aitchison_c45_distances$Group_Sample_Type_env <- all_envs_map$Group_Sample_Type_env
all_envs_beta_dist_aitchison_c45_distances$Section <- all_envs_map$Section

all_envs_beta_dist_aitchison_c45_distances_stats <- all_envs_beta_dist_aitchison_c45_distances %>%                               
  group_by(Group_Sample_Type_env) %>% 
  summarize(min = min(Distances),
            q1 = quantile(Distances, 0.25),
            median = median(Distances),
            mean = mean(Distances),
            q3 = quantile(Distances, 0.75),
            max = max(Distances))

write.csv(all_envs_beta_dist_aitchison_c45_distances_stats, "./results/environmental_analysis/all_envs/beta_disp/all_envs_SVs_c45s_distances.csv")

write.csv(all_envs_beta_dist_aitchison_c45_centroids, "./results/environmental_analysis/all_envs/beta_disp/all_envs_SVs_c45s_centroids.csv")
```


NMDS
```{r}
all_SVs_clr_c45_weekall_NMDS = metaMDS(all_envs_clr_SVs_dist_c45, parallel = 2, k = 2)
```
stress is 0.15

Add metadata
```{r}
all_SVs_clr_c45_weekall_scores = as.data.frame(scores(all_SVs_clr_c45_weekall_NMDS))
all_SVs_clr_c45_weekall_scores$Cohort <- all_envs_map$Cohort
all_SVs_clr_c45_weekall_scores$Sex <- all_envs_map$Sex
all_SVs_clr_c45_weekall_scores$Group <- all_envs_map$Group
all_SVs_clr_c45_weekall_scores$Genotype <- all_envs_map$Genotype
all_SVs_clr_c45_weekall_scores$Sample_Type_Env <- all_envs_map$Sample_Type_Env
all_SVs_clr_c45_weekall_scores$Sample_Type <- all_envs_map$Sample_Type
all_SVs_clr_c45_weekall_scores$Section <- all_envs_map$Section
all_SVs_clr_c45_weekall_scores$Week <- all_envs_map$Week
all_SVs_clr_c45_weekall_scores$Group_Week <- all_envs_map$Group_Week
```

Graph
Look at the different sections

```{r}
p_NMDS_all_SVs_clr_c45_all_envs_sections <- ggplot(data = all_SVs_clr_c45_weekall_scores[(all_SVs_clr_c45_weekall_scores$Week=="10"),], 
                           aes(x = NMDS1, y = NMDS2, color = Section, fill = Section)) + 
  geom_point(aes(shape = Sample_Type)) + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
  scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  #scale_color_brewer(palette = "Paired") +
   #scale_fill_brewer(palette = "Paired") +
  theme_classic() +
  #facet_grid(Genotype~Week) +
  theme(text = element_text(size = 15)) 
p_NMDS_all_SVs_clr_c45_all_envs_sections

p_NMDS_all_SVs_clr_c45_all_envs_sections_facet_group <- p_NMDS_all_SVs_clr_c45_all_envs_sections +
  facet_grid(Genotype~Sex)
p_NMDS_all_SVs_clr_c45_all_envs_sections_facet_group

p_NMDS_all_SVs_clr_c45_all_envs_sections_sample <- ggplot(data = all_SVs_clr_c45_weekall_scores[(all_SVs_clr_c45_weekall_scores$Week=="10"),], 
                           aes(x = NMDS1, y = NMDS2, color = Sample_Type_Env, fill = Sample_Type_Env)) + 
  geom_point() + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
  #scale_color_brewer(palette = "Paired") +
   #scale_fill_brewer(palette = "Paired") +
   scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  theme_classic() +
  #facet_grid(Genotype~Week) +
  theme(text = element_text(size = 15)) 
p_NMDS_all_SVs_clr_c45_all_envs_sections_sample

p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group <- p_NMDS_all_SVs_clr_c45_all_envs_sections_sample +
  facet_grid(Genotype~Sex)
p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group


#Save
#ggsave('./results/environmental_analysis/all_envs/NMDS/SVs_clr/NMDS_all_SVs_clr_c45_all_envs_sections.png',width=8, height=5, plot = p_NMDS_all_SVs_clr_c45_all_envs_sections)

#ggsave('./results/environmental_analysis/all_envs/NMDS/SVs_clr/NMDS_all_SVs_clr_c45_all_envs_sections_facet_group.png',width=8, height=5, plot = p_NMDS_all_SVs_clr_c45_all_envs_sections_facet_group)

#ggsave('./results/environmental_analysis/all_envs/NMDS/SVs_clr/NMDS_all_SVs_clr_c45_all_envs_sections_sample.png',width=8, height=5, plot = p_NMDS_all_SVs_clr_c45_all_envs_sections_sample)

#ggsave('./results/environmental_analysis/all_envs/NMDS/SVs_clr/NMDS_all_SVs_clr_c45_all_envs_sections_sample_facet_group.png',width=8, height=5, plot = p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group)
```


#### 2.3.4.2 family clr aitchison
Zero replace the SV table
```{r}
all_family_zerorep <- all_family
all_family_zerorep[all_family_zerorep == 0] <- 0.001
tail(all_family_zerorep)

all_family_tx<- data.frame(t(clr(t(all_family_zerorep))))
head(all_family_tx)

#Get distances
all_envs_clr_family_dist_c45<-vegdist(as.matrix(t(all_family_tx)) , method='aitchison')
```

Both Cohorts, all weeks
Permanova- mixed effects 
```{r}
all_family_clr_perm_c45_mixed_effects <- with(all_envs_map, adonis2(all_envs_clr_family_dist_c45 ~ Sex*Genotype*Sample_Type_Env, data = all_envs_map, permutations = 999, strata = all_envs_map$Cohort))

all_family_clr_perm_c45_mixed_effects

capture.output(all_family_clr_perm_c45_mixed_effects,file="./results/environmental_analysis/all_envs/permanova/family_clr/all_envs_c45s_permanova_mixed_effects.csv")

```
NMDS
```{r}
all_family_clr_c45_weekall_NMDS = metaMDS(all_envs_clr_family_dist_c45, parallel = 2, k = 2)
```
stress is 0.14
Add metadata
```{r}
all_family_clr_c45_weekall_scores = as.data.frame(scores(all_family_clr_c45_weekall_NMDS))
all_family_clr_c45_weekall_scores$Cohort <- all_envs_map$Cohort
all_family_clr_c45_weekall_scores$Sex <- all_envs_map$Sex
all_family_clr_c45_weekall_scores$Group <- all_envs_map$Group
all_family_clr_c45_weekall_scores$Genotype <- all_envs_map$Genotype
all_family_clr_c45_weekall_scores$Sample_Type_Env <- all_envs_map$Sample_Type_Env
all_family_clr_c45_weekall_scores$Sample_Type <- all_envs_map$Sample_Type
all_family_clr_c45_weekall_scores$Section <- all_envs_map$Section
all_family_clr_c45_weekall_scores$Week <- all_envs_map$Week
all_family_clr_c45_weekall_scores$Group_Week <- all_envs_map$Group_Week
```
Graph
```{r}
p_NMDS_all_family_clr_c45_all_envs_sections <- ggplot(data = all_family_clr_c45_weekall_scores[(all_family_clr_c45_weekall_scores$Week=="10"),], 
                           aes(x = NMDS1, y = NMDS2, color = Section, fill = Section)) + 
  geom_point(aes(shape = Sample_Type)) + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
     scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  #scale_color_brewer(palette = "Paired") +
   #scale_fill_brewer(palette = "Paired") +
  theme_classic() +
  labs(title= "") +
  #facet_grid(Genotype~Week) +
  theme(text = element_text(size = 15)) 
p_NMDS_all_family_clr_c45_all_envs_sections

p_NMDS_all_family_clr_c45_all_envs_sections_facet_group <- p_NMDS_all_family_clr_c45_all_envs_sections +
  facet_grid(Genotype~Sex)
p_NMDS_all_family_clr_c45_all_envs_sections_facet_group

p_NMDS_all_family_clr_c45_all_envs_sections_sample <- ggplot(data = all_family_clr_c45_weekall_scores[(all_family_clr_c45_weekall_scores$Week=="10"),], 
                           aes(x = NMDS1, y = NMDS2, 
                               color = Sample_Type_Env, fill = Sample_Type_Env)) + 
  geom_point() + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
  #scale_color_brewer(palette = "Paired") +
   #scale_fill_brewer(palette = "Paired") +
     scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  theme_classic() +
  labs(title= "") +
  #facet_grid(Genotype~Week) +
  theme(text = element_text(size = 15)) 
p_NMDS_all_family_clr_c45_all_envs_sections_sample

p_NMDS_all_family_clr_c45_all_envs_sections_sample_group <- p_NMDS_all_family_clr_c45_all_envs_sections_sample +
  facet_grid(Genotype~Sex)
p_NMDS_all_family_clr_c45_all_envs_sections_sample_group

p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces <- ggplot(data = all_family_clr_c45_weekall_scores[!(all_family_clr_c45_weekall_scores$Section=="Feces"),], 
                           aes(x = NMDS1, y = NMDS2, color = Sample_Type_Env, fill = Sample_Type_Env)) + 
  geom_point() + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
  #scale_color_brewer(palette = "Paired") +
   #scale_fill_brewer(palette = "Paired") +
     scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  theme_classic() +
  labs(title= "") +
  #facet_grid(Genotype~Week) +
  theme(text = element_text(size = 15)) 
p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces

#ggsave('./results/environmental_analysis/all_envs/NMDS/family_clr/NMDS_all_family_clr_c45_all_envs_sections.png',width=8, height=5, plot = p_NMDS_all_family_clr_c45_all_envs_sections)

#ggsave('./results/environmental_analysis/all_envs/NMDS/family_clr/NMDS_all_family_clr_c45_all_envs_sections_facet_group.png',width=8, height=5, plot = p_NMDS_all_family_clr_c45_all_envs_sections_facet_group)

#ggsave('./results/environmental_analysis/all_envs/NMDS/family_clr/NMDS_all_family_clr_c45_all_envs_sections_sample.png',width=8, height=5, plot = p_NMDS_all_family_clr_c45_all_envs_sections_sample)

#ggsave('./results/environmental_analysis/all_envs/NMDS/family_clr/NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces.png',width=8, height=5, plot =p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces)

#ggsave('./results/environmental_analysis/all_envs/NMDS/family_clr/NMDS_all_family_clr_c45_all_envs_sections_sample_facet_group.png',width=8, height=5, plot = p_NMDS_all_family_clr_c45_all_envs_sections_sample_group)
```







## 2.4 unweighted unifrac 
Make distance object
```{r}
duodenum_unweighted_unifrac_dist <- as.dist(duodenum_unweighted_filtered_c45)
ileum_unweighted_unifrac_dist <- as.dist(ileum_unweighted_filtered_c45)
cecum_unweighted_unifrac_dist <- as.dist(cecum_unweighted_filtered_c45)
```

Output descriptive stats about dispersion and centroids
```{r}
#duodenum
duodenum_beta_dist_unweighted_unifrac_c45 <- betadisper(duodenum_unweighted_unifrac_dist, group = duodenum_map_alpha$Group_Sample_Type, type = "centroid", bias.adjust = FALSE, sqrt.dist = FALSE)

duodenum_beta_dist_unweighted_unifrac_c45_distances <- as.data.frame(duodenum_beta_dist_unweighted_unifrac_c45$distances)
colnames(duodenum_beta_dist_unweighted_unifrac_c45_distances) <- c("Distances")

duodenum_beta_dist_unweighted_unifrac_c45_distances_centroids <- as.data.frame(duodenum_beta_dist_unweighted_unifrac_c45$centroids)

duodenum_beta_dist_unweighted_unifrac_c45_distances$Group_Sample_Type <- duodenum_map_alpha$Group_Sample_Type

duodenum_beta_dist_unweighted_unifrac_c45_distances_stats <- duodenum_beta_dist_unweighted_unifrac_c45_distances %>%                               
  group_by(Group_Sample_Type) %>% 
  summarize(min = min(Distances),
            q1 = quantile(Distances, 0.25),
            median = median(Distances),
            mean = mean(Distances),
            q3 = quantile(Distances, 0.75),
            max = max(Distances))

write.csv(duodenum_beta_dist_unweighted_unifrac_c45_distances_stats, "./results/environmental_analysis/duodenum/beta_disp/unweighted_unifrac/duodenum_unweighted_unifrac_c45s_distances.csv")

write.csv(duodenum_beta_dist_unweighted_unifrac_c45_distances_centroids, "./results/environmental_analysis/duodenum/beta_disp/unweighted_unifrac/duodenum_unweighted_unifrac_c45s_centroids.csv")

#ileum
ileum_beta_dist_unweighted_unifrac_c45 <- betadisper(ileum_unweighted_unifrac_dist, group = ileum_map_alpha$Group_Sample_Type, type = "centroid", bias.adjust = FALSE, sqrt.dist = FALSE)

ileum_beta_dist_unweighted_unifrac_c45_distances <- as.data.frame(ileum_beta_dist_unweighted_unifrac_c45$distances)
colnames(ileum_beta_dist_unweighted_unifrac_c45_distances) <- c("Distances")

ileum_beta_dist_unweighted_unifrac_c45_distances_centroids <- as.data.frame(ileum_beta_dist_unweighted_unifrac_c45$centroids)

ileum_beta_dist_unweighted_unifrac_c45_distances$Group_Sample_Type <- ileum_map_alpha$Group_Sample_Type

ileum_beta_dist_unweighted_unifrac_c45_distances_stats <- ileum_beta_dist_unweighted_unifrac_c45_distances %>%                               
  group_by(Group_Sample_Type) %>% 
  summarize(min = min(Distances),
            q1 = quantile(Distances, 0.25),
            median = median(Distances),
            mean = mean(Distances),
            q3 = quantile(Distances, 0.75),
            max = max(Distances))

write.csv(ileum_beta_dist_unweighted_unifrac_c45_distances_stats, "./results/environmental_analysis/ileum/beta_disp/unweighted_unifrac/ileum_unweighted_unifrac_c45s_distances.csv")

write.csv(ileum_beta_dist_unweighted_unifrac_c45_distances_centroids, "./results/environmental_analysis/ileum/beta_disp/unweighted_unifrac/ileum_unweighted_unifrac_c45s_centroids.csv")

#cecum
cecum_beta_dist_unweighted_unifrac_c45 <- betadisper(cecum_unweighted_unifrac_dist, group = cecum_map_alpha$Group_Sample_Type, type = "centroid", bias.adjust = FALSE, sqrt.dist = FALSE)

cecum_beta_dist_unweighted_unifrac_c45_distances <- as.data.frame(cecum_beta_dist_unweighted_unifrac_c45$distances)
colnames(cecum_beta_dist_unweighted_unifrac_c45_distances) <- c("Distances")

cecum_beta_dist_unweighted_unifrac_c45_distances_centroids <- as.data.frame(cecum_beta_dist_unweighted_unifrac_c45$centroids)

cecum_beta_dist_unweighted_unifrac_c45_distances$Group_Sample_Type <- cecum_map_alpha$Group_Sample_Type

cecum_beta_dist_unweighted_unifrac_c45_distances_stats <- cecum_beta_dist_unweighted_unifrac_c45_distances %>%                               
  group_by(Group_Sample_Type) %>% 
  summarize(min = min(Distances),
            q1 = quantile(Distances, 0.25),
            median = median(Distances),
            mean = mean(Distances),
            q3 = quantile(Distances, 0.75),
            max = max(Distances))

write.csv(cecum_beta_dist_unweighted_unifrac_c45_distances_stats, "./results/environmental_analysis/cecum/beta_disp/unweighted_unifrac/cecum_unweighted_unifrac_c45s_distances.csv")

write.csv(cecum_beta_dist_unweighted_unifrac_c45_distances_centroids, "./results/environmental_analysis/cecum/beta_disp/unweighted_unifrac/cecum_unweighted_unifrac_c45s_centroids.csv")
```
### 2.4.1 unweighted unifrac mixed effects permanovas
1. Duodenum
```{r}
#mixed effects
duodenum_unweighted_unifrac_c45_mixed_permanova <- with(duodenum_map_alpha, adonis2(duodenum_unweighted_unifrac_dist ~ Sex*Genotype*Sample_Type, data = duodenum_map_alpha, permutations = 9999, strata = duodenum_map_alpha$Cohort))

capture.output(duodenum_unweighted_unifrac_c45_mixed_permanova, file = './results/environmental_analysis/duodenum/permanova/unweighted_unifrac/both_cohorts/duodenum_unweighted_unifrac_c45_mixed_permanova.csv')
```
2. Ileum
```{r}
#mixed effects
ileum_unweighted_unifrac_c45_mixed_permanova <- with(ileum_map_alpha, adonis2(ileum_unweighted_unifrac_dist ~ Sex*Genotype*Sample_Type, data = ileum_map_alpha, permutations = 9999, strata = ileum_map_alpha$Cohort))

capture.output(ileum_unweighted_unifrac_c45_mixed_permanova, file = './results/environmental_analysis/ileum/permanova/unweighted_unifrac/both_cohorts/ileum_unweighted_unifrac_c45_mixed_permanova.csv')
```
3. Cecum
```{r}
#mixed effects
cecum_unweighted_unifrac_c45_mixed_permanova <- with(cecum_map_alpha, adonis2(cecum_unweighted_unifrac_dist ~ Sex*Genotype*Sample_Type, data = cecum_map_alpha, permutations = 9999, strata = cecum_map_alpha$Cohort))

capture.output(cecum_unweighted_unifrac_c45_mixed_permanova, file = './results/environmental_analysis/cecum/permanova/unweighted_unifrac/both_cohorts/cecum_unweighted_unifrac_c45_mixed_permanova.csv')
```

### 2.4.2 unweighted unifrac NMDS
1. Duodenum
```{r}
duodenum_unweighted_unifrac_NMDS = metaMDS(duodenum_unweighted_unifrac_dist, parallel = 2, k = 2)
```
stress 0.167
To add genus scores:
To add genera to the NMDS for the sample types: 
```{r}
#duodenum_genus_SV_counts_df <- duodenum_genus %>% select(-contains("461_DL"))

#dim(duodenum_genus_SV_counts_df)
#dim(duodenum_unweighted_filtered_c45)
 #rarefy
#duodenum_genus_SV_counts_colmin <- min(colSums(duodenum_genus_SV_counts_df))
#duodenum_genus_rarefied_df <- rrarefy(t(duodenum_genus_SV_counts_df), sample = duodenum_genus_SV_counts_colmin)
```

```{r}
duodenum_genus_SV_counts_df_t <- data.frame(t(duodenum_genus_tx %>% select(-contains("461_DL"))))
duodenum_unweighted_unifrac_genus <- envfit(ord = duodenum_unweighted_unifrac_NMDS, env = duodenum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(duodenum_unweighted_unifrac_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
duodenum_unweighted_unifrac_genus_scores <- as.data.frame(scores(duodenum_unweighted_unifrac_genus, display = "vectors"))

#add genus names
duodenum_unweighted_unifrac_genus_scores$Genus <- rownames(duodenum_unweighted_unifrac_genus_scores)

#add p values to filter
duodenum_unweighted_unifrac_genus_scores$pval <- duodenum_unweighted_unifrac_genus$vectors$pvals

duodenum_unweighted_unifrac_genus_scores$r2 <- duodenum_unweighted_unifrac_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
duodenum_unweighted_unifrac_genus_scores_sig <- arrange(subset(duodenum_unweighted_unifrac_genus_scores, pval<=0.00066), desc(r2))

write.csv(duodenum_unweighted_unifrac_genus_scores, './results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/duodenum_unweighted_unifrac_genus_scores_sig_clrgenus.csv')

duodenum_unweighted_unifrac_genus_scores_sig <-duodenum_unweighted_unifrac_genus_scores_sig[1:10,] 
dim(duodenum_unweighted_unifrac_genus_scores_sig)


#and add metadata to scores DF
duodenum_unweighted_unifrac_scores = as.data.frame(scores(duodenum_unweighted_unifrac_NMDS))
duodenum_unweighted_unifrac_scores$Cohort <- duodenum_map_alpha$Cohort
duodenum_unweighted_unifrac_scores$Sex <- duodenum_map_alpha$Sex
duodenum_unweighted_unifrac_scores$Group <- duodenum_map_alpha$Group
duodenum_unweighted_unifrac_scores$Genotype <- duodenum_map_alpha$Genotype
duodenum_unweighted_unifrac_scores$Sample_Type <- duodenum_map_alpha$Sample_Type
duodenum_unweighted_unifrac_scores$Group_Sample_Type <- duodenum_map_alpha$Group_Sample_type
```
graph
```{r}
p_NMDS_duodenum_unweighted_unifrac_genotype <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_unweighted_unifrac_genotype

p_NMDS_duodenum_unweighted_unifrac_sex <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_unweighted_unifrac_sex

#COhort
p_NMDS_duodenum_unweighted_unifrac_cohort <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_unweighted_unifrac_cohort

#Sample type
p_NMDS_duodenum_unweighted_unifrac_sample_type <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  theme(text = element_text(size = 15)) 
p_NMDS_duodenum_unweighted_unifrac_sample_type

p_NMDS_duodenum_unweighted_unifrac_sample_type_group <-
  p_NMDS_duodenum_unweighted_unifrac_sample_type +
  facet_grid(Genotype~Sex)
p_NMDS_duodenum_unweighted_unifrac_sample_type_group

#add genus
p_NMDS_duodenum_unweighted_unifrac_sample_type_group_genus <- p_NMDS_duodenum_unweighted_unifrac_sample_type +
  geom_segment(data = duodenum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.5, y=0, yend=NMDS2*0.5),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = duodenum_unweighted_unifrac_genus_scores_sig,
            aes(x=NMDS1*0.5, y=NMDS2*0.5, label = Genus), cex = 4, direction = "both", segment.size = 0.25) 
p_NMDS_duodenum_unweighted_unifrac_sample_type_group_genus

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sex_within_genotype_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_genotype)

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_genotype_within_sex_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_sex)

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_cohort.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_cohort)

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_sample_type)

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_group.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_sample_type_group)

ggsave('./results/environmental_analysis/duodenum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_genus.png',width=8, height=5, plot = p_NMDS_duodenum_unweighted_unifrac_sample_type_group_genus)
```

2. Ileum

```{r}
ileum_unweighted_unifrac_NMDS = metaMDS(ileum_unweighted_unifrac_dist, parallel = 2, k = 2)
```
stress 0.19

To add genus scores:
To add genera to the NMDS for the sample types: 
```{r}
#ileum_genus_SV_counts_df <- ileum_genus 

#dim(ileum_genus_SV_counts_df)
#dim(ileum_unweighted_filtered_c45)
 #rarefy
#ileum_genus_SV_counts_colmin <- min(colSums(ileum_genus_SV_counts_df))
#ileum_genus_rarefied_df <- rrarefy(t(ileum_genus_SV_counts_df), sample = ileum_genus_SV_counts_colmin)
```

```{r}
ileum_genus_SV_counts_df_t <- data.frame(t(ileum_genus_tx))
ileum_unweighted_unifrac_genus <- envfit(ord = ileum_unweighted_unifrac_NMDS, env = ileum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(ileum_unweighted_unifrac_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
ileum_unweighted_unifrac_genus_scores <- as.data.frame(scores(ileum_unweighted_unifrac_genus, display = "vectors"))

#add genus names
ileum_unweighted_unifrac_genus_scores$Genus <- rownames(ileum_unweighted_unifrac_genus_scores)

#add p values to filter
ileum_unweighted_unifrac_genus_scores$pval <- ileum_unweighted_unifrac_genus$vectors$pvals
ileum_unweighted_unifrac_genus_scores$r2 <- ileum_unweighted_unifrac_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
ileum_unweighted_unifrac_genus_scores_sig <- arrange(subset(ileum_unweighted_unifrac_genus_scores, pval<=0.009), desc(r2))

write.csv(ileum_unweighted_unifrac_genus_scores, './results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/ileum_unweighted_unifrac_genus_scores_sig_clr.csv')

ileum_unweighted_unifrac_genus_scores_sig <-ileum_unweighted_unifrac_genus_scores_sig[1:10,] 
dim(ileum_unweighted_unifrac_genus_scores_sig)
#and add metadata to scores DF
ileum_unweighted_unifrac_scores = as.data.frame(scores(ileum_unweighted_unifrac_NMDS))
ileum_unweighted_unifrac_scores$Cohort <- ileum_map_alpha$Cohort
ileum_unweighted_unifrac_scores$Sex <- ileum_map_alpha$Sex
ileum_unweighted_unifrac_scores$Group <- ileum_map_alpha$Group
ileum_unweighted_unifrac_scores$Genotype <- ileum_map_alpha$Genotype
ileum_unweighted_unifrac_scores$Sample_Type <- ileum_map_alpha$Sample_Type
ileum_unweighted_unifrac_scores$Group_Sample_Type <- ileum_map_alpha$Group_Sample_type
```

graph
```{r}
p_NMDS_ileum_unweighted_unifrac_genotype <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_unweighted_unifrac_genotype

p_NMDS_ileum_unweighted_unifrac_sex <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_unweighted_unifrac_sex

#COhort
p_NMDS_ileum_unweighted_unifrac_cohort <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_unweighted_unifrac_cohort

#Sample type
p_NMDS_ileum_unweighted_unifrac_sample_type <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  theme(text = element_text(size = 15)) 
p_NMDS_ileum_unweighted_unifrac_sample_type

p_NMDS_ileum_unweighted_unifrac_sample_type_group <-
  p_NMDS_ileum_unweighted_unifrac_sample_type +
  facet_grid(Genotype~Sex)
p_NMDS_ileum_unweighted_unifrac_sample_type_group

#add genus
p_NMDS_ileum_unweighted_unifrac_sample_type_group_genus <- p_NMDS_ileum_unweighted_unifrac_sample_type +
  geom_segment(data = ileum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.5, y=0, yend=NMDS2*0.5),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = ileum_unweighted_unifrac_genus_scores_sig,
            aes(x=NMDS1*0.5, y=NMDS2*0.5, label = Genus), max.overlaps = Inf, cex = 4, direction = "both", segment.size = 0.25) 
p_NMDS_ileum_unweighted_unifrac_sample_type_group_genus

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sex_within_genotype_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_genotype)

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_genotype_within_sex_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_sex)

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_cohort.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_cohort)

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_sample_type)

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_group.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_sample_type_group)

ggsave('./results/environmental_analysis/ileum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_genus.png',width=8, height=5, plot = p_NMDS_ileum_unweighted_unifrac_sample_type_group_genus)
```

3. Cecum
```{r}
cecum_unweighted_unifrac_NMDS = metaMDS(cecum_unweighted_unifrac_dist, parallel = 2, k = 2)
```
stress 0.23
To add genus scores:
To add genera to the NMDS for the sample types: 
```{r}
#cecum_genus_SV_counts_df <- cecum_genus %>% select(-contains("414_CL"))

#dim(cecum_genus_SV_counts_df)
#dim(cecum_unweighted_filtered_c45)
 #rarefy
#cecum_genus_SV_counts_colmin <- min(colSums(cecum_genus_SV_counts_df))
#cecum_genus_rarefied_df <- rrarefy(t(cecum_genus_SV_counts_df), sample = cecum_genus_SV_counts_colmin)
```

```{r}
cecum_genus_SV_counts_df_t <- data.frame(t(cecum_genus_tx %>% select(-contains("414_CL"))))
cecum_unweighted_unifrac_genus <- envfit(ord = cecum_unweighted_unifrac_NMDS, env = cecum_genus_SV_counts_df_t, permutations = 9999, na.rm = TRUE)
head(cecum_unweighted_unifrac_genus)
```

Select "significant" genera for adding to sample type plots
```{r}
cecum_unweighted_unifrac_genus_scores <- as.data.frame(scores(cecum_unweighted_unifrac_genus, display = "vectors"))

#add genus names
cecum_unweighted_unifrac_genus_scores$Genus <- rownames(cecum_unweighted_unifrac_genus_scores)

#add p values to filter
cecum_unweighted_unifrac_genus_scores$pval <- cecum_unweighted_unifrac_genus$vectors$pvals

cecum_unweighted_unifrac_genus_scores$r2 <- cecum_unweighted_unifrac_genus$vectors$r
#cut off at PDF adjusted p = 0.05 and R2 
cecum_unweighted_unifrac_genus_scores_sig <- arrange(subset(cecum_unweighted_unifrac_genus_scores, pval<=0.00065), desc(r2))

write.csv(cecum_unweighted_unifrac_genus_scores, './results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/cecum_unweighted_unifrac_genus_scores_sig_clr.csv')

cecum_unweighted_unifrac_genus_scores_sig <-cecum_unweighted_unifrac_genus_scores_sig[1,]
dim(cecum_unweighted_unifrac_genus_scores_sig)


#and add metadata to scores DF
cecum_unweighted_unifrac_scores = as.data.frame(scores(cecum_unweighted_unifrac_NMDS))
cecum_unweighted_unifrac_scores$Cohort <- cecum_map_alpha$Cohort
cecum_unweighted_unifrac_scores$Sex <- cecum_map_alpha$Sex
cecum_unweighted_unifrac_scores$Group <- cecum_map_alpha$Group
cecum_unweighted_unifrac_scores$Genotype <- cecum_map_alpha$Genotype
cecum_unweighted_unifrac_scores$Sample_Type <- cecum_map_alpha$Sample_Type
cecum_unweighted_unifrac_scores$Group_Sample_Type <- cecum_map_alpha$Group_Sample_type
```

graph
```{r}
cecum_unweighted_unifrac_scores <- cecum_unweighted_unifrac_scores[!(row.names(cecum_unweighted_unifrac_scores)=="452_CC"),]

p_NMDS_cecum_unweighted_unifrac_genotype <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Genotype~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_unweighted_unifrac_genotype

p_NMDS_cecum_unweighted_unifrac_sex <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(Sex~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_unweighted_unifrac_sex

#COhort
p_NMDS_cecum_unweighted_unifrac_cohort <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Cohort)) + 
    scale_color_manual(values = cohort_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Cohort, color = Cohort), alpha =0.18) +
   scale_fill_manual(values = cohort_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  facet_grid(~Sample_Type) +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_unweighted_unifrac_cohort

#Sample type
p_NMDS_cecum_unweighted_unifrac_sample_type <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.18) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_classic() +
  labs(title= "Unweighted Unifrac") +
  theme(text = element_text(size = 15)) 
p_NMDS_cecum_unweighted_unifrac_sample_type

p_NMDS_cecum_unweighted_unifrac_sample_type_group <-
  p_NMDS_cecum_unweighted_unifrac_sample_type +
  facet_grid(Genotype~Sex)
p_NMDS_cecum_unweighted_unifrac_sample_type_group

#add genus
p_NMDS_cecum_unweighted_unifrac_sample_type_group_genus <- p_NMDS_cecum_unweighted_unifrac_sample_type +
  geom_segment(data = cecum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.05, y=0, yend=NMDS2*0.05),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = cecum_unweighted_unifrac_genus_scores_sig,
            aes(x=NMDS1*0.05, y=NMDS2*0.05, label = Genus), cex = 4, direction = "both", segment.size = 0.25) 
p_NMDS_cecum_unweighted_unifrac_sample_type_group_genus

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sex_within_genotype_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_genotype)

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_genotype_within_sex_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_sex)

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_cohort.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_cohort)

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_sample_type)

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_group.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_sample_type_group)

ggsave('./results/environmental_analysis/cecum/NMDS/unweighted_unifrac/both_cohorts/c45_unweighted_unifrac_effect_of_sample_type_genus.png',width=8, height=5, plot = p_NMDS_cecum_unweighted_unifrac_sample_type_group_genus)
```

# 3. Differential Abundance- genera level
## 3.1 Balances
### 3.1.1 split nontransformed data into sections (because the package does log ratios itself)
1. Duodenum
```{r}
#Split by genotype, sex
duodenum_genus_wild_type <-  filter(duodenum_genus %>% select(any_of(duodenum_wild_type_cols_list))) 
dim(duodenum_genus_wild_type)

duodenum_genus_mutant <-  filter(duodenum_genus %>% select(any_of(duodenum_mutant_cols_list))) 
dim(duodenum_genus_mutant)

duodenum_genus_female <-  filter(duodenum_genus %>% select(any_of(duodenum_female_cols_list))) 
dim(duodenum_genus_female)

duodenum_genus_male <-  filter(duodenum_genus %>% select(any_of(duodenum_male_cols_list))) 
dim(duodenum_genus_male)

#split into sample types
#lumen
duodenum_genus_wild_type_lumen <-  filter(duodenum_genus_wild_type %>% select(any_of(duodenum_lumen_cols_list))) 
dim(duodenum_genus_wild_type_lumen)

duodenum_genus_mutant_lumen <-  filter(duodenum_genus_mutant %>% select(any_of(duodenum_lumen_cols_list))) 
dim(duodenum_genus_mutant_lumen)

duodenum_genus_female_lumen <-  filter(duodenum_genus_female %>% select(any_of(duodenum_lumen_cols_list))) 
dim(duodenum_genus_female_lumen)

duodenum_genus_male_lumen <-  filter(duodenum_genus_male %>% select(any_of(duodenum_lumen_cols_list))) 
dim(duodenum_genus_male_lumen)

#mucosa
duodenum_genus_wild_type_mucosa <-  filter(duodenum_genus_wild_type %>% select(any_of(duodenum_mucosa_cols_list))) 
dim(duodenum_genus_wild_type_mucosa)

duodenum_genus_mutant_mucosa <-  filter(duodenum_genus_mutant %>% select(any_of(duodenum_mucosa_cols_list))) 
dim(duodenum_genus_mutant_mucosa)

duodenum_genus_female_mucosa <-  filter(duodenum_genus_female %>% select(any_of(duodenum_mucosa_cols_list))) 
dim(duodenum_genus_female_mucosa)

duodenum_genus_male_mucosa <-  filter(duodenum_genus_male %>% select(any_of(duodenum_mucosa_cols_list))) 
dim(duodenum_genus_male_mucosa)
```
2. ileum
```{r}
#Split by genotype, sex
ileum_genus_wild_type <-  filter(ileum_genus %>% select(any_of(ileum_wild_type_cols_list))) 
dim(ileum_genus_wild_type)

ileum_genus_mutant <-  filter(ileum_genus %>% select(any_of(ileum_mutant_cols_list))) 
dim(ileum_genus_mutant)

ileum_genus_female <-  filter(ileum_genus %>% select(any_of(ileum_female_cols_list))) 
dim(ileum_genus_female)

ileum_genus_male <-  filter(ileum_genus %>% select(any_of(ileum_male_cols_list))) 
dim(ileum_genus_male)

#split into sample types
#lumen
ileum_genus_wild_type_lumen <-  filter(ileum_genus_wild_type %>% select(any_of(ileum_lumen_cols_list))) 
dim(ileum_genus_wild_type_lumen)

ileum_genus_mutant_lumen <-  filter(ileum_genus_mutant %>% select(any_of(ileum_lumen_cols_list))) 
dim(ileum_genus_mutant_lumen)

ileum_genus_female_lumen <-  filter(ileum_genus_female %>% select(any_of(ileum_lumen_cols_list))) 
dim(ileum_genus_female_lumen)

ileum_genus_male_lumen <-  filter(ileum_genus_male %>% select(any_of(ileum_lumen_cols_list))) 
dim(ileum_genus_male_lumen)

#mucosa
ileum_genus_wild_type_mucosa <-  filter(ileum_genus_wild_type %>% select(any_of(ileum_mucosa_cols_list))) 
dim(ileum_genus_wild_type_mucosa)

ileum_genus_mutant_mucosa <-  filter(ileum_genus_mutant %>% select(any_of(ileum_mucosa_cols_list))) 
dim(ileum_genus_mutant_mucosa)

ileum_genus_female_mucosa <-  filter(ileum_genus_female %>% select(any_of(ileum_mucosa_cols_list))) 
dim(ileum_genus_female_mucosa)

ileum_genus_male_mucosa <-  filter(ileum_genus_male %>% select(any_of(ileum_mucosa_cols_list))) 
dim(ileum_genus_male_mucosa)
```

1. cecum
```{r}
#Split by genotype, sex
cecum_genus_wild_type <-  filter(cecum_genus %>% select(any_of(cecum_wild_type_cols_list))) 
dim(cecum_genus_wild_type)

cecum_genus_mutant <-  filter(cecum_genus %>% select(any_of(cecum_mutant_cols_list))) 
dim(cecum_genus_mutant)

cecum_genus_female <-  filter(cecum_genus %>% select(any_of(cecum_female_cols_list))) 
dim(cecum_genus_female)

cecum_genus_male <-  filter(cecum_genus %>% select(any_of(cecum_male_cols_list))) 
dim(cecum_genus_male)

#split into sample types
#lumen
cecum_genus_wild_type_lumen <-  filter(cecum_genus_wild_type %>% select(any_of(cecum_lumen_cols_list))) 
dim(cecum_genus_wild_type_lumen)

cecum_genus_mutant_lumen <-  filter(cecum_genus_mutant %>% select(any_of(cecum_lumen_cols_list))) 
dim(cecum_genus_mutant_lumen)

cecum_genus_female_lumen <-  filter(cecum_genus_female %>% select(any_of(cecum_lumen_cols_list))) 
dim(cecum_genus_female_lumen)

cecum_genus_male_lumen <-  filter(cecum_genus_male %>% select(any_of(cecum_lumen_cols_list))) 
dim(cecum_genus_male_lumen)

#mucosa
cecum_genus_wild_type_mucosa <-  filter(cecum_genus_wild_type %>% select(any_of(cecum_mucosa_cols_list))) 
dim(cecum_genus_wild_type_mucosa)

cecum_genus_mutant_mucosa <-  filter(cecum_genus_mutant %>% select(any_of(cecum_mucosa_cols_list))) 
dim(cecum_genus_mutant_mucosa)

cecum_genus_female_mucosa <-  filter(cecum_genus_female %>% select(any_of(cecum_mucosa_cols_list))) 
dim(cecum_genus_female_mucosa)

cecum_genus_male_mucosa <-  filter(cecum_genus_male %>% select(any_of(cecum_mucosa_cols_list))) 
dim(cecum_genus_male_mucosa)
```
### 3.1.2 Balances for sex differences
#### 3.1.2.1 do balances for sex differences Duodenum
Combined sample types (AUC is too low with separate lumen/mucosa)
1. Wild_Type Differences
```{r}
#Wild_Type Differences
duodenum_genus_wild_type_vector_sex <- factor(duodenum_map[(duodenum_map$Genotype=="Wild_Type" ),]$Sex)

duodenum_genus_wild_type_t <- data.frame(t(duodenum_genus_wild_type))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_duodenum_genus_wild_type<-coda_glmnet(x=duodenum_genus_wild_type_t, y=duodenum_genus_wild_type_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_duodenum_genus_wild_type$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_duodenum_genus_wild_type <- data.frame(cbind(coda_glmnet_duodenum_genus_wild_type$`log-contrast coefficients`, coda_glmnet_duodenum_genus_wild_type$taxa.name))

rownames(coda_balance_coeff_duodenum_genus_wild_type) <- coda_glmnet_duodenum_genus_wild_type$taxa.name

colnames(coda_balance_coeff_duodenum_genus_wild_type) <- c("Coefficient_full", "Genus")

coda_balance_coeff_duodenum_genus_wild_type$Coefficient_full <- as.numeric(coda_balance_coeff_duodenum_genus_wild_type$Coefficient_full)

coda_balance_coeff_duodenum_genus_wild_type$Coefficient <-as.numeric( format(round(coda_balance_coeff_duodenum_genus_wild_type$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_duodenum_genus_wild_type$PosNeg <- with(coda_balance_coeff_duodenum_genus_wild_type, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_duodenum_genus_wild_type$Group <- with(coda_balance_coeff_duodenum_genus_wild_type, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Female_Wild_Type"))

#Save
##write.csv(coda_balance_coeff_duodenum_genus_wild_type,file = "./results/environmental_analysis/duodenum/balances/duodenum_wild_type_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_duodenum_genus_wild_type_predictions <- data.frame(coda_glmnet_duodenum_genus_wild_type$predictions)

colnames(coda_balances_duodenum_genus_wild_type_predictions) <- c("Balance")

coda_balances_duodenum_genus_wild_type_predictions$Sex <- duodenum_genus_wild_type_vector_sex

coda_balances_duodenum_genus_wild_type_predictions$Group <- factor(duodenum_map[(duodenum_map$Genotype=="Wild_Type" ),]$Group)

coda_balances_duodenum_genus_wild_type_predictions$Sample_Type <- factor(duodenum_map[(duodenum_map$Genotype=="Wild_Type" ),]$Sample_Type)

##write.csv(coda_balances_duodenum_genus_wild_type_predictions,file = "./results/environmental_analysis/duodenum/balances/duodenum_wild_type_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_duodenum_coda_sample_balances_wild_type <- ggplot(data = coda_balances_duodenum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_duodenum_coda_sample_balances_wild_type

ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_balances_wild_type.png",width=10, height=4, p_duodenum_coda_sample_balances_wild_type)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_duodenum_genus_wild_type_10p <- coda_balance_coeff_duodenum_genus_wild_type[(coda_balance_coeff_duodenum_genus_wild_type$Coefficient>= 0.099 | coda_balance_coeff_duodenum_genus_wild_type$Coefficient<= -0.099),]


p_duodenum_coda_sample_genera_wild_type <- ggplot(
  data = coda_balance_coeff_duodenum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_wild_type

#ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_genera_coeffieints_wild_type.png",width=10, height=5, p_duodenum_coda_sample_genera_wild_type)
```

Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_duodenum_coda_sample_genera_wild_type_all <- ggplot(
  data = coda_balance_coeff_duodenum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_wild_type_all

#ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_genera_coeffieints_wild_type_all.png",width=10, height=6, p_duodenum_coda_sample_genera_wild_type_all)
```

2. Mutant Differences
```{r}
duodenum_genus_mutant_vector_sex <- factor(duodenum_map[(duodenum_map$Genotype=="Mutant" ),]$Sex)

duodenum_genus_mutant_t <- data.frame(t(duodenum_genus_mutant))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_duodenum_genus_mutant<-coda_glmnet(x=duodenum_genus_mutant_t, y=duodenum_genus_mutant_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 6,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_duodenum_genus_mutant$`apparent AUC`
```
Make data frame with the genera and coefficients 
```{r}
coda_balance_coeff_duodenum_genus_mutant <- data.frame(cbind(coda_glmnet_duodenum_genus_mutant$`log-contrast coefficients`, coda_glmnet_duodenum_genus_mutant$taxa.name))

rownames(coda_balance_coeff_duodenum_genus_mutant) <- coda_glmnet_duodenum_genus_mutant$taxa.name

colnames(coda_balance_coeff_duodenum_genus_mutant) <- c("Coefficient_full", "Genus")

coda_balance_coeff_duodenum_genus_mutant$Coefficient_full <- as.numeric(coda_balance_coeff_duodenum_genus_mutant$Coefficient_full)

coda_balance_coeff_duodenum_genus_mutant$Coefficient <-as.numeric( format(round(coda_balance_coeff_duodenum_genus_mutant$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_duodenum_genus_mutant$PosNeg <- with(coda_balance_coeff_duodenum_genus_mutant, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_duodenum_genus_mutant$Group <- with(coda_balance_coeff_duodenum_genus_mutant, ifelse(PosNeg == "Positive", "Male_Mutant", "Female_Mutant"))

#Save
##write.csv(coda_balance_coeff_duodenum_genus_mutant,file = "./results/environmental_analysis/duodenum/balances/duodenum_mutant_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_duodenum_genus_mutant_predictions <- data.frame(coda_glmnet_duodenum_genus_mutant$predictions)

colnames(coda_balances_duodenum_genus_mutant_predictions) <- c("Balance")

coda_balances_duodenum_genus_mutant_predictions$Sex <- duodenum_genus_mutant_vector_sex

coda_balances_duodenum_genus_mutant_predictions$Group <- factor(duodenum_map[(duodenum_map$Genotype=="Mutant" ),]$Group)

coda_balances_duodenum_genus_mutant_predictions$Sample_Type <- factor(duodenum_map[(duodenum_map$Genotype=="Mutant" ),]$Sample_Type)

##write.csv(coda_balances_duodenum_genus_mutant_predictions,file = "./results/environmental_analysis/duodenum/balances/duodenum_mutant_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients
```{r}
#balance prediction plot
p_duodenum_coda_sample_balances_mutant <- ggplot(data = coda_balances_duodenum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_duodenum_coda_sample_balances_mutant

#ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_balances_mutant.png",width=10, height=4, p_duodenum_coda_sample_balances_mutant)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_duodenum_genus_mutant_10p <- coda_balance_coeff_duodenum_genus_mutant[(coda_balance_coeff_duodenum_genus_mutant$Coefficient>= 0.099 | coda_balance_coeff_duodenum_genus_mutant$Coefficient<= -0.099),]


p_duodenum_coda_sample_genera_mutant <- ggplot(
  data = coda_balance_coeff_duodenum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_mutant

#ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_genera_coeffieints_mutant.png",width=10, height=3, p_duodenum_coda_sample_genera_mutant)
```

Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_duodenum_coda_sample_genera_mutant_all <- ggplot(
  data = coda_balance_coeff_duodenum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_mutant_all

#ggsave("./results/environmental_analysis/duodenum/balances/duodenum_coda_sample_genera_coeffieints_mutant_all.png",width=10, height=9, p_duodenum_coda_sample_genera_mutant_all)
```

#### 3.1.2.2 do balances for sex differences ileum
Combined sample types (AUC is too low with separate lumen/mucosa)
1. Wild_Type Differences
```{r}
#Wild_Type Differences
ileum_genus_wild_type_vector_sex <- factor(ileum_map[(ileum_map$Genotype=="Wild_Type" ),]$Sex)

ileum_genus_wild_type_t <- data.frame(t(ileum_genus_wild_type))

set.seed(123)
coda_glmnet_ileum_genus_wild_type<-coda_glmnet(x=ileum_genus_wild_type_t, y=ileum_genus_wild_type_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 7,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_ileum_genus_wild_type$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_ileum_genus_wild_type <- data.frame(cbind(coda_glmnet_ileum_genus_wild_type$`log-contrast coefficients`, coda_glmnet_ileum_genus_wild_type$taxa.name))

rownames(coda_balance_coeff_ileum_genus_wild_type) <- coda_glmnet_ileum_genus_wild_type$taxa.name

colnames(coda_balance_coeff_ileum_genus_wild_type) <- c("Coefficient_full", "Genus")

coda_balance_coeff_ileum_genus_wild_type$Coefficient_full <- as.numeric(coda_balance_coeff_ileum_genus_wild_type$Coefficient_full)

coda_balance_coeff_ileum_genus_wild_type$Coefficient <-as.numeric( format(round(coda_balance_coeff_ileum_genus_wild_type$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_ileum_genus_wild_type$PosNeg <- with(coda_balance_coeff_ileum_genus_wild_type, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_ileum_genus_wild_type$Group <- with(coda_balance_coeff_ileum_genus_wild_type, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Female_Wild_Type"))

#Save
##write.csv(coda_balance_coeff_ileum_genus_wild_type,file = "./results/environmental_analysis/ileum/balances/ileum_wild_type_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_ileum_genus_wild_type_predictions <- data.frame(coda_glmnet_ileum_genus_wild_type$predictions)

colnames(coda_balances_ileum_genus_wild_type_predictions) <- c("Balance")

coda_balances_ileum_genus_wild_type_predictions$Sex <- ileum_genus_wild_type_vector_sex

coda_balances_ileum_genus_wild_type_predictions$Group <- factor(ileum_map[(ileum_map$Genotype=="Wild_Type" ),]$Group)

coda_balances_ileum_genus_wild_type_predictions$Sample_Type <- factor(ileum_map[(ileum_map$Genotype=="Wild_Type" ),]$Sample_Type)

##write.csv(coda_balances_ileum_genus_wild_type_predictions,file = "./results/environmental_analysis/ileum/balances/ileum_wild_type_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients
```{r}
#balance prediction plot
p_ileum_coda_sample_balances_wild_type <- ggplot(data = coda_balances_ileum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_ileum_coda_sample_balances_wild_type

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_balances_wild_type.png",width=10, height=4, p_ileum_coda_sample_balances_wild_type)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_ileum_genus_wild_type_10p <- coda_balance_coeff_ileum_genus_wild_type[(coda_balance_coeff_ileum_genus_wild_type$Coefficient>= 0.099 | coda_balance_coeff_ileum_genus_wild_type$Coefficient<= -0.099),]


p_ileum_coda_sample_genera_wild_type <- ggplot(
  data = coda_balance_coeff_ileum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_wild_type

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_genera_coeffieints_wild_type.png",width=10, height=5, p_ileum_coda_sample_genera_wild_type)
```

Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_ileum_coda_sample_genera_wild_type_all <- ggplot(
  data = coda_balance_coeff_ileum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_wild_type_all

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_genera_coeffieints_wild_type_all.png",width=10, height=6, p_ileum_coda_sample_genera_wild_type_all)
```

2. Mutant Differences
```{r}
ileum_genus_mutant_vector_sex <- factor(ileum_map[(ileum_map$Genotype=="Mutant" ),]$Sex)

ileum_genus_mutant_t <- data.frame(t(ileum_genus_mutant))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_ileum_genus_mutant<-coda_glmnet(x=ileum_genus_mutant_t, y=ileum_genus_mutant_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 7,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_ileum_genus_mutant$`apparent AUC`
```
Make data frame with the genera and coefficients 
```{r}
coda_balance_coeff_ileum_genus_mutant <- data.frame(cbind(coda_glmnet_ileum_genus_mutant$`log-contrast coefficients`, coda_glmnet_ileum_genus_mutant$taxa.name))

rownames(coda_balance_coeff_ileum_genus_mutant) <- coda_glmnet_ileum_genus_mutant$taxa.name

colnames(coda_balance_coeff_ileum_genus_mutant) <- c("Coefficient_full", "Genus")

coda_balance_coeff_ileum_genus_mutant$Coefficient_full <- as.numeric(coda_balance_coeff_ileum_genus_mutant$Coefficient_full)

coda_balance_coeff_ileum_genus_mutant$Coefficient <-as.numeric( format(round(coda_balance_coeff_ileum_genus_mutant$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_ileum_genus_mutant$PosNeg <- with(coda_balance_coeff_ileum_genus_mutant, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_ileum_genus_mutant$Group <- with(coda_balance_coeff_ileum_genus_mutant, ifelse(PosNeg == "Positive", "Male_Mutant", "Female_Mutant"))

#Save
#write.csv(coda_balance_coeff_ileum_genus_mutant,file = "./results/environmental_analysis/ileum/balances/ileum_mutant_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_ileum_genus_mutant_predictions <- data.frame(coda_glmnet_ileum_genus_mutant$predictions)

colnames(coda_balances_ileum_genus_mutant_predictions) <- c("Balance")

coda_balances_ileum_genus_mutant_predictions$Sex <- ileum_genus_mutant_vector_sex

coda_balances_ileum_genus_mutant_predictions$Group <- factor(ileum_map[(ileum_map$Genotype=="Mutant" ),]$Group)

coda_balances_ileum_genus_mutant_predictions$Sample_Type <- factor(ileum_map[(ileum_map$Genotype=="Mutant" ),]$Sample_Type)

#write.csv(coda_balances_ileum_genus_mutant_predictions,file = "./results/environmental_analysis/ileum/balances/ileum_mutant_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients
```{r}
#balance prediction plot
p_ileum_coda_sample_balances_mutant <- ggplot(data = coda_balances_ileum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_ileum_coda_sample_balances_mutant

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_balances_mutant.png",width=10, height=4, p_ileum_coda_sample_balances_mutant)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_ileum_genus_mutant_10p <- coda_balance_coeff_ileum_genus_mutant[(coda_balance_coeff_ileum_genus_mutant$Coefficient>= 0.099 | coda_balance_coeff_ileum_genus_mutant$Coefficient<= -0.099),]


p_ileum_coda_sample_genera_mutant <- ggplot(
  data = coda_balance_coeff_ileum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_mutant

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_genera_coeffieints_mutant.png",width=10, height=2, p_ileum_coda_sample_genera_mutant)
```
Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_ileum_coda_sample_genera_mutant_all <- ggplot(
  data = coda_balance_coeff_ileum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_mutant_all

ggsave("./results/environmental_analysis/ileum/balances/ileum_coda_sample_genera_coeffieints_mutant_all.png",width=10, height=6, p_ileum_coda_sample_genera_mutant_all)
```

#### 3.1.2.3 do balances for sex differences cecum
Combined sample types (AUC is too low with separate lumen/mucosa)
1. Wild_Type Differences
```{r}
#Wild_Type Differences
cecum_genus_wild_type_vector_sex <- factor(cecum_map[(cecum_map$Genotype=="Wild_Type" ),]$Sex)

cecum_genus_wild_type_t <- data.frame(t(cecum_genus_wild_type))


set.seed(123)
coda_glmnet_cecum_genus_wild_type<-coda_glmnet(x=cecum_genus_wild_type_t, y=cecum_genus_wild_type_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_cecum_genus_wild_type$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_cecum_genus_wild_type <- data.frame(cbind(coda_glmnet_cecum_genus_wild_type$`log-contrast coefficients`, coda_glmnet_cecum_genus_wild_type$taxa.name))

rownames(coda_balance_coeff_cecum_genus_wild_type) <- coda_glmnet_cecum_genus_wild_type$taxa.name

colnames(coda_balance_coeff_cecum_genus_wild_type) <- c("Coefficient_full", "Genus")

coda_balance_coeff_cecum_genus_wild_type$Coefficient_full <- as.numeric(coda_balance_coeff_cecum_genus_wild_type$Coefficient_full)

coda_balance_coeff_cecum_genus_wild_type$Coefficient <-as.numeric( format(round(coda_balance_coeff_cecum_genus_wild_type$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_cecum_genus_wild_type$PosNeg <- with(coda_balance_coeff_cecum_genus_wild_type, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_cecum_genus_wild_type$Group <- with(coda_balance_coeff_cecum_genus_wild_type, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Female_Wild_Type"))

#Save
#write.csv(coda_balance_coeff_cecum_genus_wild_type,file = "./results/environmental_analysis/cecum/balances/cecum_wild_type_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_cecum_genus_wild_type_predictions <- data.frame(coda_glmnet_cecum_genus_wild_type$predictions)

colnames(coda_balances_cecum_genus_wild_type_predictions) <- c("Balance")

coda_balances_cecum_genus_wild_type_predictions$Sex <- cecum_genus_wild_type_vector_sex

coda_balances_cecum_genus_wild_type_predictions$Group <- factor(cecum_map[(cecum_map$Genotype=="Wild_Type" ),]$Group)

coda_balances_cecum_genus_wild_type_predictions$Sample_Type <- factor(cecum_map[(cecum_map$Genotype=="Wild_Type" ),]$Sample_Type)

#write.csv(coda_balances_cecum_genus_wild_type_predictions,file = "./results/environmental_analysis/cecum/balances/cecum_wild_type_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients
```{r}
#balance prediction plot
p_cecum_coda_sample_balances_wild_type <- ggplot(data = coda_balances_cecum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_cecum_coda_sample_balances_wild_type

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_balances_wild_type.png",width=10, height=4, p_cecum_coda_sample_balances_wild_type)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_cecum_genus_wild_type_10p <- coda_balance_coeff_cecum_genus_wild_type[(coda_balance_coeff_cecum_genus_wild_type$Coefficient>= 0.099 | coda_balance_coeff_cecum_genus_wild_type$Coefficient<= -0.099),]


p_cecum_coda_sample_genera_wild_type <- ggplot(
  data = coda_balance_coeff_cecum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_wild_type

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_genera_coeffieints_wild_type.png",width=10, height=5, p_cecum_coda_sample_genera_wild_type)
```
Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_cecum_coda_sample_genera_wild_type_all <- ggplot(
  data = coda_balance_coeff_cecum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_wild_type_all

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_genera_coeffieints_wild_type_all.png",width=10, height=6, p_cecum_coda_sample_genera_wild_type_all)
```

2. Mutant Differences
```{r}
cecum_genus_mutant_vector_sex <- factor(cecum_map[(cecum_map$Genotype=="Mutant" ),]$Sex)

cecum_genus_mutant_t <- data.frame(t(cecum_genus_mutant))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_cecum_genus_mutant<-coda_glmnet(x=cecum_genus_mutant_t, y=cecum_genus_mutant_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 6,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_cecum_genus_mutant$`apparent AUC`
```
Make data frame with the genera and coefficients 
```{r}
coda_balance_coeff_cecum_genus_mutant <- data.frame(cbind(coda_glmnet_cecum_genus_mutant$`log-contrast coefficients`, coda_glmnet_cecum_genus_mutant$taxa.name))

rownames(coda_balance_coeff_cecum_genus_mutant) <- coda_glmnet_cecum_genus_mutant$taxa.name

colnames(coda_balance_coeff_cecum_genus_mutant) <- c("Coefficient_full", "Genus")

coda_balance_coeff_cecum_genus_mutant$Coefficient_full <- as.numeric(coda_balance_coeff_cecum_genus_mutant$Coefficient_full)

coda_balance_coeff_cecum_genus_mutant$Coefficient <-as.numeric( format(round(coda_balance_coeff_cecum_genus_mutant$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_cecum_genus_mutant$PosNeg <- with(coda_balance_coeff_cecum_genus_mutant, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_cecum_genus_mutant$Group <- with(coda_balance_coeff_cecum_genus_mutant, ifelse(PosNeg == "Positive", "Male_Mutant", "Female_Mutant"))

#Save
#write.csv(coda_balance_coeff_cecum_genus_mutant,file = "./results/environmental_analysis/cecum/balances/cecum_mutant_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_cecum_genus_mutant_predictions <- data.frame(coda_glmnet_cecum_genus_mutant$predictions)

colnames(coda_balances_cecum_genus_mutant_predictions) <- c("Balance")

coda_balances_cecum_genus_mutant_predictions$Sex <- cecum_genus_mutant_vector_sex

coda_balances_cecum_genus_mutant_predictions$Group <- factor(cecum_map[(cecum_map$Genotype=="Mutant" ),]$Group)

coda_balances_cecum_genus_mutant_predictions$Sample_Type <- factor(cecum_map[(cecum_map$Genotype=="Mutant" ),]$Sample_Type)

#write.csv(coda_balances_cecum_genus_mutant_predictions,file = "./results/environmental_analysis/cecum/balances/cecum_mutant_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients
```{r}
#balance prediction plot
p_cecum_coda_sample_balances_mutant <- ggplot(data = coda_balances_cecum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_cecum_coda_sample_balances_mutant

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_balances_mutant.png",width=10, height=4, p_cecum_coda_sample_balances_mutant)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_cecum_genus_mutant_10p <- coda_balance_coeff_cecum_genus_mutant[(coda_balance_coeff_cecum_genus_mutant$Coefficient>= 0.099 | coda_balance_coeff_cecum_genus_mutant$Coefficient<= -0.099),]


p_cecum_coda_sample_genera_mutant <- ggplot(
  data = coda_balance_coeff_cecum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_mutant

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_genera_coeffieints_mutant.png",width=10, height=4, p_cecum_coda_sample_genera_mutant)
```
Graph Genus Coefficients- all genus
```{r}
#Genus coefficient plot
p_cecum_coda_sample_genera_mutant_all <- ggplot(
  data = coda_balance_coeff_cecum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_mutant_all

ggsave("./results/environmental_analysis/cecum/balances/cecum_coda_sample_genera_coeffieints_mutant_all.png",width=10, height=8, p_cecum_coda_sample_genera_mutant_all)
```

Make AUC table for all environments
```{r}
coda_glmnet_Sex_differences_AUC <- c(coda_glmnet_duodenum_genus_wild_type$`apparent AUC`, coda_glmnet_duodenum_genus_mutant$`apparent AUC`, coda_glmnet_ileum_genus_wild_type$`apparent AUC`, 
                                     coda_glmnet_ileum_genus_mutant$`apparent AUC`,coda_glmnet_cecum_genus_wild_type$`apparent AUC`, coda_glmnet_cecum_genus_mutant$`apparent AUC`) 
coda_glmnet_Sex_differences_AUC

coda_glmnet_Sex_differences_AUC_names <- c("duodenum_wild_type", "duodenum_mutant", "ileum_wild_type", "ileum_mutant", "cecum_wild_type", "cecum_mutant")

coda_glmnet_Sex_differences_AUC_table <- cbind(coda_glmnet_Sex_differences_AUC_names, coda_glmnet_Sex_differences_AUC)

#write.csv(coda_glmnet_Sex_differences_AUC_table, file = "./results/environmental_analysis/duodenum/balances/coda_glmnet_Sex_differences_AUC_table.csv")
                                     
```

### 3.1.3 Balances for genotype differences
#### 3.1.3.1 Duodenum Balances for genotype differences
Combined sample types (AUC is too low with separate lumen/mucosa)
1. female Differences
```{r}
#female Differences
duodenum_genus_female_vector_sex <- factor(duodenum_map[(duodenum_map$Sex=="Female" ),]$Genotype)

duodenum_genus_female_t <- data.frame(t(duodenum_genus_female))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_duodenum_genus_female<-coda_glmnet(x=duodenum_genus_female_t, y=duodenum_genus_female_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 7,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_duodenum_genus_female$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_duodenum_genus_female <- data.frame(cbind(coda_glmnet_duodenum_genus_female$`log-contrast coefficients`, coda_glmnet_duodenum_genus_female$taxa.name))

rownames(coda_balance_coeff_duodenum_genus_female) <- coda_glmnet_duodenum_genus_female$taxa.name

colnames(coda_balance_coeff_duodenum_genus_female) <- c("Coefficient_full", "Genus")

coda_balance_coeff_duodenum_genus_female$Coefficient_full <- as.numeric(coda_balance_coeff_duodenum_genus_female$Coefficient_full)

coda_balance_coeff_duodenum_genus_female$Coefficient <-as.numeric( format(round(coda_balance_coeff_duodenum_genus_female$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_duodenum_genus_female$PosNeg <- with(coda_balance_coeff_duodenum_genus_female, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_duodenum_genus_female$Group <- with(coda_balance_coeff_duodenum_genus_female, ifelse(PosNeg == "Positive", "Female_Wild_Type", "Female_Mutant"))

#Save
#write.csv(coda_balance_coeff_duodenum_genus_female,file = "./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_female_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_duodenum_genus_female_predictions <- data.frame(coda_glmnet_duodenum_genus_female$predictions)

colnames(coda_balances_duodenum_genus_female_predictions) <- c("Balance")

coda_balances_duodenum_genus_female_predictions$Sex <- duodenum_genus_female_vector_sex

coda_balances_duodenum_genus_female_predictions$Group <- factor(duodenum_map[(duodenum_map$Sex=="Female" ),]$Group)

coda_balances_duodenum_genus_female_predictions$Sample_Type <- factor(duodenum_map[(duodenum_map$Sex=="Female" ),]$Sample_Type)

#write.csv(coda_balances_duodenum_genus_female_predictions,file = "./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_female_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_duodenum_coda_sample_balances_female <- ggplot(data = coda_balances_duodenum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_duodenum_coda_sample_balances_female

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_balances_female.png",width=10, height=4, p_duodenum_coda_sample_balances_female)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_duodenum_genus_female_10p <- coda_balance_coeff_duodenum_genus_female[(coda_balance_coeff_duodenum_genus_female$Coefficient>= 0.099 | coda_balance_coeff_duodenum_genus_female$Coefficient<= -0.099),]


p_duodenum_coda_sample_genera_female <- ggplot(
  data = coda_balance_coeff_duodenum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_female

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_genera_coeffieints_female.png",width=10, height=5, p_duodenum_coda_sample_genera_female)

#Graph Genus Coefficients- all genus
p_duodenum_coda_sample_genera_female_all <- ggplot(
  data = coda_balance_coeff_duodenum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_female_all

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_genera_coeffieints_female_all.png",width=10, height=6, p_duodenum_coda_sample_genera_female_all)
```

2. male Differences
```{r}
#male Differences
duodenum_genus_male_vector_sex <- factor(duodenum_map[(duodenum_map$Sex=="Male" ),]$Genotype)

duodenum_genus_male_t <- data.frame(t(duodenum_genus_male))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_duodenum_genus_male<-coda_glmnet(x=duodenum_genus_male_t, y=duodenum_genus_male_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_duodenum_genus_male$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_duodenum_genus_male <- data.frame(cbind(coda_glmnet_duodenum_genus_male$`log-contrast coefficients`, coda_glmnet_duodenum_genus_male$taxa.name))

rownames(coda_balance_coeff_duodenum_genus_male) <- coda_glmnet_duodenum_genus_male$taxa.name

colnames(coda_balance_coeff_duodenum_genus_male) <- c("Coefficient_full", "Genus")

coda_balance_coeff_duodenum_genus_male$Coefficient_full <- as.numeric(coda_balance_coeff_duodenum_genus_male$Coefficient_full)

coda_balance_coeff_duodenum_genus_male$Coefficient <-as.numeric( format(round(coda_balance_coeff_duodenum_genus_male$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_duodenum_genus_male$PosNeg <- with(coda_balance_coeff_duodenum_genus_male, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_duodenum_genus_male$Group <- with(coda_balance_coeff_duodenum_genus_male, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Male_Mutant"))

#Save
#write.csv(coda_balance_coeff_duodenum_genus_male,file = "./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_male_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_duodenum_genus_male_predictions <- data.frame(coda_glmnet_duodenum_genus_male$predictions)

colnames(coda_balances_duodenum_genus_male_predictions) <- c("Balance")

coda_balances_duodenum_genus_male_predictions$Sex <- duodenum_genus_male_vector_sex

coda_balances_duodenum_genus_male_predictions$Group <- factor(duodenum_map[(duodenum_map$Sex=="Male" ),]$Group)

coda_balances_duodenum_genus_male_predictions$Sample_Type <- factor(duodenum_map[(duodenum_map$Sex=="Male" ),]$Sample_Type)

#write.csv(coda_balances_duodenum_genus_male_predictions,file = "./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_male_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_duodenum_coda_sample_balances_male <- ggplot(data = coda_balances_duodenum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_duodenum_coda_sample_balances_male

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_balances_male.png",width=10, height=4, p_duodenum_coda_sample_balances_male)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_duodenum_genus_male_10p <- coda_balance_coeff_duodenum_genus_male[(coda_balance_coeff_duodenum_genus_male$Coefficient>= 0.099 | coda_balance_coeff_duodenum_genus_male$Coefficient<= -0.099),]


p_duodenum_coda_sample_genera_male <- ggplot(
  data = coda_balance_coeff_duodenum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_male

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_genera_coeffieints_male.png",width=10, height=5, p_duodenum_coda_sample_genera_male)

#Graph Genus Coefficients- all genus
p_duodenum_coda_sample_genera_male_all <- ggplot(
  data = coda_balance_coeff_duodenum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_duodenum_coda_sample_genera_male_all

#ggsave("./results/environmental_analysis/duodenum/balances/genotype_differences/duodenum_coda_sample_genera_coeffieints_male_all.png",width=10, height=4, p_duodenum_coda_sample_genera_male_all)
```

#### 3.1.3.2 Ileum Balances for genotype differences
Combined sample types (AUC is too low with separate lumen/mucosa)
1. female Differences
```{r}
#female Differences
ileum_genus_female_vector_sex <- factor(ileum_map[(ileum_map$Sex=="Female" ),]$Genotype)

ileum_genus_female_t <- data.frame(t(ileum_genus_female))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_ileum_genus_female<-coda_glmnet(x=ileum_genus_female_t, y=ileum_genus_female_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 7,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_ileum_genus_female$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_ileum_genus_female <- data.frame(cbind(coda_glmnet_ileum_genus_female$`log-contrast coefficients`, coda_glmnet_ileum_genus_female$taxa.name))

rownames(coda_balance_coeff_ileum_genus_female) <- coda_glmnet_ileum_genus_female$taxa.name

colnames(coda_balance_coeff_ileum_genus_female) <- c("Coefficient_full", "Genus")

coda_balance_coeff_ileum_genus_female$Coefficient_full <- as.numeric(coda_balance_coeff_ileum_genus_female$Coefficient_full)

coda_balance_coeff_ileum_genus_female$Coefficient <-as.numeric( format(round(coda_balance_coeff_ileum_genus_female$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_ileum_genus_female$PosNeg <- with(coda_balance_coeff_ileum_genus_female, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_ileum_genus_female$Group <- with(coda_balance_coeff_ileum_genus_female, ifelse(PosNeg == "Positive", "Female_Wild_Type", "Female_Mutant"))

#Save
#write.csv(coda_balance_coeff_ileum_genus_female,file = "./results/environmental_analysis/ileum/balances/genotype_differences/ileum_female_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_ileum_genus_female_predictions <- data.frame(coda_glmnet_ileum_genus_female$predictions)

colnames(coda_balances_ileum_genus_female_predictions) <- c("Balance")

coda_balances_ileum_genus_female_predictions$Sex <- ileum_genus_female_vector_sex

coda_balances_ileum_genus_female_predictions$Group <- factor(ileum_map[(ileum_map$Sex=="Female" ),]$Group)

coda_balances_ileum_genus_female_predictions$Sample_Type <- factor(ileum_map[(ileum_map$Sex=="Female" ),]$Sample_Type)

#write.csv(coda_balances_ileum_genus_female_predictions,file = "./results/environmental_analysis/ileum/balances/genotype_differences/ileum_female_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_ileum_coda_sample_balances_female <- ggplot(data = coda_balances_ileum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_ileum_coda_sample_balances_female

ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_balances_female.png",width=10, height=4, p_ileum_coda_sample_balances_female)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_ileum_genus_female_10p <- coda_balance_coeff_ileum_genus_female[(coda_balance_coeff_ileum_genus_female$Coefficient>= 0.099 | coda_balance_coeff_ileum_genus_female$Coefficient<= -0.099),]


p_ileum_coda_sample_genera_female <- ggplot(
  data = coda_balance_coeff_ileum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_female

#ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_genera_coeffieints_female.png",width=10, height=5, p_ileum_coda_sample_genera_female)

#Graph Genus Coefficients- all genus
p_ileum_coda_sample_genera_female_all <- ggplot(
  data = coda_balance_coeff_ileum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_female_all

#ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_genera_coeffieints_female_all.png",width=10, height=7, p_ileum_coda_sample_genera_female_all)
```

2. male Differences
```{r}
#male Differences
ileum_genus_male_vector_sex <- factor(ileum_map[(ileum_map$Sex=="Male" ),]$Genotype)

ileum_genus_male_t <- data.frame(t(ileum_genus_male))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_ileum_genus_male<-coda_glmnet(x=ileum_genus_male_t, y=ileum_genus_male_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_ileum_genus_male$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_ileum_genus_male <- data.frame(cbind(coda_glmnet_ileum_genus_male$`log-contrast coefficients`, coda_glmnet_ileum_genus_male$taxa.name))

rownames(coda_balance_coeff_ileum_genus_male) <- coda_glmnet_ileum_genus_male$taxa.name

colnames(coda_balance_coeff_ileum_genus_male) <- c("Coefficient_full", "Genus")

coda_balance_coeff_ileum_genus_male$Coefficient_full <- as.numeric(coda_balance_coeff_ileum_genus_male$Coefficient_full)

coda_balance_coeff_ileum_genus_male$Coefficient <-as.numeric( format(round(coda_balance_coeff_ileum_genus_male$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_ileum_genus_male$PosNeg <- with(coda_balance_coeff_ileum_genus_male, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_ileum_genus_male$Group <- with(coda_balance_coeff_ileum_genus_male, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Male_Mutant"))

#Save
#write.csv(coda_balance_coeff_ileum_genus_male,file = "./results/environmental_analysis/ileum/balances/genotype_differences/ileum_male_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_ileum_genus_male_predictions <- data.frame(coda_glmnet_ileum_genus_male$predictions)

colnames(coda_balances_ileum_genus_male_predictions) <- c("Balance")

coda_balances_ileum_genus_male_predictions$Sex <- ileum_genus_male_vector_sex

coda_balances_ileum_genus_male_predictions$Group <- factor(ileum_map[(ileum_map$Sex=="Male" ),]$Group)

coda_balances_ileum_genus_male_predictions$Sample_Type <- factor(ileum_map[(ileum_map$Sex=="Male" ),]$Sample_Type)

#write.csv(coda_balances_ileum_genus_male_predictions,file = "./results/environmental_analysis/ileum/balances/genotype_differences/ileum_male_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_ileum_coda_sample_balances_male <- ggplot(data = coda_balances_ileum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_ileum_coda_sample_balances_male

#ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_balances_male.png",width=10, height=4, p_ileum_coda_sample_balances_male)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_ileum_genus_male_10p <- coda_balance_coeff_ileum_genus_male[(coda_balance_coeff_ileum_genus_male$Coefficient>= 0.099 | coda_balance_coeff_ileum_genus_male$Coefficient<= -0.099),]


p_ileum_coda_sample_genera_male <- ggplot(
  data = coda_balance_coeff_ileum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_male

#ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_genera_coeffieints_male.png",width=10, height=3, p_ileum_coda_sample_genera_male)

#Graph Genus Coefficients- all genus
p_ileum_coda_sample_genera_male_all <- ggplot(
  data = coda_balance_coeff_ileum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_ileum_coda_sample_genera_male_all

#ggsave("./results/environmental_analysis/ileum/balances/genotype_differences/ileum_coda_sample_genera_coeffieints_male_all.png",width=10, height=3, p_ileum_coda_sample_genera_male_all)
```

#### 3.1.3.3 Cecum Balances for genotype differences
Combined sample types (AUC is too low with separate lumen/mucosa)
1. female Differences
```{r}
#female Differences
cecum_genus_female_vector_sex <- factor(cecum_map[(cecum_map$Sex=="Female" ),]$Genotype)

cecum_genus_female_t <- data.frame(t(cecum_genus_female))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_cecum_genus_female<-coda_glmnet(x=cecum_genus_female_t, y=cecum_genus_female_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_cecum_genus_female$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_cecum_genus_female <- data.frame(cbind(coda_glmnet_cecum_genus_female$`log-contrast coefficients`, coda_glmnet_cecum_genus_female$taxa.name))

rownames(coda_balance_coeff_cecum_genus_female) <- coda_glmnet_cecum_genus_female$taxa.name

colnames(coda_balance_coeff_cecum_genus_female) <- c("Coefficient_full", "Genus")

coda_balance_coeff_cecum_genus_female$Coefficient_full <- as.numeric(coda_balance_coeff_cecum_genus_female$Coefficient_full)

coda_balance_coeff_cecum_genus_female$Coefficient <-as.numeric( format(round(coda_balance_coeff_cecum_genus_female$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_cecum_genus_female$PosNeg <- with(coda_balance_coeff_cecum_genus_female, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_cecum_genus_female$Group <- with(coda_balance_coeff_cecum_genus_female, ifelse(PosNeg == "Positive", "Female_Wild_Type", "Female_Mutant"))

#Save
#write.csv(coda_balance_coeff_cecum_genus_female,file = "./results/environmental_analysis/cecum/balances/genotype_differences/cecum_female_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_cecum_genus_female_predictions <- data.frame(coda_glmnet_cecum_genus_female$predictions)

colnames(coda_balances_cecum_genus_female_predictions) <- c("Balance")

coda_balances_cecum_genus_female_predictions$Sex <- cecum_genus_female_vector_sex

coda_balances_cecum_genus_female_predictions$Group <- factor(cecum_map[(cecum_map$Sex=="Female" ),]$Group)

coda_balances_cecum_genus_female_predictions$Sample_Type <- factor(cecum_map[(cecum_map$Sex=="Female" ),]$Sample_Type)

#write.csv(coda_balances_cecum_genus_female_predictions,file = "./results/environmental_analysis/cecum/balances/genotype_differences/cecum_female_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_cecum_coda_sample_balances_female <- ggplot(data = coda_balances_cecum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_cecum_coda_sample_balances_female

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_balances_female.png",width=10, height=4, p_cecum_coda_sample_balances_female)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_cecum_genus_female_10p <- coda_balance_coeff_cecum_genus_female[(coda_balance_coeff_cecum_genus_female$Coefficient>= 0.099 | coda_balance_coeff_cecum_genus_female$Coefficient<= -0.099),]


p_cecum_coda_sample_genera_female <- ggplot(
  data = coda_balance_coeff_cecum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_female

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_genera_coeffieints_female.png",width=10, height=3, p_cecum_coda_sample_genera_female)

#Graph Genus Coefficients- all genus
p_cecum_coda_sample_genera_female_all <- ggplot(
  data = coda_balance_coeff_cecum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_female_all

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_genera_coeffieints_female_all.png",width=10, height=4, p_cecum_coda_sample_genera_female_all)
```

2. male Differences
```{r}
#male Differences
cecum_genus_male_vector_sex <- factor(cecum_map[(cecum_map$Sex=="Male" ),]$Genotype)

cecum_genus_male_t <- data.frame(t(cecum_genus_male))

#only take taxa over 10% contribution to variance
set.seed(123)
coda_glmnet_cecum_genus_male<-coda_glmnet(x=cecum_genus_male_t, y=cecum_genus_male_vector_sex,
lambda = "lambda.1se",
nvar = NULL,
alpha = 0.9,
nfolds = 8,
showPlots = TRUE,
coef_threshold = 0)

#Accuracy measure
coda_glmnet_cecum_genus_male$`apparent AUC`
```
Make data frame with the genera and coefficients
```{r}
coda_balance_coeff_cecum_genus_male <- data.frame(cbind(coda_glmnet_cecum_genus_male$`log-contrast coefficients`, coda_glmnet_cecum_genus_male$taxa.name))

rownames(coda_balance_coeff_cecum_genus_male) <- coda_glmnet_cecum_genus_male$taxa.name

colnames(coda_balance_coeff_cecum_genus_male) <- c("Coefficient_full", "Genus")

coda_balance_coeff_cecum_genus_male$Coefficient_full <- as.numeric(coda_balance_coeff_cecum_genus_male$Coefficient_full)

coda_balance_coeff_cecum_genus_male$Coefficient <-as.numeric( format(round(coda_balance_coeff_cecum_genus_male$Coefficient_full, 2), nsmall = 2))

coda_balance_coeff_cecum_genus_male$PosNeg <- with(coda_balance_coeff_cecum_genus_male, ifelse(Coefficient > 0, "Positive", "Negative"))

coda_balance_coeff_cecum_genus_male$Group <- with(coda_balance_coeff_cecum_genus_male, ifelse(PosNeg == "Positive", "Male_Wild_Type", "Male_Mutant"))

#Save
#write.csv(coda_balance_coeff_cecum_genus_male,file = "./results/environmental_analysis/cecum/balances/genotype_differences/cecum_male_balance_genus_and_coefficients.csv" )

#Make data frame with the samples, balance predictions, and variables
coda_balances_cecum_genus_male_predictions <- data.frame(coda_glmnet_cecum_genus_male$predictions)

colnames(coda_balances_cecum_genus_male_predictions) <- c("Balance")

coda_balances_cecum_genus_male_predictions$Sex <- cecum_genus_male_vector_sex

coda_balances_cecum_genus_male_predictions$Group <- factor(cecum_map[(cecum_map$Sex=="Male" ),]$Group)

coda_balances_cecum_genus_male_predictions$Sample_Type <- factor(cecum_map[(cecum_map$Sex=="Male" ),]$Sample_Type)

#write.csv(coda_balances_cecum_genus_male_predictions,file = "./results/environmental_analysis/cecum/balances/genotype_differences/cecum_male_balance_predictions.csv" )
```

Graph Sample Balances and Genus Coefficients (coeff > 0.099)
```{r}
#balance prediction plot
p_cecum_coda_sample_balances_male <- ggplot(data = coda_balances_cecum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Microbial Balances") +
  #facet_grid(~Sample_Type) +
    theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  #theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
  #theme(axis.text.y = element_text(angle = 90, hjust = 0.5)) +
  coord_flip() 
p_cecum_coda_sample_balances_male

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_balances_male.png",width=10, height=4, p_cecum_coda_sample_balances_male)

#Genus coefficient plot
#plot only the genera that are at least 10% of the contribution to the ballance
coda_balance_coeff_cecum_genus_male_10p <- coda_balance_coeff_cecum_genus_male[(coda_balance_coeff_cecum_genus_male$Coefficient>= 0.099 | coda_balance_coeff_cecum_genus_male$Coefficient<= -0.099),]


p_cecum_coda_sample_genera_male <- ggplot(
  data = coda_balance_coeff_cecum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_male

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_genera_coeffieints_male.png",width=10, height=3, p_cecum_coda_sample_genera_male)

#Graph Genus Coefficients- all genus
p_cecum_coda_sample_genera_male_all <- ggplot(
  data = coda_balance_coeff_cecum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_classic() +
  labs(title = "Genera in Selected Balance") +
  xlab(" ") + ylab("Coefficient") +
  theme(text = element_text(size = 20))  +
  theme(plot.title = element_text(size=16)) +
  coord_flip() 
p_cecum_coda_sample_genera_male_all

#ggsave("./results/environmental_analysis/cecum/balances/genotype_differences/cecum_coda_sample_genera_coeffieints_male_all.png",width=10, height=3, p_cecum_coda_sample_genera_male_all)
```

### 3.1.4 Make AUC table for all environments 
```{r}
coda_glmnet_genotype_differences_AUC <- c(coda_glmnet_duodenum_genus_female$`apparent AUC`, coda_glmnet_duodenum_genus_male$`apparent AUC`, coda_glmnet_ileum_genus_female$`apparent AUC`, 
                                     coda_glmnet_ileum_genus_male$`apparent AUC`,coda_glmnet_cecum_genus_female$`apparent AUC`, coda_glmnet_cecum_genus_male$`apparent AUC`) 
coda_glmnet_genotype_differences_AUC

coda_glmnet_genotype_differences_AUC_names <- c("duodenum_female", "duodenum_male", "ileum_female", "ileum_male", "cecum_female", "cecum_male")

coda_glmnet_genotype_differences_AUC_table <- cbind(coda_glmnet_genotype_differences_AUC_names, coda_glmnet_genotype_differences_AUC)

#write.csv(coda_glmnet_genotype_differences_AUC_table, file = "./results/environmental_analysis/duodenum/balances/genotype_differences/coda_glmnet_genotype_differences_AUC_table.csv")
                                     
```
### 3.1.5 Heatmap of just the selected different genera
All Envs
```{r}
#coda_balance_coeff_duodenum_genus_sex_and_genotype_diff_merge_10p$Genus
#Select genera different in WT and in genotype differences
genera_interesting_balances <- list("Hydrogenibacillus", "Pseudomonas", "Lachnoclostridium", "Escherichia.Shigella", "Eggerthellaceae", "Anaerotruncus", "Incertae_Sedis" , "Intestinimonas", "Peptococcaceae1", "Peptococcaceae"  )
intestine_genus_tx_group_by_group_interesting_balances_stack <- intestine_genus_tx_group_by_group_sample_stack[(intestine_genus_tx_group_by_group_sample_stack$Genus %in% genera_interesting_balances),]
```
```{r}
p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance <- ggplot(intestine_genus_tx_group_by_group_interesting_balances_stack, aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  #scale_fill_distiller(palette = "YlGnBu") +
  scale_fill_viridis_c(option = "D", begin = 0.1) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Sample_Type_Env, nrow = 1) +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_sample_type_interesting_balance.png',width=27, height=11, plot = p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance)
```

WT Differences by section
```{r}
#genera_interesting_balances_duo <- list("Hydrogenibacillus", "Pseudomonas", "Lachnoclostridium", "Escherichia.Shigella", "Eggerthellaceae", "Anaerotruncus", "Incertae_Sedis" , "Intestinimonas", "Peptococcaceae1", "Peptococcaceae"  )
intestine_genus_tx_group_by_group_interesting_balances_stack <- intestine_genus_tx_group_by_group_sample_stack[(intestine_genus_tx_group_by_group_sample_stack$Genus %in% genera_interesting_balances),]


#Duodenum WT differences (10p)
intestine_genus_tx_group_by_group_duodenum_interesting_balances_stack <- intestine_genus_tx_group_by_group_sample_stack[(intestine_genus_tx_group_by_group_sample_stack$Genus %in% coda_balance_coeff_duodenum_genus_wild_type_10p$Genus & intestine_genus_tx_group_by_group_sample_stack$Sample_Type_Env %in% list("Duodenum_Lumen", "Duodenum_Mucosa")),]

p_heatmap_genus_sample_type_env_facet_duodenum_interesting_balance <- ggplot(intestine_genus_tx_group_by_group_duodenum_interesting_balances_stack , aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  #scale_fill_distiller(palette = "YlGnBu") +
  scale_fill_viridis_c(option = "D", begin = 0.1) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Sample_Type_Env, nrow = 1) +
  theme(axis.text.x = element_text(angle = 50, vjust = 1, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_duodenum_interesting_balance.png',width=14, height=9, plot = p_heatmap_genus_sample_type_env_facet_duodenum_interesting_balance)

#ileum WT differences (10p)
intestine_genus_tx_group_by_group_ileum_interesting_balances_stack <- intestine_genus_tx_group_by_group_sample_stack[(intestine_genus_tx_group_by_group_sample_stack$Genus %in% coda_balance_coeff_ileum_genus_wild_type_10p$Genus & intestine_genus_tx_group_by_group_sample_stack$Sample_Type_Env %in% list("Ileum_Lumen", "Ileum_Mucosa")),]

p_heatmap_genus_sample_type_env_facet_ileum_interesting_balance <- ggplot(intestine_genus_tx_group_by_group_ileum_interesting_balances_stack , aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  #scale_fill_distiller(palette = "YlGnBu") +
  scale_fill_viridis_c(option = "D", begin = 0.1) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Sample_Type_Env, nrow = 1) +
  theme(axis.text.x = element_text(angle = 50, vjust = 1, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_ileum_interesting_balance.png',width=14, height=7, plot = p_heatmap_genus_sample_type_env_facet_ileum_interesting_balance)

#ileum WT differences (10p)
intestine_genus_tx_group_by_group_cecum_interesting_balances_stack <- intestine_genus_tx_group_by_group_sample_stack[(intestine_genus_tx_group_by_group_sample_stack$Genus %in% coda_balance_coeff_cecum_genus_wild_type_10p$Genus & intestine_genus_tx_group_by_group_sample_stack$Sample_Type_Env %in% list("Cecum_Lumen", "Cecum_Mucosa")),]

p_heatmap_genus_sample_type_env_facet_cecum_interesting_balance <- ggplot(intestine_genus_tx_group_by_group_cecum_interesting_balances_stack , aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  #scale_fill_distiller(palette = "YlGnBu") +
  scale_fill_viridis_c(option = "D", begin = 0.12) +
  theme_classic() +
   theme(text = element_text(size = 30))  +  
  facet_wrap(~Sample_Type_Env, nrow = 1) +
  theme(axis.text.x = element_text(angle = 50, vjust = 1, hjust=1)) +
  theme(axis.text = element_text(face="bold"))
p_heatmap_genus_sample_type_env_facet_sample_type_interesting_balance

ggsave('./results/environmental_analysis/all_envs/heatmap/genus/heatmap_genus_sample_types_facet_cecum_interesting_balance.png',width=14, height=8, plot = p_heatmap_genus_sample_type_env_facet_cecum_interesting_balance)
```

## 3.2 genera with glucuronidase /BSH/ 
Select genera
```{r}
#row.names(intestine_genus_tx)
glucuronidase_genera <- c("Alistipes" , "Bacteroides" , "Escherichia-Shigella", "Anaerotruncus" , "Blautia"  , "Enterococcus" , "Faecalibaculum" , "Lactobacillus", "Marvinbryantia"  , "Roseburia" , "Streptococcus" , "Bifidobacterium" )

BSH_genera <- c("Bifidobacterium"  , "Listeria" , "Bacillus" , "Staphylococcus" , "Lactobacillus", "Enterococcus" , "Streptococcus", "Blautia", "Lachnoclostridium" , "Marvinbryantia"  , "Roseburia" , "Anaerotruncus", "Faecalibaculum" ,"Methylobacterium-Methylorubrum" , "Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", "Burkholderia-Caballeronia-Paraburkholderia" , "Escherichia-Shigella", "Acinetobacter", "Pseudomonas"  )

glucuronidase_BSH_genera <- unique(c(glucuronidase_genera, BSH_genera))
#this is only 21 
length(glucuronidase_BSH_genera)
```

# 4. Sourcetracker graphs
```{r}
p_fecal_sourcetracker_means_bar = ggplot(fecal_sourcetracker_means) +
    scale_fill_manual(values = section_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 0.5, aes(x = Group, fill = Source, y = Mean)) + 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    scale_y_continuous(expand = c(0,0)) + 
    #ggtitle(label = "") +
    #facet_wrap(~Cohort) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Source Proportion", fill = "Source") 
p_fecal_sourcetracker_means_bar

p_fecal_sourcetracker_means_bar_flip <- p_fecal_sourcetracker_means_bar + 
    coord_flip() 
p_fecal_sourcetracker_means_bar_flip

ggsave("./results/environmental_analysis/all_envs/sourcetracker/fecal_sourcetracker_means_bar.png",p_fecal_sourcetracker_means_bar, width = 7, height = 7)

ggsave("./results/environmental_analysis/all_envs/sourcetracker/fecal_sourcetracker_means_bar_flip.png",p_fecal_sourcetracker_means_bar_flip, width = 12, height = 6)

```

With CI bars
```{r}
p_fecal_sourcetracker_means_bar_facet_group = ggplot(fecal_sourcetracker_means) +
    scale_fill_manual(values = section_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Source, fill = Source, y = Mean)) + 
  geom_errorbar(aes(x = Source, ymin=Low_CI, ymax=High_CI), width=.2) +
                 #position=position_dodge(.9)) 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    #ggtitle(label = "") +
    facet_wrap(~Group) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Source Proportion", fill = "Source")  + 
    coord_flip() 
p_fecal_sourcetracker_means_bar_facet_group

ggsave("./results/environmental_analysis/all_envs/sourcetracker/p_fecal_sourcetracker_means_confidence_intervals_facet_group.png",p_fecal_sourcetracker_means_bar_facet_group, width = 13, height = 7)

p_fecal_sourcetracker_means_bar_facet_source = ggplot(fecal_sourcetracker_means) +
    scale_fill_manual(values = group_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Group, fill = Group, y = Mean)) + 
  geom_errorbar(aes(x = Group, ymin=Low_CI, ymax=High_CI), width=.2) +
                 #position=position_dodge(.9)) 
    theme_classic() +
    theme(text = element_text(size = 17))  +
    #ggtitle(label = "") +
    facet_wrap(~Source) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Source Proportion", fill = "Source")  + 
    coord_flip() 
p_fecal_sourcetracker_means_bar_facet_source

ggsave("./results/environmental_analysis/all_envs/sourcetracker/p_fecal_sourcetracker_means_confidence_intervals_facet_source.png",p_fecal_sourcetracker_means_bar_facet_source, width = 13, height = 7)
```




# 5. Figures
## 5.1 Supplemental Figures 1/2/3/4/5
### 5.1.1 Supplemental Figure 1: Source tracker groups
```{r}
p_fecal_sourcetracker_means_bar_facet_group_fig = ggplot(fecal_sourcetracker_means) +
    scale_fill_manual(values = section_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Source, fill = Source, y = Mean)) + 
  geom_errorbar(aes(x = Source, ymin=Low_CI, ymax=High_CI), width=.2) +
    theme_classic() +
    theme(text = element_text(size = 17))  +
    #ggtitle(label = "") +
    facet_wrap(~Group) +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Source Proportion", fill = "Source")  + 
    coord_flip() 
p_fecal_sourcetracker_means_bar_facet_group_fig

ggsave("./results/environmental_analysis/figures/supplemental_figure_1_sourcetracker_groups.svg",p_fecal_sourcetracker_means_bar_facet_group_fig, width = 13, height = 7)
```

### 5.1.2 Supplemental figure 2: Rarefaction curves

Get average counts
```{r}
duodenum_SVs_for_rarecurve <- data.frame(t(duodenum_SVs)) 
ileum_SVs_for_rarecurve <- data.frame(t(ileum_SVs)) 
cecum_SVs_for_rarecurve <- data.frame(t(cecum_SVs)) 

#duodenum_SVs_for_rarecurve <- data.frame(t(duodenum_SVs)) 
#duodenum_SVs_for_rarecurve$Group_Sample_Type <- duodenum_map$Sample_Type

duodenum_SVs_for_rarecurve_all <- data.frame(duodenum_SVs_for_rarecurve %>%  summarise_all("mean")) %>% mutate_if(is.numeric, round)

ileum_SVs_for_rarecurve_all <- data.frame(ileum_SVs_for_rarecurve %>%  summarise_all("mean")) %>% mutate_if(is.numeric, round)

cecum_SVs_for_rarecurve_all <- data.frame(cecum_SVs_for_rarecurve %>%  summarise_all("mean")) %>% mutate_if(is.numeric, round)
```

```{r}
intestine_SVs_for_rarecurve_all <- bind_rows(duodenum_SVs_for_rarecurve_all, ileum_SVs_for_rarecurve_all)

intestine_SVs_for_rarecurve_all <- bind_rows(intestine_SVs_for_rarecurve_all, cecum_SVs_for_rarecurve_all)
  
intestine_SVs_for_rarecurve_all[is.na(intestine_SVs_for_rarecurve_all)] <- 0

row.names(intestine_SVs_for_rarecurve_all) <- c("Duodenum", "Ileum", "Cecum")

intestine_SVs_for_rarecurve_all_mean <- rarecurve(intestine_SVs_for_rarecurve_all, step = 50, xlab = "Number of Samples", 
        ylab = "Observed SVs",label = True,
          tidy = TRUE)
```

Grouped-dont use
```{r}
duodenum_SVs_for_rarecurve <- data.frame(t(duodenum_SVs)) 
duodenum_SVs_for_rarecurve$Group_Sample_Type <- duodenum_map$Group_Sample_Type

duodenum_SVs_for_rarecurve_grouped <- data.frame(duodenum_SVs_for_rarecurve %>% group_by(Group_Sample_Type) %>% summarise_all("mean")) %>% mutate_if(is.numeric, round)

row.names(duodenum_SVs_for_rarecurve_grouped) <- duodenum_SVs_for_rarecurve_grouped$Group_Sample_Type

duodenum_SVs_for_rarecurve_grouped_length <- length(colnames(duodenum_SVs_for_rarecurve_grouped))

duodenum_SVs_for_rarecurve_grouped <- duodenum_SVs_for_rarecurve_grouped[,2:duodenum_SVs_for_rarecurve_grouped_length]

rarecurve_duodenum_mean <- rarecurve(duodenum_SVs_for_rarecurve_grouped, step = 50, sample = 1012, xlab = "Number of Samples", 
          ylab = "Observed SVs",
          label = FALSE, tidy = TRUE)
#ileum
ileum_SVs_for_rarecurve <- data.frame(t(ileum_SVs)) 
ileum_SVs_for_rarecurve$Group_Sample_Type <- ileum_map$Group_Sample_Type

ileum_SVs_for_rarecurve_grouped <- data.frame(ileum_SVs_for_rarecurve %>% group_by(Group_Sample_Type) %>% summarise_all("mean")) %>% mutate_if(is.numeric, round)

row.names(ileum_SVs_for_rarecurve_grouped) <- ileum_SVs_for_rarecurve_grouped$Group_Sample_Type

ileum_SVs_for_rarecurve_grouped_length <- length(colnames(ileum_SVs_for_rarecurve_grouped))

ileum_SVs_for_rarecurve_grouped <- ileum_SVs_for_rarecurve_grouped[,2:ileum_SVs_for_rarecurve_grouped_length]

rarecurve_ileum_mean <- rarecurve(ileum_SVs_for_rarecurve_grouped, step = 50, sample = 1012, xlab = "Number of Samples", 
          ylab = "Observed SVs",
          label = FALSE, tidy = TRUE)
#Cecum
cecum_SVs_for_rarecurve <- data.frame(t(cecum_SVs)) 
cecum_SVs_for_rarecurve$Group_Sample_Type <- cecum_map$Group_Sample_Type

cecum_SVs_for_rarecurve_grouped <- data.frame(cecum_SVs_for_rarecurve %>% group_by(Group_Sample_Type) %>% summarise_all("mean")) %>% mutate_if(is.numeric, round)

row.names(cecum_SVs_for_rarecurve_grouped) <- cecum_SVs_for_rarecurve_grouped$Group_Sample_Type

cecum_SVs_for_rarecurve_grouped_length <- length(colnames(cecum_SVs_for_rarecurve_grouped))

cecum_SVs_for_rarecurve_grouped <- cecum_SVs_for_rarecurve_grouped[,2:cecum_SVs_for_rarecurve_grouped_length]

rarecurve_cecum_mean <- rarecurve(cecum_SVs_for_rarecurve_grouped, step = 50, sample = 1012, xlab = "Number of Samples", 
          ylab = "Observed SVs",
          label = FALSE, tidy = TRUE)
```
Theme
```{r}
theme_rarecurve <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      text = element_text(size = 10),
      legend.position = "right",
      legend.title= element_blank(),
      #axis.title.x=element_blank(),
      #axis.text.x=element_blank(),
      #axis.ticks.x=element_blank(),
      #plot.title = element_text(size = 15, face = "bold"), 
      plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm") 
      )
}
```

ALl together
```{r}

p_rarecurve_intestine_mean <- ggplot(data = intestine_SVs_for_rarecurve_all_mean, aes(x = Sample, y = Species, color = Site)) +
  geom_line(size = 1, aes(linetype = Site)) +
  scale_colour_manual(values = section_color_list) +
  theme_rarecurve() +
  xlab("Sampling Depth") +
  ylab("Observed SVs")
 p_rarecurve_intestine_mean 
``` 
```{r}
 
 ggsave("./results/environmental_analysis/figures/supplemental_figure_2_rarefaction_all.png", p_rarecurve_intestine_mean , width = 6, height = 5)

ggsave("./results/environmental_analysis/figures/supplemental_figure_2_rarefaction_all.svg", p_rarecurve_intestine_mean , width = 6, height = 5)

```




Graphs- by group don't use
```{r}
p_rarecurve_duodenum_mean <- ggplot(data = rarecurve_duodenum_mean, aes(x = Sample, y = Species, color = Site)) +
  geom_line() +
  scale_colour_brewer(palette = "Paired") +
  theme_rarecurve() +
  ggtitle("Duodenum") +
  xlab("Sampling Depth") +
  ylab("Observed SVs")+
  xlim(0, 75000) 
 p_rarecurve_duodenum_mean 
 
 p_rarecurve_ileum_mean <- ggplot(data = rarecurve_ileum_mean, aes(x = Sample, y = Species, color = Site)) +
  geom_line() +
   scale_colour_brewer(palette = "Paired") +
  theme_rarecurve() +
  ggtitle("Ileum") +
  xlab("Sampling Depth") +
  ylab("Observed SVs")+
  xlim(0, 75000) 
 p_rarecurve_ileum_mean 
 
 p_rarecurve_cecum_mean <- ggplot(data = rarecurve_cecum_mean, aes(x = Sample, y = Species)) +
  geom_line(aes(color = Site)) +
  scale_colour_brewer(palette = "Paired") +
  theme_rarecurve() +
  ggtitle("Cecum") +
  xlab("Sampling Depth") +
  ylab("Observed SVs")+
  xlim(0, 75000) 
 p_rarecurve_cecum_mean 

```
Don't use
```{r}
supplemental_figure_2 <- plot_grid(p_rarecurve_duodenum_mean,p_rarecurve_ileum_mean,
                                   p_rarecurve_cecum_mean,
                  labels = c('A', 'B', 'C'),
                  label_size = 12, 
                  rel_widths = c(1,1),
                  rel_heights = c(1,1),
                  align = "h",
                  axis = "br",
                  ncol = 2, nrow = 2)

#ggsave("./results/environmental_analysis/figures/supplemental_figure_2_rarefaction.png",supplemental_figure_2, width = 12, height = 6)

#ggsave("./results/environmental_analysis/figures/supplemental_figure_2_rarefaction.svg",supplemental_figure_2, width = 12, height = 6)
```



### 5.1.3 Supplemental Figure 3: Mucosa vs Lumen
```{r}
theme_sample_type_plot <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      text = element_text(size = 10),
      plot.title = element_text(size = 10, hjust = 0.5, face = "bold"), 
      plot.margin = unit(c(0, 0.25, 0, 0), "cm") 
      )
}
```

Plots
```{r}
p_NMDS_duodenum_SVs_sample_type_group_genus_supp <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.12) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  theme(legend.position="none") +
  ggtitle("Duodenum (Euclidean)") +
  geom_segment(data = duodenum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*35, y=0, yend=NMDS2*35),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = duodenum_SVs_genus_scores_sig, size = 3,
            aes(x=NMDS1*35, y=NMDS2*35, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) 
p_NMDS_duodenum_SVs_sample_type_group_genus_supp

p_NMDS_ileum_SVs_sample_type_group_genus_supp <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.12) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  ggtitle("Ileum (Euclidean)") +
  theme(legend.position="none") +
  geom_segment(data = ileum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*35, y=0, yend=NMDS2*35),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = ileum_SVs_genus_scores_sig, size = 3,
            aes(x=NMDS1*35, y=NMDS2*35, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) 
p_NMDS_ileum_SVs_sample_type_group_genus_supp

p_NMDS_cecum_SVs_sample_type_group_genus_supp <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
  scale_color_manual(values = sample_type_color_list) +
  stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type,
                                     color =Sample_Type), alpha =0.12) +
  scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  ggtitle("Cecum (Euclidean)") +
  geom_segment(data = cecum_SVs_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*35, y=0, yend=NMDS2*35),
               arrow = arrow(length = unit(0.25, "cm"), type = "open"))  + 
  ggrepel::geom_text_repel(data = cecum_SVs_genus_scores_sig, size = 3,
            aes(x=NMDS1*35, y=NMDS2*35, label = Genus), cex = 4, 
            max.overlaps = Inf, direction = "both", segment.size = 0.25) 
p_NMDS_cecum_SVs_sample_type_group_genus_supp

p_NMDS_duodenum_unweighted_unifrac_sample_type_supp <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.12) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  theme(legend.position="none") +
  labs(title= "Duodenum (Unweighted Unifrac)") +
  geom_segment(data = duodenum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.3, y=0, yend=NMDS2*0.3),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = duodenum_unweighted_unifrac_genus_scores_sig, size = 3,
            aes(x=NMDS1*0.3, y=NMDS2*0.3, label = Genus), cex = 4, direction = "both", segment.size = 0.25, max.overlaps = Inf) 
p_NMDS_duodenum_unweighted_unifrac_sample_type_supp

p_NMDS_ileum_unweighted_unifrac_sample_type_supp <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.12) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  theme(legend.position="none") +
  labs(title= "Ileum (Unweighted Unifrac)") +
  geom_segment(data = ileum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.3, y=0, yend=NMDS2*0.3),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = ileum_unweighted_unifrac_genus_scores_sig, size = 3,
            aes(x=NMDS1*0.3, y=NMDS2*0.3, label = Genus), cex = 4, max.overlaps = Inf, direction = "both", segment.size = 0.25) 
p_NMDS_ileum_unweighted_unifrac_sample_type_supp


p_NMDS_cecum_unweighted_unifrac_sample_type_supp <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Sample_Type)) + 
    scale_color_manual(values = sample_type_color_list) +
    stat_ellipse(geom ="polygon",aes(x = NMDS1, y = NMDS2, fill = Sample_Type, color =Sample_Type), alpha =0.12) +
   scale_fill_manual(values = sample_type_color_list) +
  theme_sample_type_plot() +
  labs(title= "Cecum (Unweighted Unifrac)") +
  geom_segment(data = cecum_unweighted_unifrac_genus_scores_sig, aes(x = 0,   
               xend=NMDS1*0.02, y=0, yend=NMDS2*0.02),arrow = arrow(length =
              unit(0.25, "cm")), colour = "grey10", lwd=0.3) + 
  ggrepel::geom_text_repel(data = cecum_unweighted_unifrac_genus_scores_sig, size = 3,
            aes(x=NMDS1*0.02, y=NMDS2*0.02, label = Genus), cex = 4, max.overlaps = Inf, direction = "both", segment.size = 0.25) 
p_NMDS_cecum_unweighted_unifrac_sample_type_supp


```
Figure
```{r}
supplemental_figure_3 <- plot_grid(p_NMDS_duodenum_SVs_sample_type_group_genus_supp,
                                   p_NMDS_ileum_SVs_sample_type_group_genus_supp,
                                   p_NMDS_cecum_SVs_sample_type_group_genus_supp, 
                               p_NMDS_duodenum_unweighted_unifrac_sample_type_supp,
                                  p_NMDS_ileum_unweighted_unifrac_sample_type_supp,
                                  p_NMDS_cecum_unweighted_unifrac_sample_type_supp,
                  labels = c('A','','', 'B', '', ''),
                  label_size = 12, 
                  rel_heights = c(1,1),
                  rel_widths = c(3,3,4),
                  align = "h",
                  axis = "br",
                  ncol = 3, nrow = 2)
supplemental_figure_3

ggsave("./results/environmental_analysis/figures/supplemental_figure_2.png",supplemental_figure_3, width = 13, height = 7)

ggsave("./results/environmental_analysis/figures/supplemental_figure_2.svg",supplemental_figure_3, width = 13, height = 7)
```

### 5.1.5 Supplemental figure 5: NMDS all groups together for families
```{r}

p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces_2 <- ggplot(data = all_family_clr_c45_weekall_scores[!(all_family_clr_c45_weekall_scores$Section=="Feces"),],aes(x = NMDS1, y = NMDS2, color = Sample_Type_Env, fill = Sample_Type_Env)) + 
  geom_point() + 
  stat_ellipse(geom = "polygon", alpha = 0.18) +
  scale_color_manual(values = section_color_list, name="") +
  scale_fill_manual(values = section_color_list, name="") +
  theme_classic() +
  theme(text = element_text(size = 10)) 
p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces_2

ggsave("./results/environmental_analysis/figures/supplemental_figure_4_familyNMDS.png",p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces_2, width = 6, height = 4)

ggsave("./results/environmental_analysis/figures/supplemental_figure_4_familyNMDS.svg",p_NMDS_all_family_clr_c45_all_envs_sections_sample_no_feces_2, width = 6, height = 4)

```
### 5.1.4 Supplemental Figure 4: Taxa barplots for individual samples
Get top families to show for duodenum, ileum, cecum
```{r}
top_family_duodenum <- duodenum_family_mean_both_cohorts_rel_t_stack[duodenum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type=="Female_Mutant_Mucosa", ] %>% top_n(10, value) %>% pull(variable)

top_family_ileum <- ileum_family_mean_both_cohorts_rel_t_stack[ileum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type=="Female_Mutant_Mucosa", ] %>% top_n(10, value) %>% pull(variable)

top_family_cecum <- cecum_family_mean_both_cohorts_rel_t_stack[cecum_family_mean_both_cohorts_rel_t_stack$Group_Sample_Type=="Female_Mutant_Mucosa", ] %>% top_n(10, value) %>% pull(variable)
```

Rename labeling group to have spaces
```{r}
duodenum_family_rel_t_stack$Group_Sample_Type_Space <-gsub("_", " ", duodenum_family_rel_t_stack$Group_Sample_Type) 

ileum_family_rel_t_stack$Group_Sample_Type_Space <-gsub("_", " ", ileum_family_rel_t_stack$Group_Sample_Type) 

cecum_family_rel_t_stack$Group_Sample_Type_Space <-gsub("_", " ", cecum_family_rel_t_stack$Group_Sample_Type) 

#order the variables 
duodenum_family_rel_t_stack$Group_Sample_Type_Space <- factor(duodenum_family_rel_t_stack$Group_Sample_Type_Space, levels = c("Female Mutant Mucosa", "Female Wild Type Mucosa", "Male Mutant Mucosa", "Male Wild Type Mucosa", "Female Mutant Lumen", "Female Wild Type Lumen", "Male Mutant Lumen", "Male Wild Type Lumen" ))

ileum_family_rel_t_stack$Group_Sample_Type_Space <- factor(ileum_family_rel_t_stack$Group_Sample_Type_Space, levels = c("Female Mutant Mucosa", "Female Wild Type Mucosa", "Male Mutant Mucosa", "Male Wild Type Mucosa", "Female Mutant Lumen", "Female Wild Type Lumen", "Male Mutant Lumen", "Male Wild Type Lumen" ))

cecum_family_rel_t_stack$Group_Sample_Type_Space <- factor(cecum_family_rel_t_stack$Group_Sample_Type_Space, levels = c("Female Mutant Mucosa", "Female Wild Type Mucosa", "Male Mutant Mucosa", "Male Wild Type Mucosa", "Female Mutant Lumen", "Female Wild Type Lumen", "Male Mutant Lumen", "Male Wild Type Lumen" ))
```

Theme
```{r}
theme_taxa_plot <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      text = element_text(size = 10),
      axis.title.x=element_blank(),
      axis.text.x=element_blank(),
      axis.ticks.x=element_blank(),
      plot.title = element_text(size = 15, face = "bold"), 
      plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm") 
      )
}

p_duodenum_family_supp = ggplot(duodenum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_duodenum) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_taxa_plot() +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum") +
    facet_wrap(~Group_Sample_Type_Space, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(y = "Relative Abundance", fill = "Family") 
p_duodenum_family_supp

p_ileum_family_supp = ggplot(ileum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_ileum) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_taxa_plot() +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum") +
    facet_wrap(~Group_Sample_Type_Space, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(y = "Relative Abundance", fill = "Family") 
p_ileum_family_supp

p_cecum_family_supp = ggplot(cecum_family_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_cecum) +
    geom_bar(stat = "identity", color = "Black", width = 1, aes(x = Mouse_Label, fill = variable, y = value)) + 
    theme_taxa_plot() +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum") +
    facet_wrap(~Group_Sample_Type_Space, scales = "free_x", nrow = 1, shrink = TRUE) +
    labs(y = "Relative Abundance", fill = "Family") 
p_cecum_family_supp
```

```{r}
supplemental_figure_4 <- plot_grid(p_duodenum_family_supp,
          p_ileum_family_supp, 
          p_cecum_family_supp,
          labels = c('A  ','B  ','C   '), label_size = 14, ncol = 1, nrow = 3,
          align = 'v',
          rel_heights = c(1,1,1))

ggsave("./results/environmental_analysis/figures/supplemental_figure_4.png",supplemental_figure_4, width = 14, height = 12)

ggsave("./results/environmental_analysis/figures/supplemental_figure_4.svg",supplemental_figure_4, width = 14, height = 12)
```



## 5.2 Figure 2: NMDS by section/group and sourcetracker
Plots
```{r}
#A
p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group_2 <- ggplot(data = all_SVs_clr_c45_weekall_scores[(all_SVs_clr_c45_weekall_scores$Week=="10"),], 
                           aes(x = NMDS1, y = NMDS2, color = Sample_Type_Env, fill = Sample_Type_Env)) + 
  geom_point() + 
    stat_ellipse(geom = "polygon", alpha = 0.18) +
   scale_color_manual(values = section_color_list) +
  scale_fill_manual(values = section_color_list) +
  theme_classic() +
  theme(text = element_text(size = 10)) +
  facet_grid(Genotype~Sex)
p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group_2

#B
p_fecal_sourcetracker_means_bar_flip_2 <- ggplot(fecal_sourcetracker_means) +
    scale_fill_manual(values = section_color_list) +
    geom_bar(stat = "identity", color = "Black", width = 0.8, aes(x = Group, fill = Source, y = Mean)) + 
    theme_classic() +
    theme(text = element_text(size = 10))  +
    scale_y_continuous(expand = c(0,0)) + 
    theme(axis.text.x = element_text(angle = 0, hjust = 0.5)) +
    labs(x = "", y = "Source Proportion", fill = "Source") + 
    coord_flip() 
p_fecal_sourcetracker_means_bar_flip_2




```

Fig 2
```{r}
figure_2 <- plot_grid(p_NMDS_all_SVs_clr_c45_all_envs_sections_sample_group_2, 
                  p_fecal_sourcetracker_means_bar_flip_2,  labels = c('A', 'B'),
                  label_size = 12, 
                  rel_widths = c(1, 1),
                  ncol = 2, nrow = 1)

ggsave("./results/environmental_analysis/figures/figure_2_NMDS_SOURCE.png",figure_2, width = 12, height = 4)

ggsave("./results/environmental_analysis/figures/figure_2_NMDS_SOURCE.svg",figure_2, width = 12, height = 4)
```

## 5.3 Figure 3 Alpha diversity
Theme
```{r}
theme_alpha_plot <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      text = element_text(size = 12),
      legend.title= element_blank(), 
      legend.position="none",
      plot.title = element_text(size = 12, hjust = 0.5),
      axis.title.x=element_blank(),
      axis.ticks.x=element_blank(),
      axis.text.x=element_blank(),
      plot.margin = unit(c(0, 0, 0, 0.5), "cm") 
      )
    }
```
Graphs
```{r}
#A
p_shannon_intestine_SVs_group_2 <- ggplot(data = intestine_alpha) +
    geom_boxplot(aes(x = Sample_Type, y = Shannon, color = Sample_Type))+
  geom_point(aes(x = Sample_Type, 
                 y = Shannon, color = Sample_Type), size = 1) + 
  scale_color_manual(values = sample_type_color_list) +
  theme_alpha_plot() +
  labs(title = "Shannon Index") +
  guides(color=guide_legend(title="")) +
  facet_grid(~Section) +
      theme(axis.text.x=element_blank())+
  ylim(0,8) 

#B
p_faith_intestine_SVs_group_2 <- ggplot(data = intestine_alpha) +
    geom_boxplot(aes(x = Sample_Type, y = Faith_PD, color = Sample_Type))+
  geom_point(aes(x = Sample_Type, y = Faith_PD, color = Sample_Type), size = 1) + 
  scale_color_manual(values = sample_type_color_list) +
  labs(title = "Faith's PD") +
  theme_alpha_plot() +
  guides(color=guide_legend(title="")) +
  facet_grid(~Section)  +
    theme(axis.text.x=element_blank()) +
  ylim(0,8) 


p_shannon_duodenum_SVs_fig <- ggplot(data = duodenum_alpha) +
    geom_boxplot(aes(x = Group, y = Shannon, color = Group))+
  geom_point(aes(x = Group, y = Shannon, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
  theme(axis.text.x=element_blank()) +
  ylim(0,8) 


p_faith_duodenum_SVs_fig <- ggplot(data = duodenum_alpha) +
    geom_boxplot(aes(x = Group, y = Faith_PD, color = Group))+
  geom_point(aes(x = Group, y = Faith_PD, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
    theme(axis.text.x=element_blank()) +
  ylim(0,8) 

p_shannon_ileum_SVs_fig <- ggplot(data = ileum_alpha) +
    geom_boxplot(aes(x = Group, y = Shannon, color = Group))+
  geom_point(aes(x = Group, y = Shannon, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
  theme(axis.text.x=element_blank()) +
  ylim(0,8) 

p_faith_ileum_SVs_fig <- ggplot(data = ileum_alpha) +
    geom_boxplot(aes(x = Group, y = Faith_PD, color = Group))+
  geom_point(aes(x = Group, y = Faith_PD, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
    theme(axis.text.x=element_blank()) +
  ylim(0,8) 

p_shannon_cecum_SVs_fig <- ggplot(data = cecum_alpha) +
    geom_boxplot(aes(x = Group, y = Shannon, color = Group))+
  geom_point(aes(x = Group, y = Shannon, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
  ylim(0,8)

p_faith_cecum_SVs_fig <- ggplot(data = cecum_alpha) +
    geom_boxplot(aes(x = Group, y = Faith_PD, color = Group))+
  geom_point(aes(x = Group, y = Faith_PD, color = Group)) + 
    scale_color_manual(values = group_color_list) +
  theme_alpha_plot() +
  facet_grid(~Sample_Type) +
  ylim(0,8)
```

Legends
```{r}
legend_sample_type_side <- as_ggplot(get_legend(p_shannon_intestine_SVs_group_2 + theme(legend.position = "right")))
legend_sample_type_side

legend_sample_type_flat <- as_ggplot(get_legend(p_shannon_intestine_SVs_group_2 + theme(legend.position = "bottom")))


legend_groups_side <- as_ggplot(get_legend(p_faith_cecum_SVs_fig + theme(legend.position = "right")))
legend_groups_side

legend_groups_flat <- as_ggplot(get_legend(p_faith_cecum_SVs_fig + theme(legend.position = "bottom")))
legend_groups_flat
```


figure 3
```{r}

figure_3_alpha_AB <- plot_grid(p_shannon_intestine_SVs_group_2,
                                   p_faith_intestine_SVs_group_2,
                               labels = c('A', 'B'), 
                               rel_widths = c(3,3),
                               label_size = 12,
                                   ncol = 2, nrow = 1)

figure_3_alpha_CD <- plot_grid(p_shannon_duodenum_SVs_fig,
                                   p_faith_duodenum_SVs_fig,
                               labels = c('C', 'D'), 
                               rel_widths = c(3,3),
                               label_size = 12,
                                   ncol = 2, nrow = 1)   

figure_3_alpha_EF <- plot_grid(p_shannon_ileum_SVs_fig,
                                   p_faith_ileum_SVs_fig,
                               labels = c('E', 'F'), 
                               rel_widths = c(3,3),
                               label_size = 12,
                                   ncol = 2, nrow = 1)

figure_3_alpha_GH <- plot_grid(p_shannon_cecum_SVs_fig,
                                   p_faith_cecum_SVs_fig,
                               labels = c('G', 'H'), 
                               rel_widths = c(3,3),
                               label_size = 12,
                                   ncol = 2, nrow = 1)
                               
                               
 figure_3_alpha <- plot_grid(figure_3_alpha_AB, legend_sample_type_flat,
                       figure_3_alpha_CD, legend_groups_flat,
                       figure_3_alpha_EF, legend_groups_flat,
                       figure_3_alpha_GH,
                                   rel_heights = c(6,1,6,1,6,1,6),
                                  label_size = 12,
                                   ncol = 1, nrow = 7)

ggsave("./results/environmental_analysis/figures/figure_3_alpha.png",figure_3_alpha, width = 6, height = 12)

ggsave("./results/environmental_analysis/figures/figure_3_alpha.svg",figure_3_alpha, width = 6, height = 12)
```


## 5.4 Figure 4 Taxa family barplots (averages)
```{r}
#A
p_duodenum_family_mean_both_cohorts_flip_genotype <- ggplot(duodenum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_duodenum) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 10))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Duodenum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "Family") +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=1))
p_duodenum_family_mean_both_cohorts_flip_genotype

#B
p_ileum_family_mean_both_cohorts_flip_genotype <- ggplot(ileum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_ileum) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 10))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Ileum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "") +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=1))
p_ileum_family_mean_both_cohorts_flip_genotype

#C
p_cecum_family_mean_both_cohorts_flip_genotype <- ggplot(cecum_family_mean_both_cohorts_rel_t_stack) +
    scale_fill_manual(values = family_color_list, breaks = top_family_cecum) +
    geom_bar(stat = "identity", color = "Black", width = 1, 
             aes(x = Group_Sample_Type_gen, fill = variable, y = value)) + 
    theme_classic() +
    theme(text = element_text(size = 10))  +
    scale_y_continuous(expand = c(0,0)) + 
    ggtitle(label = "Cecum Family Mean Relative Abundance") +
    theme(axis.text.x = element_text(angle = 90, hjust = 0.5)) +
    labs(x = "", y = "Relative Abundance", fill = "") +
    coord_flip() +
  theme(axis.text.x = element_text(angle = 0, vjust = 0.5, hjust=1))
p_cecum_family_mean_both_cohorts_flip_genotype

```


```{r}

figure_4_FGH <- plot_grid( p_duodenum_family_mean_both_cohorts_flip_genotype, 
          p_ileum_family_mean_both_cohorts_flip_genotype, 
          p_cecum_family_mean_both_cohorts_flip_genotype,
          labels = c('F', 'G', 'H'), label_size = 12 ,
                  rel_heights = c(1, 1, 1), 
                  ncol = 1, nrow= 3)

figure_2 <- plot_grid(figure_2_AB,
          figure_2_CDE, 
          figure_2_FGH,
          labels = c('','',''), label_size = 1, ncol = 1, nrow = 3,
          rel_heights = c(4,4,11))

ggsave("./results/environmental_analysis/figures/figure2.png",figure_2, width = 12, height = 14)

ggsave("./results/environmental_analysis/figures/figure2.svg",figure_2, width = 12, height = 14)
```


## 5.5 Figure 5 NMDS sex/genotype plots
```{r}
duodenum_SVs_scores$Group_Space <-gsub("_", " ", duodenum_SVs_scores$Group)

ileum_SVs_scores$Group_Space <-gsub("_", " ", ileum_SVs_scores$Group)

cecum_SVs_scores$Group_Space <-gsub("_", " ", cecum_SVs_scores$Group)
```


Theme
```{r}
theme_NMDS_plot <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      text = element_text(size = 10),
      legend.position = "none",
      plot.title = element_text(size = 12, face = "bold"), 
      plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm") 
      )
}
```
get legend
```{r}
p_NMDS_duodenum_euclidean_sex_diff_legend <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_classic()  +
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(legend.position="bottom") +
  theme(legend.title=element_blank())

legend_NMDS_groups <- get_legend(p = p_NMDS_duodenum_euclidean_sex_diff_legend)
```
Plots
```{r}
#Duodenum
#Genotype Differences
p_NMDS_duodenum_euclidean_genotype_diff_figure <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Euclidean") +
  theme(axis.title.x=element_blank()) +
  facet_grid(Sex~Sample_Type) 
p_NMDS_duodenum_euclidean_genotype_diff_figure

p_NMDS_duodenum_unweighted_unifrac_genotype_diff_figure <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
  facet_grid(Sex~Sample_Type) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  scale_x_continuous(breaks = c(-0.5, 0, 0.5)) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2))
p_NMDS_duodenum_unweighted_unifrac_genotype_diff_figure

#Sex Differences
p_NMDS_duodenum_euclidean_sex_diff_figure <- ggplot(data = duodenum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  facet_grid(Genotype~Sample_Type) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  labs(title= "Euclidean") 
p_NMDS_duodenum_euclidean_sex_diff_figure

p_NMDS_duodenum_unweighted_unifrac_sex_diff_figure <- ggplot(data = duodenum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
    scale_x_continuous(breaks = c(-0.5, 0, 0.5)) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2)) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  facet_grid(Genotype~Sample_Type) 
p_NMDS_duodenum_unweighted_unifrac_sex_diff_figure

#ileum
#Genotype Differences
p_NMDS_ileum_euclidean_genotype_diff_figure <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Euclidean") +
  theme(axis.title.x=element_blank()) +
  facet_grid(Sex~Sample_Type) 
p_NMDS_ileum_euclidean_genotype_diff_figure

p_NMDS_ileum_unweighted_unifrac_genotype_diff_figure <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
  scale_x_continuous(breaks = c(-0.4, 0, 0.4)) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2)) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  facet_grid(Sex~Sample_Type) 
p_NMDS_ileum_unweighted_unifrac_genotype_diff_figure

#Sex Differences
p_NMDS_ileum_euclidean_sex_diff_figure <- ggplot(data = ileum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  facet_grid(Genotype~Sample_Type) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  labs(title= "Euclidean") 
p_NMDS_ileum_euclidean_sex_diff_figure

p_NMDS_ileum_unweighted_unifrac_sex_diff_figure <- ggplot(data = ileum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
  scale_x_continuous(breaks = c(-0.4, 0, 0.4)) +
  scale_y_continuous(breaks = c(-0.2, 0, 0.2)) +
  theme(axis.title.y=element_blank()) +
  theme(axis.title.x=element_blank()) +
  facet_grid(Genotype~Sample_Type) 
p_NMDS_ileum_unweighted_unifrac_sex_diff_figure

#cecum
#Genotype Differences
p_NMDS_cecum_euclidean_genotype_diff_figure <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Euclidean") +
  facet_grid(Sex~Sample_Type) 
p_NMDS_cecum_euclidean_genotype_diff_figure

p_NMDS_cecum_unweighted_unifrac_genotype_diff_figure <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
  scale_x_continuous(breaks = c(-0.02, 0, 0.02)) +
  scale_y_continuous(breaks = c(-0.04, 0, 0.04)) +
  theme(axis.title.y=element_blank()) +
  facet_grid(Sex~Sample_Type) 
p_NMDS_cecum_unweighted_unifrac_genotype_diff_figure

#Sex Differences
p_NMDS_cecum_euclidean_sex_diff_figure <- ggplot(data = cecum_SVs_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  facet_grid(Genotype~Sample_Type) +
  theme(axis.title.y=element_blank()) +
  labs(title= "Euclidean") 
p_NMDS_cecum_euclidean_sex_diff_figure

p_NMDS_cecum_unweighted_unifrac_sex_diff_figure <- ggplot(data = cecum_unweighted_unifrac_scores) + 
  geom_point(aes(x = NMDS1, y = NMDS2, color = Group)) + 
    scale_color_manual(values = group_color_list) +
    stat_ellipse(geom = "polygon",aes(x = NMDS1, y = NMDS2, fill = Group, color = Group), alpha = 0.18) +
   scale_fill_manual(values = group_color_list) +
  theme_NMDS_plot() +
  labs(title= "Unweighted UniFrac") +
  scale_x_continuous(breaks = c(-0.02, 0, 0.02)) +
  scale_y_continuous(breaks = c(-0.04, 0, 0.04)) +
  theme(axis.title.y=element_blank()) +
  facet_grid(Genotype~Sample_Type) 
p_NMDS_cecum_unweighted_unifrac_sex_diff_figure
```

Figure
```{r}
figure_3_duodenum <- plot_grid(p_NMDS_duodenum_euclidean_genotype_diff_figure,
          p_NMDS_duodenum_unweighted_unifrac_genotype_diff_figure,
          p_NMDS_duodenum_euclidean_sex_diff_figure, 
          p_NMDS_duodenum_unweighted_unifrac_sex_diff_figure,
          labels = c('A','', 'B', ''), label_size = 12, ncol = 4, nrow = 1,
          rel_widths = c(1,1,1,1),
          align = 'h',
          axis = "b")

figure_3_ileum <- plot_grid(p_NMDS_ileum_euclidean_genotype_diff_figure,
          p_NMDS_ileum_unweighted_unifrac_genotype_diff_figure,
          p_NMDS_ileum_euclidean_sex_diff_figure, 
          p_NMDS_ileum_unweighted_unifrac_sex_diff_figure,
          labels = c('C','', 'D', ''), label_size = 12, ncol = 4, nrow = 1,
          align = 'h',
          rel_widths = c(1,1,1,1),
          axis = "b")

figure_3_cecum <- plot_grid(p_NMDS_cecum_euclidean_genotype_diff_figure,
          p_NMDS_cecum_unweighted_unifrac_genotype_diff_figure,
          p_NMDS_cecum_euclidean_sex_diff_figure, 
          p_NMDS_cecum_unweighted_unifrac_sex_diff_figure,
          labels = c('E','', 'F', ''), label_size = 12, ncol = 4, nrow = 1,
          align = 'h',
          rel_widths = c(1,1,1,1),
          axis = "b")

figure_3 <- plot_grid(figure_3_duodenum,
                      legend_NMDS_groups,
                      figure_3_ileum,
                      legend_NMDS_groups,
                      figure_3_cecum,
          labels = c('',''), ncol = 1, nrow = 5,
          align = 'v',
          axis = "l",
          rel_heights = c(6,1, 6, 1, 6),
          rel_widths = c(1))
figure_3

ggsave("./results/environmental_analysis/figures/figure_3.png",figure_3, width = 12, height = 10)

ggsave("./results/environmental_analysis/figures/figure_3.svg",figure_3, width = 12, height = 10)
```



## 5.4 Figure 6 Balances- short version
edit genus names that are long
Wild_Type
```{r}
coda_balance_coeff_names_duodenum_genus_wild_type_10p <- coda_balance_coeff_duodenum_genus_wild_type_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_duodenum_genus_wild_type <- coda_balance_coeff_duodenum_genus_wild_type %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_wild_type_10p <- coda_balance_coeff_ileum_genus_wild_type_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_wild_type <- coda_balance_coeff_ileum_genus_wild_type %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_wild_type_10p <- coda_balance_coeff_cecum_genus_wild_type_10p %>%  mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_wild_type <- coda_balance_coeff_cecum_genus_wild_type %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 
```
mutant
```{r}
coda_balance_coeff_names_duodenum_genus_mutant_10p <- coda_balance_coeff_duodenum_genus_mutant_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_duodenum_genus_mutant <- coda_balance_coeff_duodenum_genus_mutant %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_mutant_10p <- coda_balance_coeff_ileum_genus_mutant_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_mutant <- coda_balance_coeff_ileum_genus_mutant %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_mutant_10p <- coda_balance_coeff_cecum_genus_mutant_10p %>%  mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_mutant <- coda_balance_coeff_cecum_genus_mutant %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 
```
female
```{r}
coda_balance_coeff_names_duodenum_genus_female_10p <- coda_balance_coeff_duodenum_genus_female_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_duodenum_genus_female <- coda_balance_coeff_duodenum_genus_female %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_female_10p <- coda_balance_coeff_ileum_genus_female_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_female <- coda_balance_coeff_ileum_genus_female %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_female_10p <- coda_balance_coeff_cecum_genus_female_10p %>%  mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_female <- coda_balance_coeff_cecum_genus_female %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 
```
male
```{r}
coda_balance_coeff_names_duodenum_genus_male_10p <- coda_balance_coeff_duodenum_genus_male_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_duodenum_genus_male <- coda_balance_coeff_duodenum_genus_male %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_male_10p <- coda_balance_coeff_ileum_genus_male_10p %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_ileum_genus_male <- coda_balance_coeff_ileum_genus_male %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_male_10p <- coda_balance_coeff_cecum_genus_male_10p %>%  mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 

coda_balance_coeff_names_cecum_genus_male <- coda_balance_coeff_cecum_genus_male %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium')) 
```

### 5.3.1 Fig 4 Balances - Balance prediction plots
```{r}
theme_balances_plot <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      axis.title.y=element_blank(),
      text = element_text(size = 10),
      legend.title= element_blank(), 
      legend.position="none",
      plot.title = element_text(size = 10, hjust = 0.5), 
      plot.margin = unit(c(0, 0.25, 0, 0), "cm") 
      )
}
```
Duodenum
```{r}
## wild_type 
p_duodenum_coda_sample_balances_wild_type_fig <- ggplot(data = coda_balances_duodenum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_duodenum_coda_sample_balances_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_balances_wild_type_sex_diff.png",width=5.5, height=2, p_duodenum_coda_sample_balances_wild_type_fig)

## mutant 
p_duodenum_coda_sample_balances_mutant_fig <- ggplot(data = coda_balances_duodenum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_duodenum_coda_sample_balances_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_balances_mutant_sex_diff.png",width=5.5, height=2, p_duodenum_coda_sample_balances_mutant_fig)

## female 
p_duodenum_coda_sample_balances_female_fig <- ggplot(data = coda_balances_duodenum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_duodenum_coda_sample_balances_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_balances_female_Genotype_diff.png",width=5.5, height=2, p_duodenum_coda_sample_balances_female_fig)

## male 
p_duodenum_coda_sample_balances_male_fig <- ggplot(data = coda_balances_duodenum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_duodenum_coda_sample_balances_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_balances_male_Genotype_diff.png",width=5.5, height=2, p_duodenum_coda_sample_balances_male_fig)
```

ileum
```{r}
## wild_type 
p_ileum_coda_sample_balances_wild_type_fig <- ggplot(data = coda_balances_ileum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_ileum_coda_sample_balances_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_balances_wild_type_sex_diff.png",width=5.5, height=2, p_ileum_coda_sample_balances_wild_type_fig)

## mutant 
p_ileum_coda_sample_balances_mutant_fig <- ggplot(data = coda_balances_ileum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_ileum_coda_sample_balances_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_balances_mutant_sex_diff.png",width=5.5, height=2, p_ileum_coda_sample_balances_mutant_fig)

## female 
p_ileum_coda_sample_balances_female_fig <- ggplot(data = coda_balances_ileum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_ileum_coda_sample_balances_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_balances_female_Genotype_diff.png",width=5.5, height=2, p_ileum_coda_sample_balances_female_fig)

## male 
p_ileum_coda_sample_balances_male_fig <- ggplot(data = coda_balances_ileum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_ileum_coda_sample_balances_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_balances_male_Genotype_diff.png",width=5.5, height=2, p_ileum_coda_sample_balances_male_fig)
```

cecum
```{r}
## wild_type 
p_cecum_coda_sample_balances_wild_type_fig <- ggplot(data = coda_balances_cecum_genus_wild_type_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_cecum_coda_sample_balances_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_balances_wild_type_sex_diff.png",width=5.5, height=2, p_cecum_coda_sample_balances_wild_type_fig)

## mutant 
p_cecum_coda_sample_balances_mutant_fig <- ggplot(data = coda_balances_cecum_genus_mutant_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_cecum_coda_sample_balances_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_balances_mutant_sex_diff.png",width=5.5, height=2, p_cecum_coda_sample_balances_mutant_fig)

## female 
p_cecum_coda_sample_balances_female_fig <- ggplot(data = coda_balances_cecum_genus_female_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_cecum_coda_sample_balances_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_balances_female_Genotype_diff.png",width=5.5, height=2, p_cecum_coda_sample_balances_female_fig)

## male 
p_cecum_coda_sample_balances_male_fig <- ggplot(data = coda_balances_cecum_genus_male_predictions, aes(x = Sex, y = Balance, color = Group)) +
    geom_boxplot()+
    geom_point() +
    scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_balances_plot() +
  coord_flip() +
  ylab("Sample Balance Value") 
p_cecum_coda_sample_balances_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_balances_male_Genotype_diff.png",width=5.5, height=2, p_cecum_coda_sample_balances_male_fig)
```

### 5.3.2  Fig 4  Coefficient plots Coefficients (coeff > 0.099) 
Theme
```{r}
theme_coefficient_plot_05 <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      axis.title.y=element_blank(),
      text = element_text(size = 10, face = "bold"),
      legend.title= element_blank(), 
      legend.position="none",
      plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
      plot.margin = unit(c(1, 0.25, 0, 0), "cm") 
      )
}
```

```{r}
## Wild-type
p_duodenum_coda_sample_genera_wild_type_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  labs(title = "Sex (Wild Type)") +
  xlab("") +ylab("Genus Coefficient") +
  theme_coefficient_plot_05() +
  coord_flip() 
p_duodenum_coda_sample_genera_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_coda_coeffieints_0.05_wild_type_sex_diff.png",width=6, height=4, p_duodenum_coda_sample_genera_wild_type_fig)

## mutant
p_duodenum_coda_sample_genera_mutant_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Sex (Mutant)") +
  xlab("") + ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_coda_coeffieints_0.05_mutant_sex_diff.png",width=6, height=4, p_duodenum_coda_sample_genera_mutant_fig)

## female
p_duodenum_coda_sample_genera_female_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Female)") +
  xlab("") +ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_coda_coeffieints_0.05_female_Genotype_diff.png",width=6, height=4, p_duodenum_coda_sample_genera_female_fig)

## male
p_duodenum_coda_sample_genera_male_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Male)") +
  xlab("") + ylab("Genus Coefficient") +
  #theme(text = element_text(size = 10))  +
  coord_flip() 
p_duodenum_coda_sample_genera_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_coda_coeffieints_0.05_male_Genotype_diff.png",width=6, height=4, p_duodenum_coda_sample_genera_male_fig)

## Wild-type
p_ileum_coda_sample_genera_wild_type_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  labs(title = "Sex (Wild Type)") +
  xlab("") +ylab("Genus Coefficient") +
  theme_coefficient_plot_05() +
  coord_flip() 
p_ileum_coda_sample_genera_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_coda_coeffieints_0.05_wild_type_sex_diff.png",width=6, height=4, p_ileum_coda_sample_genera_wild_type_fig)

## mutant
p_ileum_coda_sample_genera_mutant_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Sex (Mutant)") +
  xlab("") + ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_coda_coeffieints_0.05_mutant_sex_diff.png",width=6, height=4, p_ileum_coda_sample_genera_mutant_fig)

## female
p_ileum_coda_sample_genera_female_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Female)") +
  xlab("") +ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_coda_coeffieints_0.05_female_Genotype_diff.png",width=6, height=4, p_ileum_coda_sample_genera_female_fig)

## male
p_ileum_coda_sample_genera_male_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Male)") +
  xlab("") + ylab("Genus Coefficient") +
  #theme(text = element_text(size = 10))  +
  coord_flip() 
p_ileum_coda_sample_genera_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/ileum_coda_coeffieints_0.05_male_Genotype_diff.png",width=6, height=4, p_ileum_coda_sample_genera_male_fig)

## Wild-type
p_cecum_coda_sample_genera_wild_type_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_wild_type_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  labs(title = "Sex (Wild Type)") +
  xlab("") +ylab("Genus Coefficient") +
  theme_coefficient_plot_05() +
  coord_flip() 
p_cecum_coda_sample_genera_wild_type_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_coda_coeffieints_0.05_wild_type_sex_diff.png",width=6, height=4, p_cecum_coda_sample_genera_wild_type_fig)

## mutant
p_cecum_coda_sample_genera_mutant_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_mutant_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Sex (Mutant)") +
  xlab("") + ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_mutant_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_coda_coeffieints_0.05_mutant_sex_diff.png",width=6, height=4, p_cecum_coda_sample_genera_mutant_fig)

## female
p_cecum_coda_sample_genera_female_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_female_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Female)") +
  xlab("") +ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_female_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_coda_coeffieints_0.05_female_Genotype_diff.png",width=6, height=4, p_cecum_coda_sample_genera_female_fig)

## male
p_cecum_coda_sample_genera_male_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_male_10p, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group), show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_05() +
  labs(title = "Genotype (Male)") +
  xlab("") + ylab("Genus Coefficient") +
  #theme(text = element_text(size = 10))  +
  coord_flip() 
p_cecum_coda_sample_genera_male_fig

ggsave("./results/environmental_analysis/figures/fig4_balances/cecum_coda_coeffieints_0.05_male_Genotype_diff.png",width=6, height=4, p_cecum_coda_sample_genera_male_fig)
```


### 5.3.4 Figure 4
legend_groups_side
legend_groups_flat

Try putting this together differently so everything can be aligned vertically

```{r}
figure_4_duodenum_sex_wild_type <- plot_grid(
                  p_duodenum_coda_sample_genera_wild_type_fig,
                  p_duodenum_coda_sample_balances_wild_type_fig,
                  p_ileum_coda_sample_genera_wild_type_fig,
                  p_ileum_coda_sample_balances_wild_type_fig,
                  p_cecum_coda_sample_genera_wild_type_fig,
                  p_cecum_coda_sample_balances_wild_type_fig,
                  labels = c('A Duodenum', '','B Ileum', '','C Cecum', ''),
                  label_size = 12, 
                  rel_heights = c(3,1, 3, 1, 3, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
figure_4_duodenum_sex_wild_type

figure_4_duodenum_sex_mutant <- plot_grid(
                  p_duodenum_coda_sample_genera_mutant_fig,
                  p_duodenum_coda_sample_balances_mutant_fig,
                  p_ileum_coda_sample_genera_mutant_fig,
                  p_ileum_coda_sample_balances_mutant_fig,
                  p_cecum_coda_sample_genera_mutant_fig,
                  p_cecum_coda_sample_balances_mutant_fig,
                  rel_heights = c(3,1, 3, 1, 3, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
figure_4_duodenum_sex_mutant

figure_4_duodenum_sex_female <- plot_grid(
                  p_duodenum_coda_sample_genera_female_fig,
                  p_duodenum_coda_sample_balances_female_fig,
                  p_ileum_coda_sample_genera_female_fig,
                  p_ileum_coda_sample_balances_female_fig,
                  p_cecum_coda_sample_genera_female_fig,
                  p_cecum_coda_sample_balances_female_fig,
                  rel_heights = c(3,1, 3, 1, 3, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
figure_4_duodenum_sex_female

figure_4_duodenum_sex_male <- plot_grid(
                  p_duodenum_coda_sample_genera_male_fig,
                  p_duodenum_coda_sample_balances_male_fig,
                  p_ileum_coda_sample_genera_male_fig,
                  p_ileum_coda_sample_balances_male_fig,
                  p_cecum_coda_sample_genera_male_fig,
                  p_cecum_coda_sample_balances_male_fig,
                  label_size = 12, 
                  rel_heights = c(3,1, 3, 1, 3, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
figure_4_duodenum_sex_male

figure_4_balances <- plot_grid(figure_4_duodenum_sex_wild_type,
                      figure_4_duodenum_sex_mutant,
                      figure_4_duodenum_sex_female,
                      figure_4_duodenum_sex_male,
                  label_size = 12, 
                  rel_widths = c(1,1,1,1),
                  align = "h",
                  axis = "b",
                  ncol = 4, nrow = 1)

#figure_4_heatmaps <- plot_grid(p_heatmap_genus_facet_duodenum_interesting_balance_fig,
#                      p_heatmap_genus_facet_ileum_interesting_balance_fig,
#                  p_heatmap_genus_facet_cecum_interesting_balance_fig + theme(legend.position="none"),
#                  get_legend(p_heatmap_genus_facet_cecum_interesting_balance_fig),
 #                 rel_widths = c(4.3,3.9,3.9,1.2),
  #                labels = c('D', 'E','F', ""),
   #               label_size = 12,
    #              align = "h",
     #             axis = "t",
      #            ncol = 4, nrow = 1)

#figure_4 <- plot_grid(figure_4_balances,
 #                     figure_4_heatmaps,
  #                rel_heights = c(4,1),
   #               ncol = 1, nrow = 2)

ggsave("./results/environmental_analysis/figures/fig4_balances/figure_4_balances.png",figure_4_balances, width = 12, height = 10)

ggsave("./results/environmental_analysis/figures/fig4_balances/figure_4_balances.svg",figure_4_balances, width = 12, height = 10)
```


## 5.5  Supplemental Figure 6 Coefficient plots all Coefficients 
Theme
```{r}
theme_coefficient_plot_all <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      axis.title.y=element_blank(),
      text = element_text(size = 10, face = "bold"),
      legend.title= element_blank(), 
      legend.position="none",
      plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
      plot.margin = unit(c(1, 0.25, 0, 0), "cm") 
      )
}
```
Plots
```{r}
#Duodenum
#Wild_type
p_duodenum_coda_sample_genera_wild_type_all_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Wild Type)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_wild_type_all_fig

#mutant
p_duodenum_coda_sample_genera_mutant_all_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Mutant)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_mutant_all_fig

#female
p_duodenum_coda_sample_genera_female_all_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Female)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_female_all_fig

#male
p_duodenum_coda_sample_genera_male_all_fig <- ggplot(
  data = coda_balance_coeff_names_duodenum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Male)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_duodenum_coda_sample_genera_male_all_fig

#ileum
#Wild_type
p_ileum_coda_sample_genera_wild_type_all_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Wild Type)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_wild_type_all_fig

#mutant
p_ileum_coda_sample_genera_mutant_all_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Mutant)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_mutant_all_fig

#female
p_ileum_coda_sample_genera_female_all_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Female)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_female_all_fig

#male
p_ileum_coda_sample_genera_male_all_fig <- ggplot(
  data = coda_balance_coeff_names_ileum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Male)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_ileum_coda_sample_genera_male_all_fig

#cecum
#Wild_type
p_cecum_coda_sample_genera_wild_type_all_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_wild_type, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Wild Type)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_wild_type_all_fig

#mutant
p_cecum_coda_sample_genera_mutant_all_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_mutant, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Sex (Mutant)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_mutant_all_fig

#female
p_cecum_coda_sample_genera_female_all_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_female, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Female)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_female_all_fig

#male
p_cecum_coda_sample_genera_male_all_fig <- ggplot(
  data = coda_balance_coeff_names_cecum_genus_male, 
  aes(x = reorder(Genus, Coefficient), y = Coefficient)) +
  geom_bar(stat="identity", width = 0.7, aes(color = Group, fill = Group),show.legend = FALSE) +
  scale_color_manual(values = group_color_list) +
  scale_fill_manual(values = group_color_list) +
  theme_coefficient_plot_all() +
  labs(title = "Genotype (Male)") +
  ylab("Genus Coefficient") +
  coord_flip() 
p_cecum_coda_sample_genera_male_all_fig

#ggsave("./results/environmental_analysis/figures/fig4_balances/duodenum_coda_coeffieints_all_wild_type_sex_diff.png",width=7, height=10, p_duodenum_coda_sample_genera_wild_type_all_fig)
```
Add AUC values  in illustrator 
```{r}
#duodenum AUC
duodenum_wild_type_AUC <- paste(c("AUC = ", round(coda_glmnet_duodenum_genus_wild_type$`apparent AUC`, digits = 3)), collapse = " ")
duodenum_mutant_AUC <- paste(c("AUC = ", round(coda_glmnet_duodenum_genus_mutant$`apparent AUC`, digits = 3)), collapse = " ")
duodenum_female_AUC <- paste(c("AUC = ", round(coda_glmnet_duodenum_genus_female$`apparent AUC`, digits = 3)), collapse = " ")
duodenum_male_AUC <- paste(c("AUC = ", round(coda_glmnet_duodenum_genus_male$`apparent AUC`, digits = 3)), collapse = " ")

```
```{r}
coda_glmnet_duodenum_genus_wild_type$`mean cv-AUC`
coda_glmnet_ileum_genus_wild_type$`mean cv-AUC`
coda_glmnet_cecum_genus_wild_type$`mean cv-AUC`

coda_glmnet_duodenum_genus_mutant$`mean cv-AUC`
coda_glmnet_ileum_genus_mutant$`mean cv-AUC`
coda_glmnet_cecum_genus_mutant$`mean cv-AUC`

coda_glmnet_duodenum_genus_female$`mean cv-AUC`
coda_glmnet_ileum_genus_female$`mean cv-AUC`
coda_glmnet_cecum_genus_female$`mean cv-AUC`

coda_glmnet_duodenum_genus_male$`mean cv-AUC`
coda_glmnet_ileum_genus_male$`mean cv-AUC`
coda_glmnet_cecum_genus_male$`mean cv-AUC`
```


Figure
p_duodenum_coda_sample_genera_wild_type_all_fig
```{r}
supplemental_figure_5_duodenum_sex_wild_type <- plot_grid(
                  p_duodenum_coda_sample_genera_wild_type_all_fig,
                  p_duodenum_coda_sample_balances_wild_type_fig,
                  p_ileum_coda_sample_genera_wild_type_all_fig,
                  p_ileum_coda_sample_balances_wild_type_fig,
                  p_cecum_coda_sample_genera_wild_type_all_fig,
                  p_cecum_coda_sample_balances_wild_type_fig,
                  labels = c('A Duodenum', '','B Ileum', '','C Cecum', ''),
                  label_size = 12, 
                  rel_heights = c(5,1, 5, 1, 5, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
supplemental_figure_5_duodenum_sex_wild_type

supplemental_figure_5_duodenum_sex_mutant <- plot_grid(
                  p_duodenum_coda_sample_genera_mutant_all_fig,
                  p_duodenum_coda_sample_balances_mutant_fig,
                  p_ileum_coda_sample_genera_mutant_all_fig,
                  p_ileum_coda_sample_balances_mutant_fig,
                  p_cecum_coda_sample_genera_mutant_all_fig,
                  p_cecum_coda_sample_balances_mutant_fig,
                  rel_heights = c(5,1, 5, 1, 5, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
supplemental_figure_5_duodenum_sex_mutant

supplemental_figure_5_duodenum_sex_female <- plot_grid(
                  p_duodenum_coda_sample_genera_female_all_fig,
                  p_duodenum_coda_sample_balances_female_fig,
                  p_ileum_coda_sample_genera_female_all_fig,
                  p_ileum_coda_sample_balances_female_fig,
                  p_cecum_coda_sample_genera_female_all_fig,
                  p_cecum_coda_sample_balances_female_fig,
                  rel_heights = c(5,1, 5, 1, 5, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
supplemental_figure_5_duodenum_sex_female

supplemental_figure_5_duodenum_sex_male <- plot_grid(
                  p_duodenum_coda_sample_genera_male_all_fig,
                  p_duodenum_coda_sample_balances_male_fig,
                  p_ileum_coda_sample_genera_male_all_fig,
                  p_ileum_coda_sample_balances_male_fig,
                  p_cecum_coda_sample_genera_male_all_fig,
                  p_cecum_coda_sample_balances_male_fig,
                  label_size = 12, 
                  rel_heights = c(5,1, 5, 1, 5, 1),
                  align = "v",
                  ncol = 1, nrow = 6)
supplemental_figure_5_duodenum_sex_male

supplemental_figure_5 <- plot_grid(supplemental_figure_5_duodenum_sex_wild_type,
                      supplemental_figure_5_duodenum_sex_mutant,
                      supplemental_figure_5_duodenum_sex_female,
                      supplemental_figure_5_duodenum_sex_male,
                  label_size = 12, 
                  rel_widths = c(1,1,1,1),
                  align = "v",
                  axis = "l",
                  ncol = 4, nrow = 1)

ggsave("./results/environmental_analysis/figures/fig4_balances/supplemental_figure_5.png",supplemental_figure_5, width = 14, height = 18, limitsize = FALSE)

ggsave("./results/environmental_analysis/figures/fig4_balances/supplemental_figure_5.svg",supplemental_figure_5, width = 14, height = 18, limitsize = FALSE)
```

## 5.6 Supplemental Figure 7 Heatmaps- WT sex differences
WT Differences by section
```{r}
intestine_tx_genus_renamed_stacked <- intestine_genus_tx_group_by_group_sample_stack %>% mutate(across('Genus', str_replace, '_group', '')) %>% mutate(across('Genus', str_replace, 'Incertae_Sedis', 'Ruminococcaceae')) %>% mutate(across('Genus', str_replace, '_UCG.00', 'UCG')) %>% mutate(across('Genus', str_replace, 'Methylobacterium.Methylorubrum', 'Methylobacterium'))

#Duodenum WT differences (10p)
intestine_genus_duodenum_interesting_balances_stack_fig <- intestine_tx_genus_renamed_stacked[(intestine_tx_genus_renamed_stacked$Genus %in% coda_balance_coeff_names_duodenum_genus_wild_type_10p$Genus & intestine_tx_genus_renamed_stacked$Sample_Type_Env %in% list("Duodenum_Lumen", "Duodenum_Mucosa")),]
                                                                                                        intestine_genus_duodenum_interesting_balances_stack_fig  <- intestine_genus_duodenum_interesting_balances_stack_fig[(intestine_genus_duodenum_interesting_balances_stack_fig$Group=="Male_Wild_Type" |      intestine_genus_duodenum_interesting_balances_stack_fig$Group=="Female_Wild_Type"),]                  
  
 #ileum WT differences (10p)
intestine_genus_ileum_interesting_balances_stack_fig <- intestine_tx_genus_renamed_stacked[(intestine_tx_genus_renamed_stacked$Genus %in% coda_balance_coeff_names_ileum_genus_wild_type_10p$Genus & intestine_tx_genus_renamed_stacked$Sample_Type_Env %in% list("Ileum_Lumen", "Ileum_Mucosa")),]
                                                                                                        intestine_genus_ileum_interesting_balances_stack_fig  <- intestine_genus_ileum_interesting_balances_stack_fig[(intestine_genus_ileum_interesting_balances_stack_fig$Group=="Male_Wild_Type" |      intestine_genus_ileum_interesting_balances_stack_fig$Group=="Female_Wild_Type"),]  
                                                                                                        
#cecum WT differences (10p)
intestine_genus_cecum_interesting_balances_stack_fig <- intestine_tx_genus_renamed_stacked[(intestine_tx_genus_renamed_stacked$Genus %in% coda_balance_coeff_names_cecum_genus_wild_type_10p$Genus & intestine_tx_genus_renamed_stacked$Sample_Type_Env %in% list("Cecum_Lumen", "Cecum_Mucosa")),]
                                                                                                        intestine_genus_cecum_interesting_balances_stack_fig  <- intestine_genus_cecum_interesting_balances_stack_fig[(intestine_genus_cecum_interesting_balances_stack_fig$Group=="Male_Wild_Type" |      intestine_genus_cecum_interesting_balances_stack_fig$Group=="Female_Wild_Type"),]                  
                                                                                                       
```
```{r}
theme_heatmap_sex_diff_balances <- function(){ 
    font <- "Arial"   #assign font family up front
    theme_classic() %+replace%    #replace elements 
    theme( 
      axis.title.y=element_blank(),
      axis.title.x=element_blank(),
      text = element_text(size = 10, face = "bold"),
      plot.title = element_text(size = 12, hjust = 0.5, face = "bold"),
      plot.margin = unit(c(1, 0.25, 0, 0), "cm"),
      axis.text.x = element_text(angle = 50, vjust = 1, hjust=1),
      legend.position='none',
      axis.text = element_text(face="bold")
      )
}
```
Genera different in WT balances
```{r}
intestine_genus_duodenum_interesting_balances_stack_fig$Genus <- factor(intestine_genus_duodenum_interesting_balances_stack_fig$Genus, levels = c("Acetatifactor","Monoglobus", "Pseudomonas", "Anoxybacillus", "Halomonas", "Oscillospiraceae1", "Lachnoclostridium", "Escherichia.Shigella", "Peptococcaceae1", "Eggerthellaceae" ))

intestine_genus_ileum_interesting_balances_stack_fig$Genus <- factor(intestine_genus_ileum_interesting_balances_stack_fig$Genus, levels = c("Rothia","Hydrogenibacillus", "Muribaculaceae", "Blautia", "Eggerthellaceae", "Escherichia.Shigella" ))

#cecum
intestine_genus_cecum_interesting_balances_stack_fig$Genus <- factor(intestine_genus_cecum_interesting_balances_stack_fig$Genus, levels = c("Oscillibacter", "Roseburia", "Anaerotruncus", "Ruminococcaceae", "Family_XIIIUCG1", "Intestinimonas", "Lachnospiraceae","Lachnoclostridium"))
```


Plot
```{r}
#get legend
p_heatmap_genus_facet_duodenum_interesting_balance_fig_legend <- ggplot(intestine_genus_duodenum_interesting_balances_stack_fig, 
  aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_viridis_c(option = "D", begin = 0.1, end = 0.9) +
  scale_x_discrete(labels = c("Female", "Male")) +
  theme_classic() 
legend_heatmap_clr_genus <- get_legend(p_heatmap_genus_facet_duodenum_interesting_balance_fig_legend)


p_heatmap_genus_facet_duodenum_interesting_balance_fig <- ggplot(intestine_genus_duodenum_interesting_balances_stack_fig, 
  aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_viridis_c(option = "D", begin = 0.1, end = 0.9) +
  scale_x_discrete(labels = c("Female", "Male")) +
  theme_heatmap_sex_diff_balances() +
  facet_wrap(~Sample_Type_Env, nrow = 1) 
p_heatmap_genus_facet_duodenum_interesting_balance_fig


p_heatmap_genus_facet_ileum_interesting_balance_fig <- ggplot(intestine_genus_ileum_interesting_balances_stack_fig, 
  aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_viridis_c(option = "D", begin = 0.1, end = 0.9) +
    scale_x_discrete(labels = c("Female", "Male")) +
  theme_heatmap_sex_diff_balances() +
  facet_wrap(~Sample_Type_Env, nrow = 1) 
p_heatmap_genus_facet_ileum_interesting_balance_fig



p_heatmap_genus_facet_cecum_interesting_balance_fig <- ggplot(intestine_genus_cecum_interesting_balances_stack_fig, 
  aes(Group, Genus, fill= CLR_Abundance)) + 
  geom_tile() +
  scale_fill_viridis_c(option = "D", begin = 0.1, end = 0.9) +
    scale_x_discrete(labels = c("Female", "Male")) +
  theme_heatmap_sex_diff_balances() +
  facet_wrap(~Sample_Type_Env, nrow = 1) 
p_heatmap_genus_facet_cecum_interesting_balance_fig

```

```{r}

supplemental_figure_6 <- plot_grid(
  p_heatmap_genus_facet_duodenum_interesting_balance_fig,
  p_heatmap_genus_facet_ileum_interesting_balance_fig,
  p_heatmap_genus_facet_cecum_interesting_balance_fig,
  legend_heatmap_clr_genus,  
                labels =  c( 'A', 'B', 'C',""),
                     label_size = 12, 
                     align = "h", axis = "tb",
                    rel_widths = c(7,6,6,2),
                     ncol = 4, 
                      nrow = 1)
                
ggsave('./results/environmental_analysis/figures/supplemental_figure_6.png',width=12, height=4, plot = supplemental_figure_6)
ggsave('./results/environmental_analysis/figures/supplemental_figure_6.svg',width=12, height=4, plot = supplemental_figure_6)
```




